{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "10b37d89",
   "metadata": {},
   "source": [
    "# Stepwise Analysis"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "efb74511",
   "metadata": {},
   "source": [
    "Take genes from WTvex9 and WTvex20, then look at the differences between them"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "ee296a20",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "library(\"tidyverse\")\n",
    "library(\"org.Hs.eg.db\")\n",
    "library(\"ggrepel\")\n",
    "library(\"DESeq2\")\n",
    "setwd(\"/data/park_lab/adam/Total_RNA\")\n",
    "load(\"Total_RNA_env.RData\")\n",
    "ls()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1ddb4f65",
   "metadata": {},
   "outputs": [],
   "source": [
    "# save.image(file = \"stepwise.RData\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bd7ad7f4",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "load(\"stepwise.RData\")\n",
    "ls()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "38a61731",
   "metadata": {},
   "source": [
    "## Aggregate"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "db3f6314",
   "metadata": {},
   "outputs": [],
   "source": [
    "WTvex9_genes_log2fc_1.5\n",
    "WTvex20_genes_log2fc_1.5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f54b6ced",
   "metadata": {},
   "outputs": [],
   "source": [
    "WTvex9_selected <- WTvex9_genes_log2fc_1.5 %>% \n",
    "                    dplyr::select(\"X\", \"symbol\", \"entrez\")\n",
    "WTvex20_selected <- WTvex20_genes_log2fc_1.5 %>% \n",
    "                    dplyr::select(\"X\", \"symbol\", \"entrez\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ff1b6628",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "head(WTvex9_selected)\n",
    "head(WTvex20_selected)\n",
    "\n",
    "interx <- WTvex9_selected %>% \n",
    "        filter(X %in% WTvex20_selected$X)\n",
    "interx\n",
    "# 28 genes are shared"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ef2a46a0",
   "metadata": {},
   "outputs": [],
   "source": [
    "WTvex9_unique <- subset(WTvex9_selected, !(X %in% interx$X))\n",
    "WTvex20_unique <- subset(WTvex20_selected, !(X %in% interx$X))\n",
    "\n",
    "WTvex9_unique\n",
    "WTvex20_unique"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "16d7196c",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "total_counts"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2816a633",
   "metadata": {},
   "source": [
    "filter counts by the unique genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b270b3b7",
   "metadata": {},
   "outputs": [],
   "source": [
    "# unique_counts_9 <- total_counts %>% \n",
    "#                 filter(rownames(total_counts) %in% WTvex9_unique$X)\n",
    "# unique_counts_9\n",
    "# unique_counts_20 <- total_counts %>% \n",
    "#                 filter(rownames(total_counts) %in% WTvex20_unique$X)\n",
    "# unique_counts_20\n",
    "# aggregate_unique <- rbind(unique_counts_9, unique_counts_20)\n",
    "# aggregate_unique\n",
    "colnames(aggregate_unique)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1f1d99bd",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Set treatment variables for DESeq\n",
    "genotype <- c(rep(\"E545K\",3), \n",
    "             rep(\"WT\",6), \n",
    "             rep(\"E545K\",3),\n",
    "             rep(\"H1047R\",3),\n",
    "             rep(\"WT\",3),\n",
    "             rep(\"E545K\",3),\n",
    "             rep(\"H1047R\",6))\n",
    "genotype <- as.factor(genotype)\n",
    "coldata <- data.frame(genotype)\n",
    "rownames(coldata) <- colnames(aggregate_unique)\n",
    "all(rownames(coldata) == colnames(aggregate_unique))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c0a4464a",
   "metadata": {},
   "outputs": [],
   "source": [
    "#DESeq Time\n",
    "dds <- DESeqDataSetFromMatrix(countData = aggregate_unique,\n",
    "                              colData = coldata,\n",
    "                              design = ~ genotype)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3ec6cc7b",
   "metadata": {},
   "outputs": [],
   "source": [
    "dds"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7cb693c1",
   "metadata": {},
   "outputs": [],
   "source": [
    "dds <- DESeq(dds)\n",
    "resultsNames(dds)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7ca9a64e",
   "metadata": {},
   "outputs": [],
   "source": [
    "######PCA for sample similarity\n",
    "dds_vst <- varianceStabilizingTransformation(dds, blind = FALSE)\n",
    "plotPCA(dds_vst, intgroup = \"genotype\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "33ed12c0",
   "metadata": {},
   "outputs": [],
   "source": [
    "######## filter by significance\n",
    "######## ## l2fc +/- 1.5 and padj <0.05\n",
    "######## ###129 genes\n",
    "split_res <- results(dds, contrast = c(\"genotype\", \"E545K\", \"H1047R\"))\n",
    "split_res <- as.data.frame(split_res)\n",
    "split_res_df_filt <- split_res %>%\n",
    "  filter(log2FoldChange < -1.5 | log2FoldChange > 1.5) %>%\n",
    "  filter(padj < 0.05)\n",
    "split_res_df_filt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fc5cf4f4",
   "metadata": {},
   "outputs": [],
   "source": [
    "library(tidyverse)\n",
    "library(janitor)\n",
    "library(scales)\n",
    "library(DESeq2)\n",
    "library(dplyr)\n",
    "library(biomaRt)\n",
    "library(ggrepel)\n",
    "# library(clusterProfiler)\n",
    "library(org.Hs.eg.db)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "41144c33",
   "metadata": {},
   "outputs": [],
   "source": [
    "#add gene symbols\n",
    "split_res_df_filt$geneID <- rownames(split_res_df_filt) %>% \n",
    "  tools::file_path_sans_ext()\n",
    "mart <- useDataset(\"hsapiens_gene_ensembl\", useMart(\"ensembl\"))\n",
    "genes <- split_res_df_filt$geneID\n",
    "G_list <- getBM(filters= \"ensembl_gene_id\",\n",
    "                attributes= c(\"ensembl_gene_id\",\"hgnc_symbol\"),\n",
    "                values=genes,mart= mart,\n",
    "               useCache = FALSE)\n",
    "split_res_df_filt <- merge(split_res_df_filt,\n",
    "                               G_list,\n",
    "                               by.x=\"geneID\",\n",
    "                               by.y=\"ensembl_gene_id\")\n",
    "write_tsv(split_res_df_filt, file = \"Stepwise_aggregate_diff.txt\")\n",
    "head(split_res_df_filt)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0962360c",
   "metadata": {},
   "outputs": [],
   "source": [
    "#volcano plot\n",
    "split_res_df_filt$diffexpressed <- \"NO\"\n",
    "split_res_df_filt$diffexpressed[split_res_df_filt$log2FoldChange > 1.5 & split_res_df_filt$padj < 0.05] <- \"E545K\"\n",
    "split_res_df_filt$diffexpressed[split_res_df_filt$log2FoldChange < -1.5 & split_res_df_filt$padj < 0.05] <- \"H1047R\"\n",
    "#volcano plot\n",
    "volcano <- split_res_df_filt %>%\n",
    "  ggplot(aes(x = log2FoldChange, y = -log10(padj), label = hgnc_symbol, col = diffexpressed)) +\n",
    "  geom_point() +\n",
    "  theme_minimal() +\n",
    "  geom_text_repel() +\n",
    "  scale_color_manual(values = c(\"lightpink3\", \"cadetblue\"))\n",
    "volcano"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5e6db16d",
   "metadata": {},
   "outputs": [],
   "source": [
    "#GOBP\n",
    "agg_GOBP <- enrichGO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "95424568",
   "metadata": {},
   "outputs": [],
   "source": [
    "browseVignettes(\"clusterProfiler\")\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4e251ea3",
   "metadata": {},
   "source": [
    "## Individual Lines"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "58e61d56",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Cancer\n",
    "cancer_WTvex9_resSig_padj\n",
    "cancer_WTvex20_resSig_padj\n",
    "\n",
    "#H-Tert\n",
    "HTert_WTvex9_resSig_padj\n",
    "HTert_WTvex20_resSig_padj\n",
    "\n",
    "#MCF10A\n",
    "MCF10A_WTvex9_resSig_padj\n",
    "MCF10A_WTvex20_resSig_padj"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b583a6f6",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "#cancer\n",
    "cancer_WTvex9_selected <- cancer_WTvex9_resSig_padj %>% \n",
    "                    dplyr::select(\"rowname\", \"log2FoldChange\")\n",
    "cancer_WTvex20_selected <- cancer_WTvex20_resSig_padj %>% \n",
    "                    dplyr::select(\"rowname\", \"log2FoldChange\")\n",
    "head(cancer_WTvex9_selected)\n",
    "head(cancer_WTvex20_selected)\n",
    "#H-Tert\n",
    "HTert_WTvex9_selected <- HTert_WTvex9_resSig_padj %>% \n",
    "                    dplyr::select(\"rowname\", \"log2FoldChange\")\n",
    "HTert_WTvex20_selected <- HTert_WTvex20_resSig_padj %>% \n",
    "                    dplyr::select(\"rowname\", \"log2FoldChange\")\n",
    "head(HTert_WTvex9_selected)\n",
    "head(HTert_WTvex20_selected)\n",
    "#MCF10A\n",
    "MCF10A_WTvex9_selected <- MCF10A_WTvex9_resSig_padj %>% \n",
    "                    dplyr::select(\"rowname\", \"log2FoldChange\")\n",
    "MCF10A_WTvex20_selected <- MCF10A_WTvex20_resSig_padj %>% \n",
    "                    dplyr::select(\"rowname\", \"log2FoldChange\")\n",
    "head(MCF10A_WTvex9_selected)\n",
    "head(MCF10A_WTvex20_selected)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "50d69e4b",
   "metadata": {},
   "outputs": [],
   "source": [
    "cancer_interx <- cancer_WTvex9_selected %>% \n",
    "        filter(rowname %in% cancer_WTvex20_selected$rowname)\n",
    "cancer_interx\n",
    "# 548 genes are shared\n",
    "HTert_interx <- HTert_WTvex9_selected %>% \n",
    "        filter(rowname %in% HTert_WTvex20_selected$rowname)\n",
    "HTert_interx\n",
    "#530\n",
    "MCF10A_interx <- MCF10A_WTvex9_selected %>% \n",
    "        filter(rowname %in% MCF10A_WTvex20_selected$rowname)\n",
    "MCF10A_interx\n",
    "#410"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "05d501f4",
   "metadata": {},
   "outputs": [],
   "source": [
    "cancer_WTvex9_unique <- subset(cancer_WTvex9_selected, !(rowname %in% cancer_interx$rowname))\n",
    "cancer_WTvex20_unique <- subset(cancer_WTvex20_selected, !(rowname %in% cancer_interx$rowname))\n",
    "cancer_WTvex9_unique\n",
    "#819\n",
    "cancer_WTvex20_unique\n",
    "#5431\n",
    "HTert_WTvex9_unique <- subset(HTert_WTvex9_selected, !(rowname %in% HTert_interx$rowname))\n",
    "HTert_WTvex20_unique <- subset(HTert_WTvex20_selected, !(rowname %in% HTert_interx$rowname))\n",
    "HTert_WTvex9_unique\n",
    "#1851\n",
    "HTert_WTvex20_unique\n",
    "#1555\n",
    "MCF10A_WTvex9_unique <- subset(MCF10A_WTvex9_selected, !(rowname %in% MCF10A_interx$rowname))\n",
    "MCF10A_WTvex20_unique <- subset(MCF10A_WTvex20_selected, !(rowname %in% MCF10A_interx$rowname))\n",
    "MCF10A_WTvex9_unique\n",
    "#750\n",
    "MCF10A_WTvex20_unique\n",
    "#1684\n",
    "write_tsv(cancer_WTvex9_unique, file = \"cancer_WTvex9_unique.txt\")\n",
    "write_tsv(cancer_WTvex20_unique, file = \"cancer_WTvex20_unique.txt\")\n",
    "write_tsv(HTert_WTvex9_unique, file = \"HTert_WTvex9_unique.txt\")\n",
    "write_tsv(HTert_WTvex20_unique, file = \"HTert_WTvex20_unique.txt\")\n",
    "write_tsv(MCF10A_WTvex9_unique, file = \"MCF10A_WTvex9_unique.txt\")\n",
    "write_tsv(MCF10A_WTvex20_unique, file = \"MCF10A_WTvex20_unique.txt\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f36c42ba",
   "metadata": {},
   "outputs": [],
   "source": [
    "ens.str <-substr(cancer_WTvex9_unique$rowname, 1,15) \n",
    "cancer_WTvex9_unique$symbol <- mapIds(org.Hs.eg.db,\n",
    "                            keys = ens.str,\n",
    "                            column = \"SYMBOL\",\n",
    "                            keytype = \"ENSEMBL\",\n",
    "                            multiVals= \"first\")\n",
    "cancer_WTvex9_unique\n",
    "ens.str <-substr(cancer_WTvex20_unique$rowname, 1,15) \n",
    "cancer_WTvex20_unique$symbol <- mapIds(org.Hs.eg.db,\n",
    "                            keys = ens.str,\n",
    "                            column = \"SYMBOL\",\n",
    "                            keytype = \"ENSEMBL\",\n",
    "                            multiVals= \"first\")\n",
    "cancer_WTvex20_unique\n",
    "ens.str <-substr(HTert_WTvex9_unique$rowname, 1,15) \n",
    "HTert_WTvex9_unique$symbol <- mapIds(org.Hs.eg.db,\n",
    "                            keys = ens.str,\n",
    "                            column = \"SYMBOL\",\n",
    "                            keytype = \"ENSEMBL\",\n",
    "                            multiVals= \"first\")\n",
    "HTert_WTvex9_unique\n",
    "ens.str <-substr(HTert_WTvex20_unique$rowname, 1,15) \n",
    "HTert_WTvex20_unique$symbol <- mapIds(org.Hs.eg.db,\n",
    "                            keys = ens.str,\n",
    "                            column = \"SYMBOL\",\n",
    "                            keytype = \"ENSEMBL\",\n",
    "                            multiVals= \"first\")\n",
    "HTert_WTvex20_unique\n",
    "ens.str <-substr(MCF10A_WTvex9_unique$rowname, 1,15) \n",
    "MCF10A_WTvex9_unique$symbol <- mapIds(org.Hs.eg.db,\n",
    "                            keys = ens.str,\n",
    "                            column = \"SYMBOL\",\n",
    "                            keytype = \"ENSEMBL\",\n",
    "                            multiVals= \"first\")\n",
    "MCF10A_WTvex9_unique\n",
    "ens.str <-substr(MCF10A_WTvex20_unique$rowname, 1,15) \n",
    "MCF10A_WTvex20_unique$symbol <- mapIds(org.Hs.eg.db,\n",
    "                            keys = ens.str,\n",
    "                            column = \"SYMBOL\",\n",
    "                            keytype = \"ENSEMBL\",\n",
    "                            multiVals= \"first\")\n",
    "MCF10A_WTvex20_unique"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "929a8ed1",
   "metadata": {},
   "outputs": [],
   "source": [
    "cancer_WTvex9_unique <- na.omit(cancer_WTvex9_unique)\n",
    "cancer_WTvex20_unique <- na.omit(cancer_WTvex20_unique)\n",
    "HTert_WTvex9_unique <- na.omit(HTert_WTvex9_unique)\n",
    "HTert_WTvex20_unique <- na.omit(HTert_WTvex20_unique)\n",
    "MCF10A_WTvex9_unique <- na.omit(MCF10A_WTvex9_unique)\n",
    "MCF10A_WTvex20_unique <- na.omit(MCF10A_WTvex20_unique)\n",
    "cancer_WTvex9_unique\n",
    "cancer_WTvex20_unique\n",
    "HTert_WTvex9_unique\n",
    "HTert_WTvex20_unique\n",
    "MCF10A_WTvex9_unique\n",
    "MCF10A_WTvex20_unique\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "61da9b6c",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "colnames(total_counts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e9f2cfee",
   "metadata": {},
   "outputs": [],
   "source": [
    "cancer_counts <- total_counts %>%\n",
    "                dplyr::select(\"MCF7_corrected_rep1\", \"MCF7_corrected_rep2\", \"MCF7_corrected_rep3\",\n",
    "                             \"MCF7_parental_rep1\", \"MCF7_parental_rep2\", \"MCF7_parental_rep3\",\n",
    "                             \"T47D_rep1\", \"T47D_rep2\", \"T47D_rep3\")\n",
    "HTert_counts <- total_counts %>%\n",
    "                dplyr::select(\"HTert_WT_rep1\", \"HTert_WT_rep2\", \"HTert_WT_rep3\",\n",
    "                             \"HTert_ex9_rep1\", \"HTert_ex9_rep2\", \"HTert_ex9_rep3\",\n",
    "                             \"HTert_ex20_rep1\", \"HTert_ex20_rep2\", \"HTert_ex20_rep3\")\n",
    "MCF10A_counts <- total_counts %>%\n",
    "                dplyr::select(\"MCF10A_parental_rep1\", \"MCF10A_parental_rep2\", \"MCF10A_parental_rep3\",\n",
    "                             \"MCF10A_E545K_rep1\", \"MCF10A_E545K_rep2\", \"MCF10A_E545K_rep3\",\n",
    "                             \"MCF10A_H1047R_rep1\", \"MCF10A_H1047R_rep2\", \"MCF10A_H1047R_rep3\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f404de9a",
   "metadata": {},
   "outputs": [],
   "source": [
    "cancer_unique_counts_9 <- cancer_counts %>% \n",
    "                filter(rownames(cancer_counts) %in% cancer_WTvex9_unique$rowname)\n",
    "cancer_unique_counts_9\n",
    "cancer_unique_counts_20 <- cancer_counts %>% \n",
    "                filter(rownames(cancer_counts) %in% cancer_WTvex20_unique$rowname)\n",
    "cancer_unique_counts_20\n",
    "cancer_unique_counts <- rbind(cancer_unique_counts_9, cancer_unique_counts_20)\n",
    "cancer_unique_counts\n",
    "\n",
    "HTert_unique_counts_9 <- HTert_counts %>% \n",
    "                filter(rownames(cancer_counts) %in% HTert_WTvex9_unique$rowname)\n",
    "HTert_unique_counts_9\n",
    "HTert_unique_counts_20 <- HTert_counts %>% \n",
    "                filter(rownames(HTert_counts) %in% HTert_WTvex20_unique$rowname)\n",
    "HTert_unique_counts_20\n",
    "HTert_unique_counts <- rbind(HTert_unique_counts_9, HTert_unique_counts_20)\n",
    "HTert_unique_counts\n",
    "\n",
    "MCF10A_unique_counts_9 <- MCF10A_counts %>% \n",
    "                filter(rownames(MCF10A_counts) %in% MCF10A_WTvex9_unique$rowname)\n",
    "MCF10A_unique_counts_9\n",
    "MCF10A_unique_counts_20 <- MCF10A_counts %>% \n",
    "                filter(rownames(MCF10A_counts) %in% MCF10A_WTvex20_unique$rowname)\n",
    "MCF10A_unique_counts_20\n",
    "MCF10A_unique_counts <- rbind(MCF10A_unique_counts_9, MCF10A_unique_counts_20)\n",
    "MCF10A_unique_counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5a8b4ae5",
   "metadata": {},
   "outputs": [],
   "source": [
    "genotype <- c(rep(\"WT\",3), \n",
    "             rep(\"E545K\",3),\n",
    "             rep(\"H1047R\",3))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6f386493",
   "metadata": {},
   "outputs": [],
   "source": [
    "genotype <- as.factor(genotype)\n",
    "cancer_coldata <- data.frame(genotype)\n",
    "HTert_coldata <- data.frame(genotype)\n",
    "MCF10A_coldata <- data.frame(genotype)\n",
    "\n",
    "rownames(cancer_coldata) <- colnames(cancer_unique_counts)\n",
    "all(rownames(cancer_coldata) == colnames(cancer_unique_counts))\n",
    "\n",
    "rownames(HTert_coldata) <- colnames(HTert_unique_counts)\n",
    "all(rownames(HTert_coldata) == colnames(HTert_unique_counts))\n",
    "\n",
    "rownames(MCF10A_coldata) <- colnames(MCF10A_unique_counts)\n",
    "all(rownames(MCF10A_coldata) == colnames(MCF10A_unique_counts))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3f2d84b2",
   "metadata": {},
   "outputs": [],
   "source": [
    "#DESeq Time\n",
    "cancer_dds <- DESeqDataSetFromMatrix(countData = cancer_unique_counts,\n",
    "                              colData = cancer_coldata,\n",
    "                              design = ~ genotype)\n",
    "HTert_dds <- DESeqDataSetFromMatrix(countData = HTert_unique_counts,\n",
    "                              colData = HTert_coldata,\n",
    "                              design = ~ genotype)\n",
    "MCF10A_dds <- DESeqDataSetFromMatrix(countData = MCF10A_unique_counts,\n",
    "                              colData = MCF10A_coldata,\n",
    "                              design = ~ genotype)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0de78138",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "cancer_dds <- DESeq(cancer_dds)\n",
    "resultsNames(cancer_dds)\n",
    "HTert_dds <- DESeq(HTert_dds)\n",
    "resultsNames(HTert_dds)\n",
    "MCF10A_dds <- DESeq(MCF10A_dds)\n",
    "resultsNames(MCF10A_dds)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "42a7b8b9",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "cancer_dds_vst <- varianceStabilizingTransformation(cancer_dds, blind = FALSE)\n",
    "plotPCA(cancer_dds_vst, intgroup = \"genotype\")\n",
    "\n",
    "HTert_dds_vst <- varianceStabilizingTransformation(HTert_dds, blind = FALSE)\n",
    "plotPCA(HTert_dds_vst, intgroup = \"genotype\")\n",
    "\n",
    "MCF10A_dds_vst <- varianceStabilizingTransformation(MCF10A_dds, blind = FALSE)\n",
    "plotPCA(MCF10A_dds_vst, intgroup = \"genotype\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e28af92e",
   "metadata": {},
   "outputs": [],
   "source": [
    "######## filter by significance\n",
    "######## ## l2fc +/- 1.5 and padj <0.05\n",
    "######## ###129 genes\n",
    "cancer_split_res <- results(cancer_dds, contrast = c(\"genotype\", \"E545K\", \"H1047R\"))\n",
    "cancer_split_res <- as.data.frame(cancer_split_res)\n",
    "cancer_split_res_df_filt <- cancer_split_res %>%\n",
    "  filter(log2FoldChange < -1.5 | log2FoldChange > 1.5) %>%\n",
    "  filter(padj < 0.05)\n",
    "cancer_split_res_df_filt\n",
    "\n",
    "HTert_split_res <- results(HTert_dds, contrast = c(\"genotype\", \"E545K\", \"H1047R\"))\n",
    "HTert_split_res <- as.data.frame(HTert_split_res)\n",
    "HTert_split_res_df_filt <- HTert_split_res %>%\n",
    "  filter(log2FoldChange < -1.5 | log2FoldChange > 1.5) %>%\n",
    "  filter(padj < 0.05)\n",
    "HTert_split_res_df_filt\n",
    "\n",
    "MCF10A_split_res <- results(MCF10A_dds, contrast = c(\"genotype\", \"E545K\", \"H1047R\"))\n",
    "MCF10A_split_res <- as.data.frame(MCF10A_split_res)\n",
    "MCF10A_split_res_df_filt <- MCF10A_split_res %>%\n",
    "  filter(log2FoldChange < -1.5 | log2FoldChange > 1.5) %>%\n",
    "  filter(padj < 0.05)\n",
    "MCF10A_split_res_df_filt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a96f26e9",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "#add gene symbols\n",
    "cancer_split_res_df_filt$geneID <- rownames(cancer_split_res_df_filt) %>% \n",
    "  tools::file_path_sans_ext()\n",
    "mart <- useDataset(\"hsapiens_gene_ensembl\", useMart(\"ensembl\"))\n",
    "genes <- cancer_split_res_df_filt$geneID\n",
    "G_list <- getBM(filters= \"ensembl_gene_id\",\n",
    "                attributes= c(\"ensembl_gene_id\",\"hgnc_symbol\"),\n",
    "                values=genes,mart= mart,\n",
    "               useCache = FALSE)\n",
    "cancer_split_res_df_filt <- merge(cancer_split_res_df_filt,\n",
    "                               G_list,\n",
    "                               by.x=\"geneID\",\n",
    "                               by.y=\"ensembl_gene_id\")\n",
    "write_tsv(cancer_split_res_df_filt, file = \"cancer_Stepwise_aggregate_diff.txt\")\n",
    "head(cancer_split_res_df_filt)\n",
    "\n",
    "HTert_split_res_df_filt$geneID <- rownames(HTert_split_res_df_filt) %>% \n",
    "  tools::file_path_sans_ext()\n",
    "mart <- useDataset(\"hsapiens_gene_ensembl\", useMart(\"ensembl\"))\n",
    "genes <- HTert_split_res_df_filt$geneID\n",
    "G_list <- getBM(filters= \"ensembl_gene_id\",\n",
    "                attributes= c(\"ensembl_gene_id\",\"hgnc_symbol\"),\n",
    "                values=genes,mart= mart,\n",
    "               useCache = FALSE)\n",
    "HTert_split_res_df_filt <- merge(HTert_split_res_df_filt,\n",
    "                               G_list,\n",
    "                               by.x=\"geneID\",\n",
    "                               by.y=\"ensembl_gene_id\")\n",
    "write_tsv(HTert_split_res_df_filt, file = \"HTert_Stepwise_aggregate_diff.txt\")\n",
    "head(HTert_split_res_df_filt)\n",
    "\n",
    "MCF10A_split_res_df_filt$geneID <- rownames(MCF10A_split_res_df_filt) %>% \n",
    "  tools::file_path_sans_ext()\n",
    "mart <- useDataset(\"hsapiens_gene_ensembl\", useMart(\"ensembl\"))\n",
    "genes <- MCF10A_split_res_df_filt$geneID\n",
    "G_list <- getBM(filters= \"ensembl_gene_id\",\n",
    "                attributes= c(\"ensembl_gene_id\",\"hgnc_symbol\"),\n",
    "                values=genes,mart= mart,\n",
    "               useCache = FALSE)\n",
    "MCF10A_split_res_df_filt <- merge(MCF10A_split_res_df_filt,\n",
    "                               G_list,\n",
    "                               by.x=\"geneID\",\n",
    "                               by.y=\"ensembl_gene_id\")\n",
    "write_tsv(MCF10A_split_res_df_filt, file = \"MCF10A_Stepwise_aggregate_diff.txt\")\n",
    "head(MCF10A_split_res_df_filt)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ec5a4edb",
   "metadata": {},
   "outputs": [],
   "source": [
    "#volcano plot\n",
    "cancer_split_res_df_filt$diffexpressed <- \"NO\"\n",
    "cancer_split_res_df_filt$diffexpressed[cancer_split_res_df_filt$log2FoldChange > 1.5 & cancer_split_res_df_filt$padj < 0.05] <- \"E545K\"\n",
    "cancer_split_res_df_filt$diffexpressed[cancer_split_res_df_filt$log2FoldChange < -1.5 & cancer_split_res_df_filt$padj < 0.05] <- \"H1047R\"\n",
    "#volcano plot\n",
    "volcano <- cancer_split_res_df_filt %>%\n",
    "  ggplot(aes(x = log2FoldChange, y = -log10(padj), label = hgnc_symbol, col = diffexpressed)) +\n",
    "  geom_point() +\n",
    "  theme_minimal() +\n",
    "  geom_text_repel() +\n",
    "  scale_color_manual(values = c(\"lightpink3\", \"cadetblue\")) +\n",
    "    ggtitle(label = \"cancer ex9 v ex20 from stepwise analysis\")\n",
    "volcano"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "308a72f3",
   "metadata": {},
   "outputs": [],
   "source": [
    "#volcano plot\n",
    "HTert_split_res_df_filt$diffexpressed <- \"NO\"\n",
    "HTert_split_res_df_filt$diffexpressed[HTert_split_res_df_filt$log2FoldChange > 1.5 & HTert_split_res_df_filt$padj < 0.05] <- \"E545K\"\n",
    "HTert_split_res_df_filt$diffexpressed[HTert_split_res_df_filt$log2FoldChange < -1.5 & HTert_split_res_df_filt$padj < 0.05] <- \"H1047R\"\n",
    "#volcano plot\n",
    "volcano <- HTert_split_res_df_filt %>%\n",
    "  ggplot(aes(x = log2FoldChange, y = -log10(padj), label = hgnc_symbol, col = diffexpressed)) +\n",
    "  geom_point() +\n",
    "  theme_minimal() +\n",
    "  geom_text_repel() +\n",
    "  scale_color_manual(values = c(\"lightpink3\", \"cadetblue\")) + \n",
    "    ggtitle(label = \"HTert ex9 v ex20 from stepwise analysis\")\n",
    "volcano"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1f76474a",
   "metadata": {},
   "outputs": [],
   "source": [
    "#volcano plot\n",
    "MCF10A_split_res_df_filt$diffexpressed <- \"NO\"\n",
    "MCF10A_split_res_df_filt$diffexpressed[MCF10A_split_res_df_filt$log2FoldChange > 1.5 & MCF10A_split_res_df_filt$padj < 0.05] <- \"E545K\"\n",
    "MCF10A_split_res_df_filt$diffexpressed[MCF10A_split_res_df_filt$log2FoldChange < -1.5 & MCF10A_split_res_df_filt$padj < 0.05] <- \"H1047R\"\n",
    "#volcano plot\n",
    "volcano <- MCF10A_split_res_df_filt %>%\n",
    "  ggplot(aes(x = log2FoldChange, y = -log10(padj), label = hgnc_symbol, col = diffexpressed)) +\n",
    "  geom_point() +\n",
    "  theme_minimal() +\n",
    "  geom_text_repel() +\n",
    "  scale_color_manual(values = c(\"lightpink3\", \"cadetblue\")) + \n",
    "    ggtitle(label = \"MCF10A ex9 v ex20 from stepwise analysis\")\n",
    "volcano"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "42700197",
   "metadata": {},
   "source": [
    "### Shared"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "186dcd6d",
   "metadata": {},
   "outputs": [],
   "source": [
    "shared_split_res_df_filt <- cancer_split_res_df_filt %>% \n",
    "                filter(geneID %in% HTert_split_res_df_filt$geneID)\n",
    "shared_split_res_df_filt <- shared_split_res_df_filt %>%\n",
    "                filter(geneID %in% MCF10A_split_res_df_filt$geneID)\n",
    "shared_split_res_df_filt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c1adbc3b",
   "metadata": {},
   "outputs": [],
   "source": [
    "write_tsv(shared_split_res_df_filt, file = \"shared_genes_step.txt\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8678d808",
   "metadata": {},
   "outputs": [],
   "source": [
    "volcano <- shared_split_res_df_filt %>%\n",
    "  ggplot(aes(x = log2FoldChange, y = -log10(padj), label = hgnc_symbol, col = diffexpressed)) +\n",
    "  geom_point() +\n",
    "  theme_minimal() +\n",
    "  geom_text_repel() +\n",
    "  scale_color_manual(values = c(\"lightpink3\", \"cadetblue\")) + \n",
    "    ggtitle(label = \"Shared ex9 v ex20 from stepwise analysis\")\n",
    "volcano"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4e345f41",
   "metadata": {},
   "outputs": [],
   "source": [
    "pdf(file = \"Volcanoes\",\n",
    "   width = 8,\n",
    "   height = 6)\n",
    "options(repr.plot.width=8, repr.plot.height=6)\n",
    "volcano <- cancer_split_res_df_filt %>%\n",
    "  ggplot(aes(x = log2FoldChange, y = -log10(padj), label = hgnc_symbol, col = diffexpressed)) +\n",
    "  geom_point() +\n",
    "  theme_minimal() +\n",
    "  geom_text_repel() +\n",
    "  scale_color_manual(values = c(\"lightpink3\", \"cadetblue\")) +\n",
    "    ggtitle(label = \"cancer ex9 v ex20 from stepwise analysis\")\n",
    "volcano\n",
    "volcano <- HTert_split_res_df_filt %>%\n",
    "  ggplot(aes(x = log2FoldChange, y = -log10(padj), label = hgnc_symbol, col = diffexpressed)) +\n",
    "  geom_point() +\n",
    "  theme_minimal() +\n",
    "  geom_text_repel() +\n",
    "  scale_color_manual(values = c(\"lightpink3\", \"cadetblue\")) + \n",
    "    ggtitle(label = \"HTert ex9 v ex20 from stepwise analysis\")\n",
    "volcano\n",
    "volcano <- MCF10A_split_res_df_filt %>%\n",
    "  ggplot(aes(x = log2FoldChange, y = -log10(padj), label = hgnc_symbol, col = diffexpressed)) +\n",
    "  geom_point() +\n",
    "  theme_minimal() +\n",
    "  geom_text_repel() +\n",
    "  scale_color_manual(values = c(\"lightpink3\", \"cadetblue\")) + \n",
    "    ggtitle(label = \"MCF10A ex9 v ex20 from stepwise analysis\")\n",
    "volcano\n",
    "volcano <- shared_split_res_df_filt %>%\n",
    "  ggplot(aes(x = log2FoldChange, y = -log10(padj), label = hgnc_symbol, col = diffexpressed)) +\n",
    "  geom_point() +\n",
    "  theme_minimal() +\n",
    "  geom_text_repel() +\n",
    "  scale_color_manual(values = c(\"lightpink3\", \"cadetblue\")) + \n",
    "    ggtitle(label = \"Shared ex9 v ex20 from stepwise analysis\")\n",
    "volcano\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "93788333",
   "metadata": {},
   "outputs": [],
   "source": [
    "cd /data/park_lab/adam/Total_RNA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5638ef6c",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "cat Moar_analysis.R"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3a1e2c91",
   "metadata": {},
   "source": [
    "#### Shared Directionality"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fe60debd",
   "metadata": {},
   "outputs": [],
   "source": [
    "head(cancer_split_res_df_filt)\n",
    "head(HTert_split_res_df_filt)\n",
    "head(MCF10A_split_res_df_filt)\n",
    "\n",
    "# shared_split_res_df_filt <- cancer_split_res_df_filt %>% \n",
    "#                 filter(rownames(cancer_split_res_df_filt) %in% rownames(HTert_split_res_df_filt))\n",
    "# shared_split_res_df_filt <- shared_split_res_df_filt %>%\n",
    "#                 filter(rownames(shared_split_res_df_filt) %in% rownames(MCF10A_split_res_df_filt))\n",
    "# shared_split_res_df_filt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c1563eb7",
   "metadata": {},
   "outputs": [],
   "source": [
    "cancer_split_res_df_filt_9 <- cancer_split_res_df_filt %>%\n",
    "                                dplyr::filter(log2FoldChange > 0)\n",
    "cancer_split_res_df_filt_20 <- cancer_split_res_df_filt %>%\n",
    "                                dplyr::filter(log2FoldChange < 0)\n",
    "\n",
    "HTert_split_res_df_filt_9 <- HTert_split_res_df_filt %>%\n",
    "                                dplyr::filter(log2FoldChange > 0)\n",
    "HTert_split_res_df_filt_20 <- HTert_split_res_df_filt %>%\n",
    "                                dplyr::filter(log2FoldChange < 0)\n",
    "\n",
    "MCF10A_split_res_df_filt_9 <- MCF10A_split_res_df_filt %>%\n",
    "                                dplyr::filter(log2FoldChange > 0)\n",
    "MCF10A_split_res_df_filt_20 <- MCF10A_split_res_df_filt %>%\n",
    "                                dplyr::filter(log2FoldChange < 0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1c585a25",
   "metadata": {},
   "outputs": [],
   "source": [
    "cancer_split_res_df_filt_9\n",
    "cancer_split_res_df_filt_20\n",
    "HTert_split_res_df_filt_9\n",
    "HTert_split_res_df_filt_20\n",
    "MCF10A_split_res_df_filt_9\n",
    "MCF10A_split_res_df_filt_20"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b5700d6f",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "shared_split_res_df_filt_9 <- cancer_split_res_df_filt_9 %>% \n",
    "                filter(geneID %in% HTert_split_res_df_filt_9$geneID)\n",
    "shared_split_res_df_filt_9 <- shared_split_res_df_filt_9 %>%\n",
    "                filter(geneID %in% MCF10A_split_res_df_filt_9$geneID)\n",
    "shared_split_res_df_filt_9"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7683c072",
   "metadata": {},
   "outputs": [],
   "source": [
    "shared_split_res_df_filt_20 <- cancer_split_res_df_filt_20 %>% \n",
    "                filter(geneID %in% HTert_split_res_df_filt_20$geneID)\n",
    "shared_split_res_df_filt_20 <- shared_split_res_df_filt_9 %>%\n",
    "                filter(geneID %in% MCF10A_split_res_df_filt_20$geneID)\n",
    "shared_split_res_df_filt_20"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "571b0fcc",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "f174eae7",
   "metadata": {},
   "source": [
    "## TGFB Specific Analysis of RNA-seq data"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3bead793",
   "metadata": {},
   "source": [
    "The goal is to filter the RNA-seq results for TGFB associated genes and perform dimensionality reduction to ideally stratify data by mutation status."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2da4ce80",
   "metadata": {},
   "source": [
    "Set directory to folder with RNA-seq environment from laptop."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a4c99f68",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "library(\"tidyverse\")\n",
    "library(\"org.Hs.eg.db\")\n",
    "library(\"ggrepel\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6434da0d",
   "metadata": {},
   "outputs": [],
   "source": [
    "setwd(\"/data/park_lab/adam/Total_RNA\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "95f87fa8",
   "metadata": {},
   "source": [
    "Load that environment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "923995e9",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "load(\"Total_RNA_env.RData\")\n",
    "ls()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "63713e42",
   "metadata": {},
   "outputs": [],
   "source": [
    "head(total_counts_t)\n",
    "head(normalized_total_counts_t)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "abccea53",
   "metadata": {},
   "source": [
    "Read in the list of genes from the TGFB pathway (GO: 0071559)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "552b66fe",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "TGFB_genes <- read_tsv(file = \"tgfb_gene_list_GO_0071559.txt\")\n",
    "head(TGFB_genes)\n",
    "TGFB_genes <- rbind(names(TGFB_genes), TGFB_genes)\n",
    "names(TGFB_genes) <- c(\"UniProtID\", \"Symbol\")\n",
    "head(TGFB_genes)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "30faf5f1",
   "metadata": {},
   "source": [
    "Add ENSEMBL IDs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9b653c55",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "symbol.str <- substr(TGFB_genes$Symbol, 1,15)\n",
    "symbol.str\n",
    "TGFB_genes$Ensembl <- mapIds(org.Hs.eg.db,\n",
    "                            keys = symbol.str,\n",
    "                            column = \"ENSEMBL\",\n",
    "                            keytype = \"SYMBOL\",\n",
    "                            multiVals= \"first\")\n",
    "head(TGFB_genes)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f0d8a4ee",
   "metadata": {},
   "source": [
    "### Raw Counts"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "50ab3b5c",
   "metadata": {},
   "source": [
    "Remove the Transcript ID from the Ensembl IDs in the counts data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6701f377",
   "metadata": {},
   "outputs": [],
   "source": [
    "total_counts_t <- t(total_counts)\n",
    "head(total_counts_t)\n",
    "good_names <- colnames(total_counts_t)\n",
    "good_names <- tools::file_path_sans_ext(c(good_names))\n",
    "good_names\n",
    "colnames(total_counts_t) <- good_names\n",
    "head(total_counts_t)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "40b9afdf",
   "metadata": {},
   "source": [
    "Filter the counts data by the TGFB genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "94c881dc",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "total_counts_t <- as.data.frame(total_counts_t)\n",
    "TGFB_ensembl <- TGFB_genes$Ensembl %>%\n",
    "                as.character()\n",
    "TGFB_ensembl\n",
    "TGFB_ensembl <- na.omit(TGFB_ensembl)\n",
    "TGFB_ensembl <- as.character(TGFB_ensembl)\n",
    "str(TGFB_ensembl)\n",
    "TGFB_counts <- total_counts_t[, (colnames(total_counts_t) %in% c(TGFB_ensembl))]\n",
    "head(TGFB_counts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0395be46",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "ens.str <- substr(colnames(TGFB_counts), 1,15)\n",
    "genes_for_pca <- mapIds(org.Hs.eg.db,\n",
    "                            keys = ens.str,\n",
    "                            column = \"SYMBOL\",\n",
    "                            keytype = \"ENSEMBL\",\n",
    "                            multiVals= \"first\")\n",
    "str(genes_for_pca)\n",
    "genes_for_pca <- as.character(genes_for_pca)\n",
    "genes_for_pca\n",
    "colnames(TGFB_counts) <- genes_for_pca\n",
    "colnames(TGFB_counts)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a2b429db",
   "metadata": {},
   "source": [
    "Principal Components"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a31ee8f9",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c56f807c",
   "metadata": {},
   "outputs": [],
   "source": [
    "library(\"factoextra\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8e5fcbb9",
   "metadata": {},
   "outputs": [],
   "source": [
    "res.pca <- prcomp(TGFB_counts, scale = TRUE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "33c26096",
   "metadata": {},
   "outputs": [],
   "source": [
    "fviz_eig(res.pca)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1f123d11",
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "fviz_pca_ind(res.pca,\n",
    "             col.ind = \"cos2\", # Color by the quality of representation\n",
    "             gradient.cols = c(\"#00AFBB\", \"#E7B800\", \"#FC4E07\"),\n",
    "             repel = TRUE     # Avoid text overlapping\n",
    "             )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a28e9f0e",
   "metadata": {},
   "outputs": [],
   "source": [
    "fviz_pca_var(res.pca,\n",
    "             col.var = \"contrib\", # Color by contributions to the PC\n",
    "             gradient.cols = c(\"#00AFBB\", \"#E7B800\", \"#FC4E07\"),\n",
    "             repel = FALSE     # Avoid text overlapping\n",
    "             )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9078a39f",
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "fviz_pca_biplot(res.pca, repel = FALSE,\n",
    "                col.var = \"#2E9FDF\", # Variables color\n",
    "                col.ind = \"#696969\"  # Individuals color\n",
    "                )"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "85a4523f",
   "metadata": {},
   "source": [
    "### Normalized Counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cc5a43a4",
   "metadata": {},
   "outputs": [],
   "source": [
    "head(normalized_total_counts_t)\n",
    "norm_names <- colnames(normalized_total_counts_t)\n",
    "norm_names <- tools::file_path_sans_ext(c(norm_names))\n",
    "norm_names\n",
    "colnames(normalized_total_counts_t) <- norm_names\n",
    "head(normalized_total_counts_t)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "39d0ac76",
   "metadata": {},
   "source": [
    "Filter for TGFB genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3277a523",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "TGFB_ensembl <- TGFB_genes$Ensembl %>%\n",
    "                as.character()\n",
    "TGFB_ensembl\n",
    "TGFB_ensembl <- na.omit(TGFB_ensembl)\n",
    "TGFB_ensembl <- as.character(TGFB_ensembl)\n",
    "str(TGFB_ensembl)\n",
    "TGFB_normCounts <- normalized_total_counts_t[, (colnames(normalized_total_counts_t) %in% c(TGFB_ensembl))]\n",
    "head(TGFB_normCounts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cf104e21",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "ens.str <- substr(colnames(TGFB_normCounts), 1,15)\n",
    "genes_for_pca <- mapIds(org.Hs.eg.db,\n",
    "                            keys = ens.str,\n",
    "                            column = \"SYMBOL\",\n",
    "                            keytype = \"ENSEMBL\",\n",
    "                            multiVals= \"first\")\n",
    "str(genes_for_pca)\n",
    "genes_for_pca <- as.character(genes_for_pca)\n",
    "genes_for_pca\n",
    "colnames(TGFB_normCounts) <- genes_for_pca\n",
    "colnames(TGFB_normCounts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "95fe6387",
   "metadata": {},
   "outputs": [],
   "source": [
    "res.Normpca <- prcomp(TGFB_normCounts, scale = TRUE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "318d506b",
   "metadata": {},
   "outputs": [],
   "source": [
    "fviz_eig(res.Normpca)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "efb8de54",
   "metadata": {},
   "outputs": [],
   "source": [
    "fviz_pca_ind(res.Normpca,\n",
    "             col.ind = \"cos2\", # Color by the quality of representation\n",
    "             gradient.cols = c(\"#00AFBB\", \"#E7B800\", \"#FC4E07\"),\n",
    "             repel = TRUE     # Avoid text overlapping\n",
    "             )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c7761584",
   "metadata": {},
   "outputs": [],
   "source": [
    "fviz_pca_var(res.Normpca,\n",
    "             col.var = \"contrib\", # Color by contributions to the PC\n",
    "             gradient.cols = c(\"#00AFBB\", \"#E7B800\", \"#FC4E07\"),\n",
    "             repel = TRUE     # Avoid text overlapping\n",
    "             )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "30ed3055",
   "metadata": {},
   "outputs": [],
   "source": [
    "fviz_pca_biplot(res.Normpca, repel = TRUE,\n",
    "                col.var = \"#2E9FDF\", # Variables color\n",
    "                col.ind = \"#696969\"  # Individuals color\n",
    "                )"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "40f258d5",
   "metadata": {},
   "source": [
    "## Hippo Pathway"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bc9282bf",
   "metadata": {},
   "source": [
    "Read in list of hippo pathway genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "53d47c6d",
   "metadata": {},
   "outputs": [],
   "source": [
    "hippo_genes <- read_tsv(file = \"hippo_gene_list_GO_0035329.txt\")\n",
    "head(hippo_genes)\n",
    "hippo_genes <- rbind(names(hippo_genes), hippo_genes)\n",
    "names(hippo_genes) <- c(\"UniProtID\", \"Name\", \"Symbol\")\n",
    "head(hippo_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cced82a7",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "symbol.str <- substr(hippo_genes$Symbol, 1,15)\n",
    "symbol.str\n",
    "hippo_genes$Ensembl <- mapIds(org.Hs.eg.db,\n",
    "                            keys = symbol.str,\n",
    "                            column = \"ENSEMBL\",\n",
    "                            keytype = \"SYMBOL\",\n",
    "                            multiVals= \"first\")\n",
    "hippo_genes"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "94da060d",
   "metadata": {},
   "source": [
    "Filter the raw and normalized counts by the hippo pathway"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e0043746",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "total_counts_t <- as.data.frame(total_counts_t)\n",
    "hippo_ensembl <- hippo_genes$Ensembl %>%\n",
    "                as.character()\n",
    "hippo_ensembl\n",
    "hippo_ensembl <- na.omit(hippo_ensembl)\n",
    "hippo_ensembl <- as.character(hippo_ensembl)\n",
    "str(hippo_ensembl)\n",
    "hippo_counts <- total_counts_t[, (colnames(total_counts_t) %in% c(hippo_ensembl))]\n",
    "head(hippo_counts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2d9183f0",
   "metadata": {},
   "outputs": [],
   "source": [
    "hippo_normCounts <- normalized_total_counts_t[, (colnames(normalized_total_counts_t) %in% \n",
    "                                                c(hippo_ensembl))]\n",
    "head(hippo_normCounts)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "945a4983",
   "metadata": {},
   "source": [
    "Add back the gene names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bee9a278",
   "metadata": {},
   "outputs": [],
   "source": [
    "ens.str <- substr(colnames(hippo_counts), 1,15)\n",
    "genes_for_pca <- mapIds(org.Hs.eg.db,\n",
    "                            keys = ens.str,\n",
    "                            column = \"SYMBOL\",\n",
    "                            keytype = \"ENSEMBL\",\n",
    "                            multiVals= \"first\")\n",
    "str(genes_for_pca)\n",
    "genes_for_pca <- as.character(genes_for_pca)\n",
    "genes_for_pca\n",
    "colnames(hippo_counts) <- genes_for_pca\n",
    "colnames(hippo_counts)\n",
    "\n",
    "ens.str <- substr(colnames(hippo_normCounts), 1,15)\n",
    "genes_for_pca <- mapIds(org.Hs.eg.db,\n",
    "                            keys = ens.str,\n",
    "                            column = \"SYMBOL\",\n",
    "                            keytype = \"ENSEMBL\",\n",
    "                            multiVals= \"first\")\n",
    "str(genes_for_pca)\n",
    "genes_for_pca <- as.character(genes_for_pca)\n",
    "genes_for_pca\n",
    "colnames(hippo_normCounts) <- genes_for_pca\n",
    "colnames(hippo_normCounts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "41f74e4e",
   "metadata": {},
   "outputs": [],
   "source": [
    "hippo_res.pca <- prcomp(hippo_counts, scale = TRUE)\n",
    "hippo_normres.pca <- prcomp(hippo_normCounts, scale = TRUE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "22176fea",
   "metadata": {},
   "outputs": [],
   "source": [
    "fviz_pca_ind(hippo_res.pca,\n",
    "             col.ind = \"cos2\", # Color by the quality of representation\n",
    "             gradient.cols = c(\"#00AFBB\", \"#E7B800\", \"#FC4E07\"),\n",
    "             repel = TRUE     # Avoid text overlapping\n",
    "             )\n",
    "### Raw Counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "33e30c49",
   "metadata": {},
   "outputs": [],
   "source": [
    "fviz_pca_ind(hippo_normres.pca,\n",
    "             col.ind = \"cos2\", # Color by the quality of representation\n",
    "             gradient.cols = c(\"#00AFBB\", \"#E7B800\", \"#FC4E07\"),\n",
    "             repel = TRUE     # Avoid text overlapping\n",
    "             )\n",
    "### Normalized Counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "998b0985",
   "metadata": {},
   "outputs": [],
   "source": [
    "fviz_pca_biplot(hippo_res.pca, repel = TRUE,\n",
    "                col.var = \"#2E9FDF\", # Variables color\n",
    "                col.ind = \"#696969\"  # Individuals color\n",
    "                )\n",
    "fviz_pca_biplot(hippo_normres.pca, repel = TRUE,\n",
    "                col.var = \"#2E9FDF\", # Variables color\n",
    "                col.ind = \"#696969\"  # Individuals color\n",
    "                )"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4e8edeeb",
   "metadata": {},
   "source": [
    "## AP-1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f604eb34",
   "metadata": {},
   "outputs": [],
   "source": [
    "ap1_genes <- read_tsv(file = \"ap1_gene_list_GO_0035976.txt\")\n",
    "head(ap1_genes)\n",
    "hippo_genes <- rbind(names(ap1_genes), ap1_genes)\n",
    "names(ap1_genes) <- c(\"UniProtID\", \"Name\", \"Symbol\")\n",
    "head(ap1_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3788757b",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "symbol.str <- substr(ap1_genes$Symbol, 1,15)\n",
    "symbol.str\n",
    "ap1_genes$Ensembl <- mapIds(org.Hs.eg.db,\n",
    "                            keys = symbol.str,\n",
    "                            column = \"ENSEMBL\",\n",
    "                            keytype = \"SYMBOL\",\n",
    "                            multiVals= \"first\")\n",
    "ap1_genes"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b9027c34",
   "metadata": {},
   "source": [
    "## PI3K Pathway"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "79bcc10b",
   "metadata": {},
   "source": [
    "Probably the pathway I should've tried first"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "573607a3",
   "metadata": {},
   "outputs": [],
   "source": [
    "pi3k_genes <- read_tsv(file = \"pi3k_gene_list_GO_0014065.txt\")\n",
    "head(pi3k_genes)\n",
    "hippo_genes <- rbind(names(pi3k_genes), pi3k_genes)\n",
    "names(pi3k_genes) <- c(\"UniProtID\", \"Name\", \"Symbol\")\n",
    "head(pi3k_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "557d8d4a",
   "metadata": {},
   "outputs": [],
   "source": [
    "symbol.str <- substr(pi3k_genes$Symbol, 1,15)\n",
    "symbol.str\n",
    "pi3k_genes$Ensembl <- mapIds(org.Hs.eg.db,\n",
    "                            keys = symbol.str,\n",
    "                            column = \"ENSEMBL\",\n",
    "                            keytype = \"SYMBOL\",\n",
    "                            multiVals= \"first\")\n",
    "pi3k_genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "64a2314a",
   "metadata": {},
   "outputs": [],
   "source": [
    "pi3k_ensembl <- pi3k_genes$Ensembl %>%\n",
    "                as.character()\n",
    "pi3k_ensembl\n",
    "pi3k_ensembl <- na.omit(pi3k_ensembl)\n",
    "pi3k_ensembl <- as.character(pi3k_ensembl)\n",
    "pi3k_normCounts <- normalized_total_counts_t[, (colnames(normalized_total_counts_t) %in% \n",
    "                                                c(pi3k_ensembl))]\n",
    "head(pi3k_normCounts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2be1a118",
   "metadata": {},
   "outputs": [],
   "source": [
    "ens.str <- substr(colnames(pi3k_normCounts), 1,15)\n",
    "genes_for_pca <- mapIds(org.Hs.eg.db,\n",
    "                            keys = ens.str,\n",
    "                            column = \"SYMBOL\",\n",
    "                            keytype = \"ENSEMBL\",\n",
    "                            multiVals= \"first\")\n",
    "str(genes_for_pca)\n",
    "genes_for_pca <- as.character(genes_for_pca)\n",
    "genes_for_pca\n",
    "colnames(pi3k_normCounts) <- genes_for_pca\n",
    "colnames(pi3k_normCounts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "84a3cb98",
   "metadata": {},
   "outputs": [],
   "source": [
    "pi3k_normres.pca <- prcomp(pi3k_normCounts, scale = TRUE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e8f16f4e",
   "metadata": {},
   "outputs": [],
   "source": [
    "library(\"factoextra\")\n",
    "fviz_pca_ind(pi3k_normres.pca,\n",
    "             col.ind = \"cos2\", # Color by the quality of representation\n",
    "             gradient.cols = c(\"#00AFBB\", \"#E7B800\", \"#FC4E07\"),\n",
    "             repel = TRUE     # Avoid text overlapping\n",
    "             )\n",
    "### Normalized Counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a4cbe298",
   "metadata": {},
   "outputs": [],
   "source": [
    "fviz_pca_biplot(pi3k_normres.pca, repel = TRUE,\n",
    "                col.var = \"#2E9FDF\", # Variables color\n",
    "                col.ind = \"#696969\"  # Individuals color\n",
    "                )"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6e1e38c3",
   "metadata": {},
   "source": [
    "## Growth Factor Response in general"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cedb8ab9",
   "metadata": {},
   "outputs": [],
   "source": [
    "GF_genes <- read_tsv(file = \"GF_gene_list_GO_0070848.txt\")\n",
    "head(GF_genes)\n",
    "GF_genes <- rbind(names(GF_genes), GF_genes)\n",
    "names(GF_genes) <- c(\"UniProtID\", \"Symbol\", \"Name\")\n",
    "head(GF_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "97d4d863",
   "metadata": {},
   "outputs": [],
   "source": [
    "symbol.str <- substr(GF_genes$Symbol, 1,15)\n",
    "symbol.str\n",
    "GF_genes$Ensembl <- mapIds(org.Hs.eg.db,\n",
    "                            keys = symbol.str,\n",
    "                            column = \"ENSEMBL\",\n",
    "                            keytype = \"SYMBOL\",\n",
    "                            multiVals= \"first\")\n",
    "GF_genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5893c4e0",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "GF_ensembl <- GF_genes$Ensembl %>%\n",
    "                as.character()\n",
    "GF_ensembl\n",
    "GF_ensembl <- na.omit(GF_ensembl)\n",
    "GF_ensembl <- as.character(GF_ensembl)\n",
    "GF_normCounts <- normalized_total_counts_t[, (colnames(normalized_total_counts_t) %in% \n",
    "                                                c(GF_ensembl))]\n",
    "head(GF_normCounts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b754f17b",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "ens.str <- substr(colnames(GF_normCounts), 1,15)\n",
    "genes_for_pca <- mapIds(org.Hs.eg.db,\n",
    "                            keys = ens.str,\n",
    "                            column = \"SYMBOL\",\n",
    "                            keytype = \"ENSEMBL\",\n",
    "                            multiVals= \"first\")\n",
    "str(genes_for_pca)\n",
    "genes_for_pca <- as.character(genes_for_pca)\n",
    "genes_for_pca\n",
    "colnames(GF_normCounts) <- genes_for_pca\n",
    "colnames(GF_normCounts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a881d427",
   "metadata": {},
   "outputs": [],
   "source": [
    "GF_normres.pca <- prcomp(GF_normCounts, scale = TRUE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4fcdf75e",
   "metadata": {},
   "outputs": [],
   "source": [
    "library(\"factoextra\")\n",
    "fviz_pca_ind(GF_normres.pca,\n",
    "             col.ind = \"cos2\", # Color by the quality of representation\n",
    "             gradient.cols = c(\"#00AFBB\", \"#E7B800\", \"#FC4E07\"),\n",
    "             repel = TRUE     # Avoid text overlapping\n",
    "             )\n",
    "### Normalized Counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e89be443",
   "metadata": {},
   "outputs": [],
   "source": [
    "fviz_pca_biplot(GF_normres.pca, repel = TRUE,\n",
    "                col.var = \"#2E9FDF\", # Variables color\n",
    "                col.ind = \"#696969\"  # Individuals color\n",
    "                )"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "136d9c6d",
   "metadata": {},
   "source": [
    "## RTK Pathway"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6786bf20",
   "metadata": {},
   "outputs": [],
   "source": [
    "RTK_genes <- read_tsv(file = \"RTK_path_genes.txt\")\n",
    "head(RTK_genes)\n",
    "RTK_genes <- rbind(names(RTK_genes), RTK_genes)\n",
    "names(RTK_genes) <- c(\"UniProtID\", \"Name\", \"Symbol\")\n",
    "head(RTK_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6c291156",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "symbol.str <- substr(RTK_genes$Symbol, 1,15)\n",
    "symbol.str\n",
    "RTK_genes$Ensembl <- mapIds(org.Hs.eg.db,\n",
    "                            keys = symbol.str,\n",
    "                            column = \"ENSEMBL\",\n",
    "                            keytype = \"SYMBOL\",\n",
    "                            multiVals= \"first\")\n",
    "RTK_genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b5c7975c",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "head(normalized_total_counts_t)\n",
    "norm_names <- colnames(normalized_total_counts_t)\n",
    "norm_names <- tools::file_path_sans_ext(c(norm_names))\n",
    "norm_names\n",
    "colnames(normalized_total_counts_t) <- norm_names\n",
    "head(normalized_total_counts_t)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "87edd267",
   "metadata": {},
   "outputs": [],
   "source": [
    "RTK_ensembl <- RTK_genes$Ensembl %>%\n",
    "                as.character()\n",
    "RTK_ensembl\n",
    "RTK_ensembl <- na.omit(RTK_ensembl)\n",
    "RTK_ensembl <- as.character(RTK_ensembl)\n",
    "RTK_ensembl\n",
    "RTK_normCounts <- normalized_total_counts_t[, (colnames(normalized_total_counts_t) %in% \n",
    "                                                c(RTK_ensembl))]\n",
    "head(RTK_normCounts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "12228eb5",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "ens.str <- substr(colnames(RTK_normCounts), 1,15)\n",
    "genes_for_pca <- mapIds(org.Hs.eg.db,\n",
    "                            keys = ens.str,\n",
    "                            column = \"SYMBOL\",\n",
    "                            keytype = \"ENSEMBL\",\n",
    "                            multiVals= \"first\")\n",
    "str(genes_for_pca)\n",
    "genes_for_pca <- as.character(genes_for_pca)\n",
    "genes_for_pca\n",
    "colnames(RTK_normCounts) <- genes_for_pca\n",
    "colnames(RTK_normCounts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "630631d3",
   "metadata": {},
   "outputs": [],
   "source": [
    "RTK_normres.pca <- prcomp(RTK_normCounts, scale = TRUE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c2811c7b",
   "metadata": {},
   "outputs": [],
   "source": [
    "library(\"factoextra\")\n",
    "fviz_pca_ind(RTK_normres.pca,\n",
    "             col.ind = \"cos2\", # Color by the quality of representation\n",
    "             gradient.cols = c(\"#00AFBB\", \"#E7B800\", \"#FC4E07\"),\n",
    "             repel = TRUE     # Avoid text overlapping\n",
    "             )\n",
    "### Normalized Counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bebd4d6d",
   "metadata": {},
   "outputs": [],
   "source": [
    "fviz_pca_biplot(RTK_normres.pca, repel = TRUE,\n",
    "                col.var = \"#2E9FDF\", # Variables color\n",
    "                col.ind = \"#696969\"  # Individuals color\n",
    "                )"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9661ab77",
   "metadata": {},
   "source": [
    "# New NEW Approach: Remove Lineage-specific genes"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fb9b0ba0",
   "metadata": {},
   "source": [
    "New approach from Wendy. Let's try to identify lineage-specific expression markers and eliminate those from the DESeq2 analysis."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4a65705a",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "total_counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "96ef487b",
   "metadata": {},
   "outputs": [],
   "source": [
    "names(total_counts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e7368232",
   "metadata": {},
   "outputs": [],
   "source": [
    "lineage <- c(rep(\"MCF10A\",6), \n",
    "             rep(\"cancer\",6), \n",
    "             rep(\"MCF10A\",3),\n",
    "             rep(\"HTert\",9),\n",
    "             rep(\"cancer\",3))\n",
    "lineage <- as.factor(lineage)\n",
    "coldata <- data.frame(lineage)\n",
    "rownames(coldata) <- names(total_counts)\n",
    "all(rownames(coldata) == colnames(total_counts))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "552e105b",
   "metadata": {},
   "outputs": [],
   "source": [
    "#DESeq Time\n",
    "dds <- DESeqDataSetFromMatrix(countData = total_counts,\n",
    "                              colData = coldata,\n",
    "                              design = ~ lineage)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3cbf9c11",
   "metadata": {},
   "outputs": [],
   "source": [
    "dds"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "adbc381f",
   "metadata": {},
   "outputs": [],
   "source": [
    "dds <- DESeq(dds)\n",
    "resultsNames(dds)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0ec732f2",
   "metadata": {},
   "outputs": [],
   "source": [
    "dds_vst <- varianceStabilizingTransformation(dds, blind = FALSE)\n",
    "plotPCA(dds_vst, intgroup = \"lineage\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ecff143e",
   "metadata": {},
   "outputs": [],
   "source": [
    "######## filter by significance\n",
    "######## ## l2fc +/- 1.5 and padj <0.05\n",
    "######## ###12895 genes (1.5fc)\n",
    "#10618 (2fc)\n",
    "#8936 (2.5fc)\n",
    "canvHTert_split_res <- results(dds, contrast = c(\"lineage\", \"cancer\", \"HTert\"))\n",
    "canvHTert_split_res <- as.data.frame(canvHTert_split_res)\n",
    "canvHTert_split_res_df_filt <- canvHTert_split_res %>%\n",
    "  filter(log2FoldChange < -2.5 | log2FoldChange > 2.5) %>%\n",
    "  filter(padj < 0.05)\n",
    "canvHTert_split_res_df_filt\n",
    "\n",
    "#11401 genes (1.5fc)\n",
    "#8930 genes (2fc)\n",
    "#7127 genes (2.5 fc)\n",
    "canvMCF10A_split_res <- results(dds, contrast = c(\"lineage\", \"cancer\", \"MCF10A\"))\n",
    "canvMCF10A_split_res <- as.data.frame(canvMCF10A_split_res)\n",
    "canvMCF10A_split_res_df_filt <- canvMCF10A_split_res %>%\n",
    "  filter(log2FoldChange < -2.5 | log2FoldChange > 2.5) %>%\n",
    "  filter(padj < 0.05)\n",
    "canvMCF10A_split_res_df_filt\n",
    "\n",
    "#9642 genes (1.5fc)\n",
    "#7771 genes (2fc)\n",
    "#6394 genes (2.5fc)\n",
    "HTertvMCF10A_split_res <- results(dds, contrast = c(\"lineage\", \"HTert\", \"MCF10A\"))\n",
    "HTertvMCF10A_split_res <- as.data.frame(HTertvMCF10A_split_res)\n",
    "HTertvMCF10A_split_res_df_filt <- HTertvMCF10A_split_res %>%\n",
    "  filter(log2FoldChange < -2.5 | log2FoldChange > 2.5) %>%\n",
    "  filter(padj < 0.05)\n",
    "HTertvMCF10A_split_res_df_filt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dbfa6f6a",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "#At 2.5fc threshold\n",
    "lineage_diff_genes <- c(c(rownames(canvHTert_split_res_df_filt), \n",
    "                          rownames(canvMCF10A_split_res_df_filt)), \n",
    "                          rownames(HTertvMCF10A_split_res_df_filt))\n",
    "lineage_diff_genes\n",
    "str(lineage_diff_genes)\n",
    "lineage_diff_genes <- unique(lineage_diff_genes)\n",
    "str(lineage_diff_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d6e9d5a5",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "`%!in%` = Negate(`%in%`)\n",
    "NonLineage_counts <- total_counts %>% \n",
    "                filter(rownames(total_counts) %!in% lineage_diff_genes)\n",
    "NonLineage_counts"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b4017df3",
   "metadata": {},
   "source": [
    "## Removed Non-lineage Counts"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8f6db2bd",
   "metadata": {},
   "source": [
    "### By Genotype"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c2090a78",
   "metadata": {},
   "outputs": [],
   "source": [
    "genotype <- c(rep(\"E545K\",3), \n",
    "             rep(\"WT\",6), \n",
    "             rep(\"E545K\",3),\n",
    "             rep(\"H1047R\",3),\n",
    "             rep(\"WT\",3),\n",
    "             rep(\"E545K\",3),\n",
    "             rep(\"H1047R\",6))\n",
    "genotype <- as.factor(genotype)\n",
    "coldata <- data.frame(genotype)\n",
    "rownames(coldata) <- colnames(NonLineage_counts)\n",
    "all(rownames(coldata) == colnames(NonLineage_counts))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4c0e2f98",
   "metadata": {},
   "outputs": [],
   "source": [
    "#DESeq Time\n",
    "dds <- DESeqDataSetFromMatrix(countData = NonLineage_counts,\n",
    "                              colData = coldata,\n",
    "                              design = ~ genotype)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "106d035e",
   "metadata": {},
   "outputs": [],
   "source": [
    "dds"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "512dcb9c",
   "metadata": {},
   "outputs": [],
   "source": [
    "dds <- DESeq(dds)\n",
    "resultsNames(dds)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ba1f9156",
   "metadata": {},
   "outputs": [],
   "source": [
    "dds_vst <- varianceStabilizingTransformation(dds, blind = FALSE)\n",
    "plotPCA(dds_vst, intgroup = \"genotype\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "997c9283",
   "metadata": {},
   "source": [
    "### By lineage"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ed352358",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "lineage <- c(rep(\"MCF10A\",6), \n",
    "             rep(\"cancer\",6), \n",
    "             rep(\"MCF10A\",3),\n",
    "             rep(\"HTert\",9),\n",
    "             rep(\"cancer\",3))\n",
    "lineage <- as.factor(lineage)\n",
    "coldata <- data.frame(lineage)\n",
    "rownames(coldata) <- names(NonLineage_counts)\n",
    "all(rownames(coldata) == colnames(NonLineage_counts))\n",
    "#DESeq Time\n",
    "dds <- DESeqDataSetFromMatrix(countData = NonLineage_counts,\n",
    "                              colData = coldata,\n",
    "                              design = ~ lineage)\n",
    "dds <- DESeq(dds)\n",
    "resultsNames(dds)\n",
    "dds_vst <- varianceStabilizingTransformation(dds, blind = FALSE)\n",
    "plotPCA(dds_vst, intgroup = \"lineage\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f634cf7d",
   "metadata": {},
   "source": [
    "## Throw out the Cancer cells"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ab03a800",
   "metadata": {},
   "source": [
    "Because Fuck 'em"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "71e373da",
   "metadata": {},
   "outputs": [],
   "source": [
    "names(total_counts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "60fd359b",
   "metadata": {},
   "outputs": [],
   "source": [
    "nocan_names <- c(\"MCF10A_E545K_rep1\", \"MCF10A_E545K_rep2\", \"MCF10A_E545K_rep3\",\n",
    "                \"MCF10A_parental_rep1\", \"MCF10A_parental_rep2\", \"MCF10A_parental_rep3\",\n",
    "                \"MCF10A_H1047R_rep1\", \"MCF10A_H1047R_rep2\", \"MCF10A_H1047R_rep3\", \n",
    "                \"HTert_WT_rep1\", \"HTert_WT_rep2\", \"HTert_WT_rep3\", \n",
    "                \"HTert_ex9_rep1\", \"HTert_ex9_rep2\", \"HTert_ex9_rep3\", \n",
    "                \"HTert_ex20_rep1\", \"HTert_ex20_rep2\", \"HTert_ex20_rep3\")\n",
    "str(nocan_names)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7387e340",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "nocan_counts <- total_counts %>%\n",
    "                dplyr::select(nocan_names)\n",
    "nocan_counts"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1a792ada",
   "metadata": {},
   "source": [
    "### Remove Lineage Specific Genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d33379bc",
   "metadata": {},
   "outputs": [],
   "source": [
    "lineage <- c(rep(\"MCF10A\",9),\n",
    "             rep(\"HTert\",9))\n",
    "lineage <- as.factor(lineage)\n",
    "coldata <- data.frame(lineage)\n",
    "rownames(coldata) <- names(nocan_counts)\n",
    "all(rownames(coldata) == colnames(nocan_counts))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "264b52fe",
   "metadata": {},
   "outputs": [],
   "source": [
    "#DESeq Time\n",
    "dds <- DESeqDataSetFromMatrix(countData = nocan_counts,\n",
    "                              colData = coldata,\n",
    "                              design = ~ lineage)\n",
    "dds\n",
    "dds <- DESeq(dds)\n",
    "resultsNames(dds)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ae05ce91",
   "metadata": {},
   "outputs": [],
   "source": [
    "dds_vst <- varianceStabilizingTransformation(dds, blind = FALSE)\n",
    "plotPCA(dds_vst, intgroup = \"lineage\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bcaec393",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "#9580 genes at 1.5fc\n",
    "#6284 genes at 2.5fc\n",
    "nocan_HTertvMCF10A_split_res <- results(dds, contrast = c(\"lineage\", \"HTert\", \"MCF10A\"))\n",
    "nocan_HTertvMCF10A_split_res <- as.data.frame(nocan_HTertvMCF10A_split_res)\n",
    "nocan_HTertvMCF10A_split_res_df_filt <- nocan_HTertvMCF10A_split_res %>%\n",
    "  filter(log2FoldChange < -1.5 | log2FoldChange > 1.5) %>%\n",
    "  filter(padj < 0.05)\n",
    "nocan_HTertvMCF10A_split_res_df_filt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7a9e0162",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "`%!in%` = Negate(`%in%`)\n",
    "nocan_NonLineage_counts <- nocan_counts %>% \n",
    "                filter(rownames(nocan_counts) %!in% rownames(nocan_HTertvMCF10A_split_res_df_filt))\n",
    "nocan_NonLineage_counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b6064bd6",
   "metadata": {},
   "outputs": [],
   "source": [
    "names(nocan_NonLineage_counts)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9cdb9c52",
   "metadata": {},
   "source": [
    "### No cancer cells, lineage specific genes removed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "93787424",
   "metadata": {},
   "outputs": [],
   "source": [
    "genotype <- c(rep(\"E545K\",3), \n",
    "             rep(\"WT\",3),\n",
    "             rep(\"H1047R\",3),\n",
    "             rep(\"WT\",3),\n",
    "             rep(\"E545K\",3),\n",
    "             rep(\"H1047R\",3))\n",
    "genotype <- as.factor(genotype)\n",
    "coldata <- data.frame(genotype)\n",
    "rownames(coldata) <- colnames(nocan_NonLineage_counts)\n",
    "all(rownames(coldata) == colnames(nocan_NonLineage_counts))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9b106ace",
   "metadata": {},
   "outputs": [],
   "source": [
    "#DESeq Time\n",
    "dds <- DESeqDataSetFromMatrix(countData = nocan_NonLineage_counts,\n",
    "                              colData = coldata,\n",
    "                              design = ~ genotype)\n",
    "dds\n",
    "dds <- DESeq(dds)\n",
    "resultsNames(dds)\n",
    "dds_vst <- varianceStabilizingTransformation(dds, blind = FALSE)\n",
    "plotPCA(dds_vst, intgroup = \"genotype\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9f375a56",
   "metadata": {},
   "source": [
    "# Shared Pathways"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "62fff8fc",
   "metadata": {},
   "source": [
    "First we gotta do lineage independent comparisons and KEGG enrichment for each mutational comparison"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dbd71eea",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "cat SharedPaths.R"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7a3f8054",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "library(\"tidyverse\")\n",
    "library(\"org.Hs.eg.db\")\n",
    "library(\"ggrepel\")\n",
    "library(\"DESeq2\")\n",
    "setwd(\"/data/park_lab/adam/Total_RNA\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "be0df0fc",
   "metadata": {},
   "outputs": [],
   "source": [
    "load(file = \"sharedPaths.RData\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c48f480d",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "ls()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "06355177",
   "metadata": {},
   "outputs": [],
   "source": [
    "sharedPaths_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "acf7b748",
   "metadata": {},
   "outputs": [],
   "source": [
    "WTvex20_sharedPaths_df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "116d1732",
   "metadata": {},
   "source": [
    "# Linear Model?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "78aa3e38",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "library(\"tidyverse\")\n",
    "library(\"org.Hs.eg.db\")\n",
    "library(\"ggrepel\")\n",
    "library(\"DESeq2\")\n",
    "setwd(\"/data/park_lab/adam/Total_RNA\")\n",
    "load(\"Total_RNA_env.RData\")\n",
    "ls()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "29fb9d87",
   "metadata": {},
   "outputs": [],
   "source": [
    "normalized_total_counts_t"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7fdd0a73",
   "metadata": {},
   "source": [
    "Add Categorical Columns for model to incorporate"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "948838e9",
   "metadata": {},
   "outputs": [],
   "source": [
    "Lineage <- c(rep(\"MCF10A\", 6),\n",
    "             rep(\"cancer\", 6),\n",
    "             rep(\"MCF10A\", 3),\n",
    "             rep(\"HTert\", 9),\n",
    "             rep(\"cancer\", 3))\n",
    "Lineage"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "db36b4e6",
   "metadata": {},
   "outputs": [],
   "source": [
    "Genotype <- c(rep(\"ex9\", 3), rep(\"WT\", 6), rep(\"ex9\", 3), rep(\"ex20\", 3), \n",
    "             rep(\"WT\", 3), rep(\"ex9\", 3), rep(\"ex20\", 6))\n",
    "Genotype"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d2735c98",
   "metadata": {},
   "outputs": [],
   "source": [
    "norm_counts_model <- normalized_total_counts_t\n",
    "norm_counts_model <- cbind(Lineage, Genotype, norm_counts_model)\n",
    "norm_counts_model"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "44ec4ac4",
   "metadata": {},
   "source": [
    "GLM loop from Kevin Blighe, Adding covariate of lineage"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "418e401b",
   "metadata": {},
   "outputs": [],
   "source": [
    "formula.start <- \"Genotype ~ \"\n",
    "covars <- \" + Lineage\"\n",
    "listModels <- list()\n",
    "\n",
    "for (i in 3:ncol(norm_counts_model))\n",
    "{\n",
    "    formula <- paste(formula.start, colnames(norm_counts_model)[i], covars, sep=\"\")\n",
    "    model <- glm(as.formula(formula), data=norm_counts_model, family=binomial(link=\"logit\"))\n",
    "    listModels[[i]] <- model\n",
    "    names(listModels[i]) <- colnames(norm_counts_model)[i]\n",
    "    print(summary(model))\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5c4718a1",
   "metadata": {},
   "outputs": [],
   "source": [
    "listModels"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "678f2f12",
   "metadata": {},
   "outputs": [],
   "source": [
    "save.image(file = \"GLM.RData\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "54e997c8",
   "metadata": {},
   "source": [
    "# Modeling Tom's Way"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "85336e84",
   "metadata": {},
   "source": [
    "Based on Tom's code:"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "541a9390",
   "metadata": {},
   "source": [
    "Saved at:\n",
    "/data/park_lab/adam/rclass.final.Rmd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "4db36204",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "-- \u001b[1mAttaching packages\u001b[22m --------------------------------------- tidyverse 1.3.1 --\n",
      "\n",
      "\u001b[32mv\u001b[39m \u001b[34mggplot2\u001b[39m 3.3.5     \u001b[32mv\u001b[39m \u001b[34mpurrr  \u001b[39m 0.3.4\n",
      "\u001b[32mv\u001b[39m \u001b[34mtibble \u001b[39m 3.1.6     \u001b[32mv\u001b[39m \u001b[34mdplyr  \u001b[39m 1.0.8\n",
      "\u001b[32mv\u001b[39m \u001b[34mtidyr  \u001b[39m 1.2.0     \u001b[32mv\u001b[39m \u001b[34mstringr\u001b[39m 1.4.0\n",
      "\u001b[32mv\u001b[39m \u001b[34mreadr  \u001b[39m 2.1.2     \u001b[32mv\u001b[39m \u001b[34mforcats\u001b[39m 0.5.1\n",
      "\n",
      "-- \u001b[1mConflicts\u001b[22m ------------------------------------------ tidyverse_conflicts() --\n",
      "\u001b[31mx\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mfilter()\u001b[39m  masks \u001b[34mstats\u001b[39m::filter()\n",
      "\u001b[31mx\u001b[39m \u001b[34mpurrr\u001b[39m::\u001b[32mflatten()\u001b[39m masks \u001b[34mjsonlite\u001b[39m::flatten()\n",
      "\u001b[31mx\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mlag()\u001b[39m     masks \u001b[34mstats\u001b[39m::lag()\n",
      "\n",
      "Loading required package: AnnotationDbi\n",
      "\n",
      "Loading required package: stats4\n",
      "\n",
      "Loading required package: BiocGenerics\n",
      "\n",
      "\n",
      "Attaching package: 'BiocGenerics'\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:dplyr':\n",
      "\n",
      "    combine, intersect, setdiff, union\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:stats':\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:base':\n",
      "\n",
      "    Filter, Find, Map, Position, Reduce, anyDuplicated, append,\n",
      "    as.data.frame, basename, cbind, colnames, dirname, do.call,\n",
      "    duplicated, eval, evalq, get, grep, grepl, intersect, is.unsorted,\n",
      "    lapply, mapply, match, mget, order, paste, pmax, pmax.int, pmin,\n",
      "    pmin.int, rank, rbind, rownames, sapply, setdiff, sort, table,\n",
      "    tapply, union, unique, unsplit, which.max, which.min\n",
      "\n",
      "\n",
      "Loading required package: Biobase\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Loading required package: IRanges\n",
      "\n",
      "Loading required package: S4Vectors\n",
      "\n",
      "\n",
      "Attaching package: 'S4Vectors'\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:dplyr':\n",
      "\n",
      "    first, rename\n",
      "\n",
      "\n",
      "The following object is masked from 'package:tidyr':\n",
      "\n",
      "    expand\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:base':\n",
      "\n",
      "    I, expand.grid, unname\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: 'IRanges'\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:dplyr':\n",
      "\n",
      "    collapse, desc, slice\n",
      "\n",
      "\n",
      "The following object is masked from 'package:purrr':\n",
      "\n",
      "    reduce\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: 'AnnotationDbi'\n",
      "\n",
      "\n",
      "The following object is masked from 'package:dplyr':\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Loading required package: GenomicRanges\n",
      "\n",
      "Loading required package: GenomeInfoDb\n",
      "\n",
      "Loading required package: SummarizedExperiment\n",
      "\n",
      "Loading required package: MatrixGenerics\n",
      "\n",
      "Loading required package: matrixStats\n",
      "\n",
      "\n",
      "Attaching package: 'matrixStats'\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:Biobase':\n",
      "\n",
      "    anyMissing, rowMedians\n",
      "\n",
      "\n",
      "The following object is masked from 'package:dplyr':\n",
      "\n",
      "    count\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: 'MatrixGenerics'\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:matrixStats':\n",
      "\n",
      "    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,\n",
      "    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,\n",
      "    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,\n",
      "    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,\n",
      "    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,\n",
      "    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,\n",
      "    colWeightedMeans, colWeightedMedians, colWeightedSds,\n",
      "    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,\n",
      "    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,\n",
      "    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,\n",
      "    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,\n",
      "    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,\n",
      "    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,\n",
      "    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,\n",
      "    rowWeightedSds, rowWeightedVars\n",
      "\n",
      "\n",
      "The following object is masked from 'package:Biobase':\n",
      "\n",
      "    rowMedians\n",
      "\n",
      "\n",
      "Loading required package: knitr\n",
      "\n",
      "\n",
      "\n",
      "Registered S3 method overwritten by 'GGally':\n",
      "  method from   \n",
      "  +.gg   ggplot2\n",
      "\n",
      "\n",
      "\n",
      "Loading required package: mgcv\n",
      "\n",
      "Loading required package: nlme\n",
      "\n",
      "\n",
      "Attaching package: 'nlme'\n",
      "\n",
      "\n",
      "The following object is masked from 'package:IRanges':\n",
      "\n",
      "    collapse\n",
      "\n",
      "\n",
      "The following object is masked from 'package:dplyr':\n",
      "\n",
      "    collapse\n",
      "\n",
      "\n",
      "This is mgcv 1.8-39. For overview type 'help(\"mgcv-package\")'.\n",
      "\n",
      "Loading required package: genefilter\n",
      "\n",
      "\n",
      "Attaching package: 'genefilter'\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:MatrixGenerics':\n",
      "\n",
      "    rowSds, rowVars\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:matrixStats':\n",
      "\n",
      "    rowSds, rowVars\n",
      "\n",
      "\n",
      "The following object is masked from 'package:readr':\n",
      "\n",
      "    spec\n",
      "\n",
      "\n",
      "Loading required package: BiocParallel\n",
      "\n",
      "Loading required package: annotate\n",
      "\n",
      "Loading required package: XML\n",
      "\n",
      "Loading required package: graph\n",
      "\n",
      "\n",
      "Attaching package: 'graph'\n",
      "\n",
      "\n",
      "The following object is masked from 'package:XML':\n",
      "\n",
      "    addNode\n",
      "\n",
      "\n",
      "The following object is masked from 'package:stringr':\n",
      "\n",
      "    boundary\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: 'gplots'\n",
      "\n",
      "\n",
      "The following object is masked from 'package:IRanges':\n",
      "\n",
      "    space\n",
      "\n",
      "\n",
      "The following object is masked from 'package:S4Vectors':\n",
      "\n",
      "    space\n",
      "\n",
      "\n",
      "The following object is masked from 'package:stats':\n",
      "\n",
      "    lowess\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: 'limma'\n",
      "\n",
      "\n",
      "The following object is masked from 'package:DESeq2':\n",
      "\n",
      "    plotMA\n",
      "\n",
      "\n",
      "The following object is masked from 'package:BiocGenerics':\n",
      "\n",
      "    plotMA\n",
      "\n",
      "\n",
      "Loading required package: pathfindR.data\n",
      "\n",
      "##############################################################################\n",
      "                        Welcome to pathfindR!\n",
      "\n",
      "Please cite the article below if you use pathfindR in published reseach:\n",
      "\n",
      "Ulgen E, Ozisik O, Sezerman OU. 2019. pathfindR: An R Package for Comprehensive\n",
      "Identification of Enriched Pathways in Omics Data Through Active Subnetworks.\n",
      "Front. Genet. doi:10.3389/fgene.2019.00858\n",
      "\n",
      "##############################################################################\n",
      "\n",
      "Linking to ImageMagick 7.1.0.26\n",
      "Enabled features: cairo, fontconfig, freetype, fftw, pango, rsvg, webp, x11\n",
      "Disabled features: ghostscript, heic, lcms, raw\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'AREG_exp'</li><li>'AREG_exp_raw'</li><li>'CRISPR_guides'</li><li>'E545K_expressed_E545K_interacting_overlap'</li><li>'E545K_expressed_WT_interacting_overlap'</li><li>'E545K_expressed_interacting_overlap'</li><li>'ETV5_exp'</li><li>'G_list'</li><li>'Gene_targets'</li><li>'Genotype_vec'</li><li>'HTertDists'</li><li>'HTert_WT_counts'</li><li>'HTert_WTvex20_over'</li><li>'HTert_WTvex20_res'</li><li>'HTert_WTvex20_resSig_padj'</li><li>'HTert_WTvex20_under'</li><li>'HTert_WTvex9_over'</li><li>'HTert_WTvex9_res'</li><li>'HTert_WTvex9_resSig_padj'</li><li>'HTert_WTvex9_under'</li><li>'HTert_coldata'</li><li>'HTert_counts'</li><li>'HTert_dds'</li><li>'HTert_dds_f'</li><li>'HTert_diff_genes'</li><li>'HTert_ex20_counts'</li><li>'HTert_ex9_counts'</li><li>'HTert_ex9vex20_over'</li><li>'HTert_ex9vex20_res'</li><li>'HTert_ex9vex20_resSig_padj'</li><li>'HTert_ex9vex20_under'</li><li>'HTert_filt'</li><li>'HTert_genotype'</li><li>'HTert_norm_sigset'</li><li>'HTert_normalized_counts_f'</li><li>'HTert_normalized_counts_f_nonames'</li><li>'HTert_rlog'</li><li>'HTert_sampleDistMatrix'</li><li>'HTert_samples'</li><li>'HTert_vst'</li><li>'HiC_overlap_venns'</li><li>'IRS2_exp'</li><li>'Lineage_vec'</li><li>'MCF10ADists'</li><li>'MCF10A_H1047R_counts'</li><li>'MCF10A_WTvex20_over'</li><li>'MCF10A_WTvex20_res'</li><li>'MCF10A_WTvex20_resSig_padj'</li><li>'MCF10A_WTvex20_under'</li><li>'MCF10A_WTvex9_over'</li><li>'MCF10A_WTvex9_res'</li><li>'MCF10A_WTvex9_resSig_padj'</li><li>'MCF10A_WTvex9_under'</li><li>'MCF10A_coldata'</li><li>'MCF10A_counts'</li><li>'MCF10A_dds'</li><li>'MCF10A_dds_f'</li><li>'MCF10A_diff_genes'</li><li>'MCF10A_ex9vex20_over'</li><li>'MCF10A_ex9vex20_res'</li><li>'MCF10A_ex9vex20_resSig_padj'</li><li>'MCF10A_ex9vex20_under'</li><li>'MCF10A_filt'</li><li>'MCF10A_genotype'</li><li>'MCF10A_norm_sigset'</li><li>'MCF10A_normalized_counts_f'</li><li>'MCF10A_normalized_counts_f_nonames'</li><li>'MCF10A_rlog'</li><li>'MCF10A_sampleDistMatrix'</li><li>'MCF10A_samples'</li><li>'MCF10A_vst'</li><li>'MCF7_corrected_hic_enriched'</li><li>'MCF7_parental_hic_enriched'</li><li>'MCF_7_counts'</li><li>'MCF_dual_E545K_counts'</li><li>'MCF_total_coldata'</li><li>'MCF_total_dds'</li><li>'MCF_total_dds_f'</li><li>'MCF_total_samples'</li><li>'MCF_total_treatment'</li><li>'MCF_total_vst'</li><li>'PIK3R1_exp'</li><li>'PRLR_exp'</li><li>'T47D_counts'</li><li>'TGFB_DN'</li><li>'TGFB_DN_genes'</li><li>'TGFB_DN_guides'</li><li>'TGFB_DN_targets'</li><li>'TGFB_UP'</li><li>'TGFB_UP_genes'</li><li>'TGFB_UP_guides'</li><li>'TGFB_UP_path'</li><li>'TGFB_UP_targets'</li><li>'TGFB_guides'</li><li>'THRB_exp'</li><li>'WT_expressed_E545K_interacting_overlap'</li><li>'WT_expressed_WT_interacting_overlap'</li><li>'WTvex20_GOBP'</li><li>'WTvex20_GOMF'</li><li>'WTvex20_KEGG'</li><li>'WTvex20_combined'</li><li>'WTvex20_combined_GOBP'</li><li>'WTvex20_combined_GOBP_nocan'</li><li>'WTvex20_combined_nocan'</li><li>'WTvex20_genes_log2fc_1.5'</li><li>'WTvex20_genes_log2fc_1.5_entrez'</li><li>'WTvex20_genes_log2fc_1.5_entrez_df'</li><li>'WTvex20_genes_padj'</li><li>'WTvex20_genes_padj_entrez'</li><li>'WTvex20_over_clean'</li><li>'WTvex20_over_clean_nocan'</li><li>'WTvex20_over_common'</li><li>'WTvex20_over_common_GOBP'</li><li>'WTvex20_over_common_GOBP_nocan'</li><li>'WTvex20_over_common_GOMF'</li><li>'WTvex20_over_nocan'</li><li>'WTvex20_res'</li><li>'WTvex20_resSig_padj'</li><li>'WTvex20_resSig_padj_df'</li><li>'WTvex20_resSig_padj_ordered'</li><li>'WTvex20_resSig_padj_ordered_df'</li><li>'WTvex20_res_df'</li><li>'WTvex20_under_clean'</li><li>'WTvex20_under_clean_nocan'</li><li>'WTvex20_under_common'</li><li>'WTvex20_under_common_GOBP'</li><li>'WTvex20_under_common_GOBP_nocan'</li><li>'WTvex20_under_common_GOMF'</li><li>'WTvex20_under_nocan'</li><li>'WTvex9_Comparison_gene_list_df'</li><li>'WTvex9_GOBP'</li><li>'WTvex9_GOMF'</li><li>'WTvex9_KEGG'</li><li>'WTvex9_combined'</li><li>'WTvex9_combined_GOBP'</li><li>'WTvex9_combined_GOBP_nocan'</li><li>'WTvex9_combined_nocan'</li><li>'WTvex9_genes_log2fc_1.5'</li><li>'WTvex9_genes_log2fc_1.5_entrez'</li><li>'WTvex9_genes_padj'</li><li>'WTvex9_genes_padj_entrez'</li><li>'WTvex9_over_clean'</li><li>'WTvex9_over_clean_nocan'</li><li>'WTvex9_over_common'</li><li>'WTvex9_over_common_GOBP'</li><li>'WTvex9_over_common_GOBP_nocan'</li><li>'WTvex9_over_common_GOMF'</li><li>'WTvex9_over_common_KEGG'</li><li>'WTvex9_over_nocan'</li><li>'WTvex9_res'</li><li>'WTvex9_resSig_padj'</li><li>'WTvex9_resSig_padj_df'</li><li>'WTvex9_resSig_padj_ordered'</li><li>'WTvex9_resSig_padj_ordered_df'</li><li>'WTvex9_res_df'</li><li>'WTvex9_under_clean'</li><li>'WTvex9_under_clean_nocan'</li><li>'WTvex9_under_common'</li><li>'WTvex9_under_common_GOBP'</li><li>'WTvex9_under_common_GOBP_nocan'</li><li>'WTvex9_under_common_GOMF'</li><li>'WTvex9_under_nocan'</li><li>'add.flag'</li><li>'all_diff_genes'</li><li>'bp'</li><li>'cancerDists'</li><li>'cancer_WTvex20_over'</li><li>'cancer_WTvex20_res'</li><li>'cancer_WTvex20_resSig_padj'</li><li>'cancer_WTvex20_under'</li><li>'cancer_WTvex9_over'</li><li>'cancer_WTvex9_over_clean'</li><li>'cancer_WTvex9_over_symbols'</li><li>'cancer_WTvex9_res'</li><li>'cancer_WTvex9_resSig_padj'</li><li>'cancer_WTvex9_under'</li><li>'cancer_WTvex9_under_clean'</li><li>'cancer_WTvex9_under_symbols'</li><li>'cancer_coldata'</li><li>'cancer_counts'</li><li>'cancer_dds'</li><li>'cancer_dds_f'</li><li>'cancer_diff_genes'</li><li>'cancer_ex9vex20_over'</li><li>'cancer_ex9vex20_res'</li><li>'cancer_ex9vex20_resSig_padj'</li><li>'cancer_ex9vex20_under'</li><li>'cancer_filt'</li><li>'cancer_genotype'</li><li>'cancer_norm_sigset'</li><li>'cancer_normalized_counts_f'</li><li>'cancer_normalized_counts_f_nonames'</li><li>'cancer_rlog'</li><li>'cancer_sampleDistMatrix'</li><li>'cancer_samples'</li><li>'cancer_vst'</li><li>'cleaningY'</li><li>'cleany'</li><li>'cleany2'</li><li>'cleany3'</li><li>'clinsub'</li><li>'cluster3'</li><li>'cluster3_GOBP'</li><li>'cluster3_genes'</li><li>'cluster3_genes_clean'</li><li>'cluster9'</li><li>'cluster9_GOBP'</li><li>'cluster9_genes'</li><li>'cluster9_genes_clean'</li><li>'cluster_df'</li><li>'clusters'</li><li>'colMeta_df'</li><li>'colors'</li><li>'counts_deseq_sva'</li><li>'cov'</li><li>'dat'</li><li>'datares_HC'</li><li>'dds'</li><li>'dds_LRT'</li><li>'ddssva'</li><li>'diff_WTvex20'</li><li>'diff_WTvex20_sig'</li><li>'diff_WTvex9'</li><li>'diff_WTvex9_sig'</li><li>'diff_ex9vex20'</li><li>'diff_ex9vex20_sig'</li><li>'dups'</li><li>'ens.str_WTvex20'</li><li>'ens.str_WTvex9'</li><li>'ens.str_ex9vex20'</li><li>'ex9vex20_GOBP'</li><li>'ex9vex20_GOMF'</li><li>'ex9vex20_KEGG'</li><li>'ex9vex20_combined'</li><li>'ex9vex20_combined_GOBP'</li><li>'ex9vex20_combined_GOBP_nocan'</li><li>'ex9vex20_combined_nocan'</li><li>'ex9vex20_genes_log2fc_1.5'</li><li>'ex9vex20_genes_log2fc_1.5_entrez'</li><li>'ex9vex20_genes_log2fc_1.5_entrez_df'</li><li>'ex9vex20_genes_padj'</li><li>'ex9vex20_genes_padj_entrez'</li><li>'ex9vex20_over_clean'</li><li>'ex9vex20_over_clean_nocan'</li><li>'ex9vex20_over_common'</li><li>'ex9vex20_over_common_GOBP'</li><li>'ex9vex20_over_common_GOBP_nocan'</li><li>'ex9vex20_over_common_GOMF'</li><li>'ex9vex20_over_nocan'</li><li>'ex9vex20_res'</li><li>'ex9vex20_resSig_padj'</li><li>'ex9vex20_resSig_padj_df'</li><li>'ex9vex20_resSig_padj_ordered'</li><li>'ex9vex20_resSig_padj_ordered_df'</li><li>'ex9vex20_res_df'</li><li>'ex9vex20_under_clean'</li><li>'ex9vex20_under_clean_nocan'</li><li>'ex9vex20_under_common'</li><li>'ex9vex20_under_common_GOBP'</li><li>'ex9vex20_under_common_GOBP_nocan'</li><li>'ex9vex20_under_common_GOMF'</li><li>'ex9vex20_under_nocan'</li><li>'exampleSet'</li><li>'expression'</li><li>'filt'</li><li>'fullGeneList'</li><li>'geneSetClustering'</li><li>'genes'</li><li>'genes2'</li><li>'geneset'</li><li>'geno_comp'</li><li>'geno_comp_filt'</li><li>'good_names'</li><li>'gr'</li><li>'heatmapres2.1'</li><li>'iSample'</li><li>'idx'</li><li>'kmax'</li><li>'lfcShrink_df'</li><li>'lfcShrink_df_absvalue'</li><li>'lfcShrink_df_absvalue_filt'</li><li>'list'</li><li>'lp'</li><li>'mart'</li><li>'mat2'</li><li>'mat2.1'</li><li>'mn'</li><li>'mod'</li><li>'mod0'</li><li>'myBreaks'</li><li>'myC6'</li><li>'myColor'</li><li>'my_palette'</li><li>'myvars'</li><li>'nSamples'</li><li>'nas'</li><li>'norm_subset_mat'</li><li>'norm_total_counts_sig_subset'</li><li>'normalized_MCF_total_counts'</li><li>'normalized_MCF_total_counts_f'</li><li>'normalized_MCF_total_counts_f_nonames'</li><li>'normalized_total_counts'</li><li>'normalized_total_counts_f'</li><li>'normalized_total_counts_f_nonames'</li><li>'normalized_total_counts_t'</li><li>'number_col'</li><li>'p'</li><li>'p_improved'</li><li>'paletteLength'</li><li>'pca'</li><li>'prePathfindR_df'</li><li>'progressBar'</li><li>'range'</li><li>'raw_total_counts_sig_subset'</li><li>'reorder_cells'</li><li>'res'</li><li>'resAshWTvex20'</li><li>'resAshWTvex9'</li><li>'resAshex9vex20'</li><li>'resCombo'</li><li>'resLRT'</li><li>'rld.sigtest2'</li><li>'rlog_dds'</li><li>'sampleDistMatrix'</li><li>'sampleDists'</li><li>'sample_info'</li><li>'secondp'</li><li>'selectGeneList'</li><li>'sets_of_int'</li><li>'sub'</li><li>'svseq'</li><li>'test'</li><li>'test_df'</li><li>'total_coldata'</li><li>'total_counts'</li><li>'total_counts_nonames'</li><li>'total_counts_t'</li><li>'total_dds'</li><li>'total_dds_f'</li><li>'total_rlog'</li><li>'total_samples'</li><li>'total_treatment'</li><li>'total_vst'</li><li>'venn_data'</li><li>'vsd'</li><li>'wss'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'AREG\\_exp'\n",
       "\\item 'AREG\\_exp\\_raw'\n",
       "\\item 'CRISPR\\_guides'\n",
       "\\item 'E545K\\_expressed\\_E545K\\_interacting\\_overlap'\n",
       "\\item 'E545K\\_expressed\\_WT\\_interacting\\_overlap'\n",
       "\\item 'E545K\\_expressed\\_interacting\\_overlap'\n",
       "\\item 'ETV5\\_exp'\n",
       "\\item 'G\\_list'\n",
       "\\item 'Gene\\_targets'\n",
       "\\item 'Genotype\\_vec'\n",
       "\\item 'HTertDists'\n",
       "\\item 'HTert\\_WT\\_counts'\n",
       "\\item 'HTert\\_WTvex20\\_over'\n",
       "\\item 'HTert\\_WTvex20\\_res'\n",
       "\\item 'HTert\\_WTvex20\\_resSig\\_padj'\n",
       "\\item 'HTert\\_WTvex20\\_under'\n",
       "\\item 'HTert\\_WTvex9\\_over'\n",
       "\\item 'HTert\\_WTvex9\\_res'\n",
       "\\item 'HTert\\_WTvex9\\_resSig\\_padj'\n",
       "\\item 'HTert\\_WTvex9\\_under'\n",
       "\\item 'HTert\\_coldata'\n",
       "\\item 'HTert\\_counts'\n",
       "\\item 'HTert\\_dds'\n",
       "\\item 'HTert\\_dds\\_f'\n",
       "\\item 'HTert\\_diff\\_genes'\n",
       "\\item 'HTert\\_ex20\\_counts'\n",
       "\\item 'HTert\\_ex9\\_counts'\n",
       "\\item 'HTert\\_ex9vex20\\_over'\n",
       "\\item 'HTert\\_ex9vex20\\_res'\n",
       "\\item 'HTert\\_ex9vex20\\_resSig\\_padj'\n",
       "\\item 'HTert\\_ex9vex20\\_under'\n",
       "\\item 'HTert\\_filt'\n",
       "\\item 'HTert\\_genotype'\n",
       "\\item 'HTert\\_norm\\_sigset'\n",
       "\\item 'HTert\\_normalized\\_counts\\_f'\n",
       "\\item 'HTert\\_normalized\\_counts\\_f\\_nonames'\n",
       "\\item 'HTert\\_rlog'\n",
       "\\item 'HTert\\_sampleDistMatrix'\n",
       "\\item 'HTert\\_samples'\n",
       "\\item 'HTert\\_vst'\n",
       "\\item 'HiC\\_overlap\\_venns'\n",
       "\\item 'IRS2\\_exp'\n",
       "\\item 'Lineage\\_vec'\n",
       "\\item 'MCF10ADists'\n",
       "\\item 'MCF10A\\_H1047R\\_counts'\n",
       "\\item 'MCF10A\\_WTvex20\\_over'\n",
       "\\item 'MCF10A\\_WTvex20\\_res'\n",
       "\\item 'MCF10A\\_WTvex20\\_resSig\\_padj'\n",
       "\\item 'MCF10A\\_WTvex20\\_under'\n",
       "\\item 'MCF10A\\_WTvex9\\_over'\n",
       "\\item 'MCF10A\\_WTvex9\\_res'\n",
       "\\item 'MCF10A\\_WTvex9\\_resSig\\_padj'\n",
       "\\item 'MCF10A\\_WTvex9\\_under'\n",
       "\\item 'MCF10A\\_coldata'\n",
       "\\item 'MCF10A\\_counts'\n",
       "\\item 'MCF10A\\_dds'\n",
       "\\item 'MCF10A\\_dds\\_f'\n",
       "\\item 'MCF10A\\_diff\\_genes'\n",
       "\\item 'MCF10A\\_ex9vex20\\_over'\n",
       "\\item 'MCF10A\\_ex9vex20\\_res'\n",
       "\\item 'MCF10A\\_ex9vex20\\_resSig\\_padj'\n",
       "\\item 'MCF10A\\_ex9vex20\\_under'\n",
       "\\item 'MCF10A\\_filt'\n",
       "\\item 'MCF10A\\_genotype'\n",
       "\\item 'MCF10A\\_norm\\_sigset'\n",
       "\\item 'MCF10A\\_normalized\\_counts\\_f'\n",
       "\\item 'MCF10A\\_normalized\\_counts\\_f\\_nonames'\n",
       "\\item 'MCF10A\\_rlog'\n",
       "\\item 'MCF10A\\_sampleDistMatrix'\n",
       "\\item 'MCF10A\\_samples'\n",
       "\\item 'MCF10A\\_vst'\n",
       "\\item 'MCF7\\_corrected\\_hic\\_enriched'\n",
       "\\item 'MCF7\\_parental\\_hic\\_enriched'\n",
       "\\item 'MCF\\_7\\_counts'\n",
       "\\item 'MCF\\_dual\\_E545K\\_counts'\n",
       "\\item 'MCF\\_total\\_coldata'\n",
       "\\item 'MCF\\_total\\_dds'\n",
       "\\item 'MCF\\_total\\_dds\\_f'\n",
       "\\item 'MCF\\_total\\_samples'\n",
       "\\item 'MCF\\_total\\_treatment'\n",
       "\\item 'MCF\\_total\\_vst'\n",
       "\\item 'PIK3R1\\_exp'\n",
       "\\item 'PRLR\\_exp'\n",
       "\\item 'T47D\\_counts'\n",
       "\\item 'TGFB\\_DN'\n",
       "\\item 'TGFB\\_DN\\_genes'\n",
       "\\item 'TGFB\\_DN\\_guides'\n",
       "\\item 'TGFB\\_DN\\_targets'\n",
       "\\item 'TGFB\\_UP'\n",
       "\\item 'TGFB\\_UP\\_genes'\n",
       "\\item 'TGFB\\_UP\\_guides'\n",
       "\\item 'TGFB\\_UP\\_path'\n",
       "\\item 'TGFB\\_UP\\_targets'\n",
       "\\item 'TGFB\\_guides'\n",
       "\\item 'THRB\\_exp'\n",
       "\\item 'WT\\_expressed\\_E545K\\_interacting\\_overlap'\n",
       "\\item 'WT\\_expressed\\_WT\\_interacting\\_overlap'\n",
       "\\item 'WTvex20\\_GOBP'\n",
       "\\item 'WTvex20\\_GOMF'\n",
       "\\item 'WTvex20\\_KEGG'\n",
       "\\item 'WTvex20\\_combined'\n",
       "\\item 'WTvex20\\_combined\\_GOBP'\n",
       "\\item 'WTvex20\\_combined\\_GOBP\\_nocan'\n",
       "\\item 'WTvex20\\_combined\\_nocan'\n",
       "\\item 'WTvex20\\_genes\\_log2fc\\_1.5'\n",
       "\\item 'WTvex20\\_genes\\_log2fc\\_1.5\\_entrez'\n",
       "\\item 'WTvex20\\_genes\\_log2fc\\_1.5\\_entrez\\_df'\n",
       "\\item 'WTvex20\\_genes\\_padj'\n",
       "\\item 'WTvex20\\_genes\\_padj\\_entrez'\n",
       "\\item 'WTvex20\\_over\\_clean'\n",
       "\\item 'WTvex20\\_over\\_clean\\_nocan'\n",
       "\\item 'WTvex20\\_over\\_common'\n",
       "\\item 'WTvex20\\_over\\_common\\_GOBP'\n",
       "\\item 'WTvex20\\_over\\_common\\_GOBP\\_nocan'\n",
       "\\item 'WTvex20\\_over\\_common\\_GOMF'\n",
       "\\item 'WTvex20\\_over\\_nocan'\n",
       "\\item 'WTvex20\\_res'\n",
       "\\item 'WTvex20\\_resSig\\_padj'\n",
       "\\item 'WTvex20\\_resSig\\_padj\\_df'\n",
       "\\item 'WTvex20\\_resSig\\_padj\\_ordered'\n",
       "\\item 'WTvex20\\_resSig\\_padj\\_ordered\\_df'\n",
       "\\item 'WTvex20\\_res\\_df'\n",
       "\\item 'WTvex20\\_under\\_clean'\n",
       "\\item 'WTvex20\\_under\\_clean\\_nocan'\n",
       "\\item 'WTvex20\\_under\\_common'\n",
       "\\item 'WTvex20\\_under\\_common\\_GOBP'\n",
       "\\item 'WTvex20\\_under\\_common\\_GOBP\\_nocan'\n",
       "\\item 'WTvex20\\_under\\_common\\_GOMF'\n",
       "\\item 'WTvex20\\_under\\_nocan'\n",
       "\\item 'WTvex9\\_Comparison\\_gene\\_list\\_df'\n",
       "\\item 'WTvex9\\_GOBP'\n",
       "\\item 'WTvex9\\_GOMF'\n",
       "\\item 'WTvex9\\_KEGG'\n",
       "\\item 'WTvex9\\_combined'\n",
       "\\item 'WTvex9\\_combined\\_GOBP'\n",
       "\\item 'WTvex9\\_combined\\_GOBP\\_nocan'\n",
       "\\item 'WTvex9\\_combined\\_nocan'\n",
       "\\item 'WTvex9\\_genes\\_log2fc\\_1.5'\n",
       "\\item 'WTvex9\\_genes\\_log2fc\\_1.5\\_entrez'\n",
       "\\item 'WTvex9\\_genes\\_padj'\n",
       "\\item 'WTvex9\\_genes\\_padj\\_entrez'\n",
       "\\item 'WTvex9\\_over\\_clean'\n",
       "\\item 'WTvex9\\_over\\_clean\\_nocan'\n",
       "\\item 'WTvex9\\_over\\_common'\n",
       "\\item 'WTvex9\\_over\\_common\\_GOBP'\n",
       "\\item 'WTvex9\\_over\\_common\\_GOBP\\_nocan'\n",
       "\\item 'WTvex9\\_over\\_common\\_GOMF'\n",
       "\\item 'WTvex9\\_over\\_common\\_KEGG'\n",
       "\\item 'WTvex9\\_over\\_nocan'\n",
       "\\item 'WTvex9\\_res'\n",
       "\\item 'WTvex9\\_resSig\\_padj'\n",
       "\\item 'WTvex9\\_resSig\\_padj\\_df'\n",
       "\\item 'WTvex9\\_resSig\\_padj\\_ordered'\n",
       "\\item 'WTvex9\\_resSig\\_padj\\_ordered\\_df'\n",
       "\\item 'WTvex9\\_res\\_df'\n",
       "\\item 'WTvex9\\_under\\_clean'\n",
       "\\item 'WTvex9\\_under\\_clean\\_nocan'\n",
       "\\item 'WTvex9\\_under\\_common'\n",
       "\\item 'WTvex9\\_under\\_common\\_GOBP'\n",
       "\\item 'WTvex9\\_under\\_common\\_GOBP\\_nocan'\n",
       "\\item 'WTvex9\\_under\\_common\\_GOMF'\n",
       "\\item 'WTvex9\\_under\\_nocan'\n",
       "\\item 'add.flag'\n",
       "\\item 'all\\_diff\\_genes'\n",
       "\\item 'bp'\n",
       "\\item 'cancerDists'\n",
       "\\item 'cancer\\_WTvex20\\_over'\n",
       "\\item 'cancer\\_WTvex20\\_res'\n",
       "\\item 'cancer\\_WTvex20\\_resSig\\_padj'\n",
       "\\item 'cancer\\_WTvex20\\_under'\n",
       "\\item 'cancer\\_WTvex9\\_over'\n",
       "\\item 'cancer\\_WTvex9\\_over\\_clean'\n",
       "\\item 'cancer\\_WTvex9\\_over\\_symbols'\n",
       "\\item 'cancer\\_WTvex9\\_res'\n",
       "\\item 'cancer\\_WTvex9\\_resSig\\_padj'\n",
       "\\item 'cancer\\_WTvex9\\_under'\n",
       "\\item 'cancer\\_WTvex9\\_under\\_clean'\n",
       "\\item 'cancer\\_WTvex9\\_under\\_symbols'\n",
       "\\item 'cancer\\_coldata'\n",
       "\\item 'cancer\\_counts'\n",
       "\\item 'cancer\\_dds'\n",
       "\\item 'cancer\\_dds\\_f'\n",
       "\\item 'cancer\\_diff\\_genes'\n",
       "\\item 'cancer\\_ex9vex20\\_over'\n",
       "\\item 'cancer\\_ex9vex20\\_res'\n",
       "\\item 'cancer\\_ex9vex20\\_resSig\\_padj'\n",
       "\\item 'cancer\\_ex9vex20\\_under'\n",
       "\\item 'cancer\\_filt'\n",
       "\\item 'cancer\\_genotype'\n",
       "\\item 'cancer\\_norm\\_sigset'\n",
       "\\item 'cancer\\_normalized\\_counts\\_f'\n",
       "\\item 'cancer\\_normalized\\_counts\\_f\\_nonames'\n",
       "\\item 'cancer\\_rlog'\n",
       "\\item 'cancer\\_sampleDistMatrix'\n",
       "\\item 'cancer\\_samples'\n",
       "\\item 'cancer\\_vst'\n",
       "\\item 'cleaningY'\n",
       "\\item 'cleany'\n",
       "\\item 'cleany2'\n",
       "\\item 'cleany3'\n",
       "\\item 'clinsub'\n",
       "\\item 'cluster3'\n",
       "\\item 'cluster3\\_GOBP'\n",
       "\\item 'cluster3\\_genes'\n",
       "\\item 'cluster3\\_genes\\_clean'\n",
       "\\item 'cluster9'\n",
       "\\item 'cluster9\\_GOBP'\n",
       "\\item 'cluster9\\_genes'\n",
       "\\item 'cluster9\\_genes\\_clean'\n",
       "\\item 'cluster\\_df'\n",
       "\\item 'clusters'\n",
       "\\item 'colMeta\\_df'\n",
       "\\item 'colors'\n",
       "\\item 'counts\\_deseq\\_sva'\n",
       "\\item 'cov'\n",
       "\\item 'dat'\n",
       "\\item 'datares\\_HC'\n",
       "\\item 'dds'\n",
       "\\item 'dds\\_LRT'\n",
       "\\item 'ddssva'\n",
       "\\item 'diff\\_WTvex20'\n",
       "\\item 'diff\\_WTvex20\\_sig'\n",
       "\\item 'diff\\_WTvex9'\n",
       "\\item 'diff\\_WTvex9\\_sig'\n",
       "\\item 'diff\\_ex9vex20'\n",
       "\\item 'diff\\_ex9vex20\\_sig'\n",
       "\\item 'dups'\n",
       "\\item 'ens.str\\_WTvex20'\n",
       "\\item 'ens.str\\_WTvex9'\n",
       "\\item 'ens.str\\_ex9vex20'\n",
       "\\item 'ex9vex20\\_GOBP'\n",
       "\\item 'ex9vex20\\_GOMF'\n",
       "\\item 'ex9vex20\\_KEGG'\n",
       "\\item 'ex9vex20\\_combined'\n",
       "\\item 'ex9vex20\\_combined\\_GOBP'\n",
       "\\item 'ex9vex20\\_combined\\_GOBP\\_nocan'\n",
       "\\item 'ex9vex20\\_combined\\_nocan'\n",
       "\\item 'ex9vex20\\_genes\\_log2fc\\_1.5'\n",
       "\\item 'ex9vex20\\_genes\\_log2fc\\_1.5\\_entrez'\n",
       "\\item 'ex9vex20\\_genes\\_log2fc\\_1.5\\_entrez\\_df'\n",
       "\\item 'ex9vex20\\_genes\\_padj'\n",
       "\\item 'ex9vex20\\_genes\\_padj\\_entrez'\n",
       "\\item 'ex9vex20\\_over\\_clean'\n",
       "\\item 'ex9vex20\\_over\\_clean\\_nocan'\n",
       "\\item 'ex9vex20\\_over\\_common'\n",
       "\\item 'ex9vex20\\_over\\_common\\_GOBP'\n",
       "\\item 'ex9vex20\\_over\\_common\\_GOBP\\_nocan'\n",
       "\\item 'ex9vex20\\_over\\_common\\_GOMF'\n",
       "\\item 'ex9vex20\\_over\\_nocan'\n",
       "\\item 'ex9vex20\\_res'\n",
       "\\item 'ex9vex20\\_resSig\\_padj'\n",
       "\\item 'ex9vex20\\_resSig\\_padj\\_df'\n",
       "\\item 'ex9vex20\\_resSig\\_padj\\_ordered'\n",
       "\\item 'ex9vex20\\_resSig\\_padj\\_ordered\\_df'\n",
       "\\item 'ex9vex20\\_res\\_df'\n",
       "\\item 'ex9vex20\\_under\\_clean'\n",
       "\\item 'ex9vex20\\_under\\_clean\\_nocan'\n",
       "\\item 'ex9vex20\\_under\\_common'\n",
       "\\item 'ex9vex20\\_under\\_common\\_GOBP'\n",
       "\\item 'ex9vex20\\_under\\_common\\_GOBP\\_nocan'\n",
       "\\item 'ex9vex20\\_under\\_common\\_GOMF'\n",
       "\\item 'ex9vex20\\_under\\_nocan'\n",
       "\\item 'exampleSet'\n",
       "\\item 'expression'\n",
       "\\item 'filt'\n",
       "\\item 'fullGeneList'\n",
       "\\item 'geneSetClustering'\n",
       "\\item 'genes'\n",
       "\\item 'genes2'\n",
       "\\item 'geneset'\n",
       "\\item 'geno\\_comp'\n",
       "\\item 'geno\\_comp\\_filt'\n",
       "\\item 'good\\_names'\n",
       "\\item 'gr'\n",
       "\\item 'heatmapres2.1'\n",
       "\\item 'iSample'\n",
       "\\item 'idx'\n",
       "\\item 'kmax'\n",
       "\\item 'lfcShrink\\_df'\n",
       "\\item 'lfcShrink\\_df\\_absvalue'\n",
       "\\item 'lfcShrink\\_df\\_absvalue\\_filt'\n",
       "\\item 'list'\n",
       "\\item 'lp'\n",
       "\\item 'mart'\n",
       "\\item 'mat2'\n",
       "\\item 'mat2.1'\n",
       "\\item 'mn'\n",
       "\\item 'mod'\n",
       "\\item 'mod0'\n",
       "\\item 'myBreaks'\n",
       "\\item 'myC6'\n",
       "\\item 'myColor'\n",
       "\\item 'my\\_palette'\n",
       "\\item 'myvars'\n",
       "\\item 'nSamples'\n",
       "\\item 'nas'\n",
       "\\item 'norm\\_subset\\_mat'\n",
       "\\item 'norm\\_total\\_counts\\_sig\\_subset'\n",
       "\\item 'normalized\\_MCF\\_total\\_counts'\n",
       "\\item 'normalized\\_MCF\\_total\\_counts\\_f'\n",
       "\\item 'normalized\\_MCF\\_total\\_counts\\_f\\_nonames'\n",
       "\\item 'normalized\\_total\\_counts'\n",
       "\\item 'normalized\\_total\\_counts\\_f'\n",
       "\\item 'normalized\\_total\\_counts\\_f\\_nonames'\n",
       "\\item 'normalized\\_total\\_counts\\_t'\n",
       "\\item 'number\\_col'\n",
       "\\item 'p'\n",
       "\\item 'p\\_improved'\n",
       "\\item 'paletteLength'\n",
       "\\item 'pca'\n",
       "\\item 'prePathfindR\\_df'\n",
       "\\item 'progressBar'\n",
       "\\item 'range'\n",
       "\\item 'raw\\_total\\_counts\\_sig\\_subset'\n",
       "\\item 'reorder\\_cells'\n",
       "\\item 'res'\n",
       "\\item 'resAshWTvex20'\n",
       "\\item 'resAshWTvex9'\n",
       "\\item 'resAshex9vex20'\n",
       "\\item 'resCombo'\n",
       "\\item 'resLRT'\n",
       "\\item 'rld.sigtest2'\n",
       "\\item 'rlog\\_dds'\n",
       "\\item 'sampleDistMatrix'\n",
       "\\item 'sampleDists'\n",
       "\\item 'sample\\_info'\n",
       "\\item 'secondp'\n",
       "\\item 'selectGeneList'\n",
       "\\item 'sets\\_of\\_int'\n",
       "\\item 'sub'\n",
       "\\item 'svseq'\n",
       "\\item 'test'\n",
       "\\item 'test\\_df'\n",
       "\\item 'total\\_coldata'\n",
       "\\item 'total\\_counts'\n",
       "\\item 'total\\_counts\\_nonames'\n",
       "\\item 'total\\_counts\\_t'\n",
       "\\item 'total\\_dds'\n",
       "\\item 'total\\_dds\\_f'\n",
       "\\item 'total\\_rlog'\n",
       "\\item 'total\\_samples'\n",
       "\\item 'total\\_treatment'\n",
       "\\item 'total\\_vst'\n",
       "\\item 'venn\\_data'\n",
       "\\item 'vsd'\n",
       "\\item 'wss'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'AREG_exp'\n",
       "2. 'AREG_exp_raw'\n",
       "3. 'CRISPR_guides'\n",
       "4. 'E545K_expressed_E545K_interacting_overlap'\n",
       "5. 'E545K_expressed_WT_interacting_overlap'\n",
       "6. 'E545K_expressed_interacting_overlap'\n",
       "7. 'ETV5_exp'\n",
       "8. 'G_list'\n",
       "9. 'Gene_targets'\n",
       "10. 'Genotype_vec'\n",
       "11. 'HTertDists'\n",
       "12. 'HTert_WT_counts'\n",
       "13. 'HTert_WTvex20_over'\n",
       "14. 'HTert_WTvex20_res'\n",
       "15. 'HTert_WTvex20_resSig_padj'\n",
       "16. 'HTert_WTvex20_under'\n",
       "17. 'HTert_WTvex9_over'\n",
       "18. 'HTert_WTvex9_res'\n",
       "19. 'HTert_WTvex9_resSig_padj'\n",
       "20. 'HTert_WTvex9_under'\n",
       "21. 'HTert_coldata'\n",
       "22. 'HTert_counts'\n",
       "23. 'HTert_dds'\n",
       "24. 'HTert_dds_f'\n",
       "25. 'HTert_diff_genes'\n",
       "26. 'HTert_ex20_counts'\n",
       "27. 'HTert_ex9_counts'\n",
       "28. 'HTert_ex9vex20_over'\n",
       "29. 'HTert_ex9vex20_res'\n",
       "30. 'HTert_ex9vex20_resSig_padj'\n",
       "31. 'HTert_ex9vex20_under'\n",
       "32. 'HTert_filt'\n",
       "33. 'HTert_genotype'\n",
       "34. 'HTert_norm_sigset'\n",
       "35. 'HTert_normalized_counts_f'\n",
       "36. 'HTert_normalized_counts_f_nonames'\n",
       "37. 'HTert_rlog'\n",
       "38. 'HTert_sampleDistMatrix'\n",
       "39. 'HTert_samples'\n",
       "40. 'HTert_vst'\n",
       "41. 'HiC_overlap_venns'\n",
       "42. 'IRS2_exp'\n",
       "43. 'Lineage_vec'\n",
       "44. 'MCF10ADists'\n",
       "45. 'MCF10A_H1047R_counts'\n",
       "46. 'MCF10A_WTvex20_over'\n",
       "47. 'MCF10A_WTvex20_res'\n",
       "48. 'MCF10A_WTvex20_resSig_padj'\n",
       "49. 'MCF10A_WTvex20_under'\n",
       "50. 'MCF10A_WTvex9_over'\n",
       "51. 'MCF10A_WTvex9_res'\n",
       "52. 'MCF10A_WTvex9_resSig_padj'\n",
       "53. 'MCF10A_WTvex9_under'\n",
       "54. 'MCF10A_coldata'\n",
       "55. 'MCF10A_counts'\n",
       "56. 'MCF10A_dds'\n",
       "57. 'MCF10A_dds_f'\n",
       "58. 'MCF10A_diff_genes'\n",
       "59. 'MCF10A_ex9vex20_over'\n",
       "60. 'MCF10A_ex9vex20_res'\n",
       "61. 'MCF10A_ex9vex20_resSig_padj'\n",
       "62. 'MCF10A_ex9vex20_under'\n",
       "63. 'MCF10A_filt'\n",
       "64. 'MCF10A_genotype'\n",
       "65. 'MCF10A_norm_sigset'\n",
       "66. 'MCF10A_normalized_counts_f'\n",
       "67. 'MCF10A_normalized_counts_f_nonames'\n",
       "68. 'MCF10A_rlog'\n",
       "69. 'MCF10A_sampleDistMatrix'\n",
       "70. 'MCF10A_samples'\n",
       "71. 'MCF10A_vst'\n",
       "72. 'MCF7_corrected_hic_enriched'\n",
       "73. 'MCF7_parental_hic_enriched'\n",
       "74. 'MCF_7_counts'\n",
       "75. 'MCF_dual_E545K_counts'\n",
       "76. 'MCF_total_coldata'\n",
       "77. 'MCF_total_dds'\n",
       "78. 'MCF_total_dds_f'\n",
       "79. 'MCF_total_samples'\n",
       "80. 'MCF_total_treatment'\n",
       "81. 'MCF_total_vst'\n",
       "82. 'PIK3R1_exp'\n",
       "83. 'PRLR_exp'\n",
       "84. 'T47D_counts'\n",
       "85. 'TGFB_DN'\n",
       "86. 'TGFB_DN_genes'\n",
       "87. 'TGFB_DN_guides'\n",
       "88. 'TGFB_DN_targets'\n",
       "89. 'TGFB_UP'\n",
       "90. 'TGFB_UP_genes'\n",
       "91. 'TGFB_UP_guides'\n",
       "92. 'TGFB_UP_path'\n",
       "93. 'TGFB_UP_targets'\n",
       "94. 'TGFB_guides'\n",
       "95. 'THRB_exp'\n",
       "96. 'WT_expressed_E545K_interacting_overlap'\n",
       "97. 'WT_expressed_WT_interacting_overlap'\n",
       "98. 'WTvex20_GOBP'\n",
       "99. 'WTvex20_GOMF'\n",
       "100. 'WTvex20_KEGG'\n",
       "101. 'WTvex20_combined'\n",
       "102. 'WTvex20_combined_GOBP'\n",
       "103. 'WTvex20_combined_GOBP_nocan'\n",
       "104. 'WTvex20_combined_nocan'\n",
       "105. 'WTvex20_genes_log2fc_1.5'\n",
       "106. 'WTvex20_genes_log2fc_1.5_entrez'\n",
       "107. 'WTvex20_genes_log2fc_1.5_entrez_df'\n",
       "108. 'WTvex20_genes_padj'\n",
       "109. 'WTvex20_genes_padj_entrez'\n",
       "110. 'WTvex20_over_clean'\n",
       "111. 'WTvex20_over_clean_nocan'\n",
       "112. 'WTvex20_over_common'\n",
       "113. 'WTvex20_over_common_GOBP'\n",
       "114. 'WTvex20_over_common_GOBP_nocan'\n",
       "115. 'WTvex20_over_common_GOMF'\n",
       "116. 'WTvex20_over_nocan'\n",
       "117. 'WTvex20_res'\n",
       "118. 'WTvex20_resSig_padj'\n",
       "119. 'WTvex20_resSig_padj_df'\n",
       "120. 'WTvex20_resSig_padj_ordered'\n",
       "121. 'WTvex20_resSig_padj_ordered_df'\n",
       "122. 'WTvex20_res_df'\n",
       "123. 'WTvex20_under_clean'\n",
       "124. 'WTvex20_under_clean_nocan'\n",
       "125. 'WTvex20_under_common'\n",
       "126. 'WTvex20_under_common_GOBP'\n",
       "127. 'WTvex20_under_common_GOBP_nocan'\n",
       "128. 'WTvex20_under_common_GOMF'\n",
       "129. 'WTvex20_under_nocan'\n",
       "130. 'WTvex9_Comparison_gene_list_df'\n",
       "131. 'WTvex9_GOBP'\n",
       "132. 'WTvex9_GOMF'\n",
       "133. 'WTvex9_KEGG'\n",
       "134. 'WTvex9_combined'\n",
       "135. 'WTvex9_combined_GOBP'\n",
       "136. 'WTvex9_combined_GOBP_nocan'\n",
       "137. 'WTvex9_combined_nocan'\n",
       "138. 'WTvex9_genes_log2fc_1.5'\n",
       "139. 'WTvex9_genes_log2fc_1.5_entrez'\n",
       "140. 'WTvex9_genes_padj'\n",
       "141. 'WTvex9_genes_padj_entrez'\n",
       "142. 'WTvex9_over_clean'\n",
       "143. 'WTvex9_over_clean_nocan'\n",
       "144. 'WTvex9_over_common'\n",
       "145. 'WTvex9_over_common_GOBP'\n",
       "146. 'WTvex9_over_common_GOBP_nocan'\n",
       "147. 'WTvex9_over_common_GOMF'\n",
       "148. 'WTvex9_over_common_KEGG'\n",
       "149. 'WTvex9_over_nocan'\n",
       "150. 'WTvex9_res'\n",
       "151. 'WTvex9_resSig_padj'\n",
       "152. 'WTvex9_resSig_padj_df'\n",
       "153. 'WTvex9_resSig_padj_ordered'\n",
       "154. 'WTvex9_resSig_padj_ordered_df'\n",
       "155. 'WTvex9_res_df'\n",
       "156. 'WTvex9_under_clean'\n",
       "157. 'WTvex9_under_clean_nocan'\n",
       "158. 'WTvex9_under_common'\n",
       "159. 'WTvex9_under_common_GOBP'\n",
       "160. 'WTvex9_under_common_GOBP_nocan'\n",
       "161. 'WTvex9_under_common_GOMF'\n",
       "162. 'WTvex9_under_nocan'\n",
       "163. 'add.flag'\n",
       "164. 'all_diff_genes'\n",
       "165. 'bp'\n",
       "166. 'cancerDists'\n",
       "167. 'cancer_WTvex20_over'\n",
       "168. 'cancer_WTvex20_res'\n",
       "169. 'cancer_WTvex20_resSig_padj'\n",
       "170. 'cancer_WTvex20_under'\n",
       "171. 'cancer_WTvex9_over'\n",
       "172. 'cancer_WTvex9_over_clean'\n",
       "173. 'cancer_WTvex9_over_symbols'\n",
       "174. 'cancer_WTvex9_res'\n",
       "175. 'cancer_WTvex9_resSig_padj'\n",
       "176. 'cancer_WTvex9_under'\n",
       "177. 'cancer_WTvex9_under_clean'\n",
       "178. 'cancer_WTvex9_under_symbols'\n",
       "179. 'cancer_coldata'\n",
       "180. 'cancer_counts'\n",
       "181. 'cancer_dds'\n",
       "182. 'cancer_dds_f'\n",
       "183. 'cancer_diff_genes'\n",
       "184. 'cancer_ex9vex20_over'\n",
       "185. 'cancer_ex9vex20_res'\n",
       "186. 'cancer_ex9vex20_resSig_padj'\n",
       "187. 'cancer_ex9vex20_under'\n",
       "188. 'cancer_filt'\n",
       "189. 'cancer_genotype'\n",
       "190. 'cancer_norm_sigset'\n",
       "191. 'cancer_normalized_counts_f'\n",
       "192. 'cancer_normalized_counts_f_nonames'\n",
       "193. 'cancer_rlog'\n",
       "194. 'cancer_sampleDistMatrix'\n",
       "195. 'cancer_samples'\n",
       "196. 'cancer_vst'\n",
       "197. 'cleaningY'\n",
       "198. 'cleany'\n",
       "199. 'cleany2'\n",
       "200. 'cleany3'\n",
       "201. 'clinsub'\n",
       "202. 'cluster3'\n",
       "203. 'cluster3_GOBP'\n",
       "204. 'cluster3_genes'\n",
       "205. 'cluster3_genes_clean'\n",
       "206. 'cluster9'\n",
       "207. 'cluster9_GOBP'\n",
       "208. 'cluster9_genes'\n",
       "209. 'cluster9_genes_clean'\n",
       "210. 'cluster_df'\n",
       "211. 'clusters'\n",
       "212. 'colMeta_df'\n",
       "213. 'colors'\n",
       "214. 'counts_deseq_sva'\n",
       "215. 'cov'\n",
       "216. 'dat'\n",
       "217. 'datares_HC'\n",
       "218. 'dds'\n",
       "219. 'dds_LRT'\n",
       "220. 'ddssva'\n",
       "221. 'diff_WTvex20'\n",
       "222. 'diff_WTvex20_sig'\n",
       "223. 'diff_WTvex9'\n",
       "224. 'diff_WTvex9_sig'\n",
       "225. 'diff_ex9vex20'\n",
       "226. 'diff_ex9vex20_sig'\n",
       "227. 'dups'\n",
       "228. 'ens.str_WTvex20'\n",
       "229. 'ens.str_WTvex9'\n",
       "230. 'ens.str_ex9vex20'\n",
       "231. 'ex9vex20_GOBP'\n",
       "232. 'ex9vex20_GOMF'\n",
       "233. 'ex9vex20_KEGG'\n",
       "234. 'ex9vex20_combined'\n",
       "235. 'ex9vex20_combined_GOBP'\n",
       "236. 'ex9vex20_combined_GOBP_nocan'\n",
       "237. 'ex9vex20_combined_nocan'\n",
       "238. 'ex9vex20_genes_log2fc_1.5'\n",
       "239. 'ex9vex20_genes_log2fc_1.5_entrez'\n",
       "240. 'ex9vex20_genes_log2fc_1.5_entrez_df'\n",
       "241. 'ex9vex20_genes_padj'\n",
       "242. 'ex9vex20_genes_padj_entrez'\n",
       "243. 'ex9vex20_over_clean'\n",
       "244. 'ex9vex20_over_clean_nocan'\n",
       "245. 'ex9vex20_over_common'\n",
       "246. 'ex9vex20_over_common_GOBP'\n",
       "247. 'ex9vex20_over_common_GOBP_nocan'\n",
       "248. 'ex9vex20_over_common_GOMF'\n",
       "249. 'ex9vex20_over_nocan'\n",
       "250. 'ex9vex20_res'\n",
       "251. 'ex9vex20_resSig_padj'\n",
       "252. 'ex9vex20_resSig_padj_df'\n",
       "253. 'ex9vex20_resSig_padj_ordered'\n",
       "254. 'ex9vex20_resSig_padj_ordered_df'\n",
       "255. 'ex9vex20_res_df'\n",
       "256. 'ex9vex20_under_clean'\n",
       "257. 'ex9vex20_under_clean_nocan'\n",
       "258. 'ex9vex20_under_common'\n",
       "259. 'ex9vex20_under_common_GOBP'\n",
       "260. 'ex9vex20_under_common_GOBP_nocan'\n",
       "261. 'ex9vex20_under_common_GOMF'\n",
       "262. 'ex9vex20_under_nocan'\n",
       "263. 'exampleSet'\n",
       "264. 'expression'\n",
       "265. 'filt'\n",
       "266. 'fullGeneList'\n",
       "267. 'geneSetClustering'\n",
       "268. 'genes'\n",
       "269. 'genes2'\n",
       "270. 'geneset'\n",
       "271. 'geno_comp'\n",
       "272. 'geno_comp_filt'\n",
       "273. 'good_names'\n",
       "274. 'gr'\n",
       "275. 'heatmapres2.1'\n",
       "276. 'iSample'\n",
       "277. 'idx'\n",
       "278. 'kmax'\n",
       "279. 'lfcShrink_df'\n",
       "280. 'lfcShrink_df_absvalue'\n",
       "281. 'lfcShrink_df_absvalue_filt'\n",
       "282. 'list'\n",
       "283. 'lp'\n",
       "284. 'mart'\n",
       "285. 'mat2'\n",
       "286. 'mat2.1'\n",
       "287. 'mn'\n",
       "288. 'mod'\n",
       "289. 'mod0'\n",
       "290. 'myBreaks'\n",
       "291. 'myC6'\n",
       "292. 'myColor'\n",
       "293. 'my_palette'\n",
       "294. 'myvars'\n",
       "295. 'nSamples'\n",
       "296. 'nas'\n",
       "297. 'norm_subset_mat'\n",
       "298. 'norm_total_counts_sig_subset'\n",
       "299. 'normalized_MCF_total_counts'\n",
       "300. 'normalized_MCF_total_counts_f'\n",
       "301. 'normalized_MCF_total_counts_f_nonames'\n",
       "302. 'normalized_total_counts'\n",
       "303. 'normalized_total_counts_f'\n",
       "304. 'normalized_total_counts_f_nonames'\n",
       "305. 'normalized_total_counts_t'\n",
       "306. 'number_col'\n",
       "307. 'p'\n",
       "308. 'p_improved'\n",
       "309. 'paletteLength'\n",
       "310. 'pca'\n",
       "311. 'prePathfindR_df'\n",
       "312. 'progressBar'\n",
       "313. 'range'\n",
       "314. 'raw_total_counts_sig_subset'\n",
       "315. 'reorder_cells'\n",
       "316. 'res'\n",
       "317. 'resAshWTvex20'\n",
       "318. 'resAshWTvex9'\n",
       "319. 'resAshex9vex20'\n",
       "320. 'resCombo'\n",
       "321. 'resLRT'\n",
       "322. 'rld.sigtest2'\n",
       "323. 'rlog_dds'\n",
       "324. 'sampleDistMatrix'\n",
       "325. 'sampleDists'\n",
       "326. 'sample_info'\n",
       "327. 'secondp'\n",
       "328. 'selectGeneList'\n",
       "329. 'sets_of_int'\n",
       "330. 'sub'\n",
       "331. 'svseq'\n",
       "332. 'test'\n",
       "333. 'test_df'\n",
       "334. 'total_coldata'\n",
       "335. 'total_counts'\n",
       "336. 'total_counts_nonames'\n",
       "337. 'total_counts_t'\n",
       "338. 'total_dds'\n",
       "339. 'total_dds_f'\n",
       "340. 'total_rlog'\n",
       "341. 'total_samples'\n",
       "342. 'total_treatment'\n",
       "343. 'total_vst'\n",
       "344. 'venn_data'\n",
       "345. 'vsd'\n",
       "346. 'wss'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "  [1] \"AREG_exp\"                                 \n",
       "  [2] \"AREG_exp_raw\"                             \n",
       "  [3] \"CRISPR_guides\"                            \n",
       "  [4] \"E545K_expressed_E545K_interacting_overlap\"\n",
       "  [5] \"E545K_expressed_WT_interacting_overlap\"   \n",
       "  [6] \"E545K_expressed_interacting_overlap\"      \n",
       "  [7] \"ETV5_exp\"                                 \n",
       "  [8] \"G_list\"                                   \n",
       "  [9] \"Gene_targets\"                             \n",
       " [10] \"Genotype_vec\"                             \n",
       " [11] \"HTertDists\"                               \n",
       " [12] \"HTert_WT_counts\"                          \n",
       " [13] \"HTert_WTvex20_over\"                       \n",
       " [14] \"HTert_WTvex20_res\"                        \n",
       " [15] \"HTert_WTvex20_resSig_padj\"                \n",
       " [16] \"HTert_WTvex20_under\"                      \n",
       " [17] \"HTert_WTvex9_over\"                        \n",
       " [18] \"HTert_WTvex9_res\"                         \n",
       " [19] \"HTert_WTvex9_resSig_padj\"                 \n",
       " [20] \"HTert_WTvex9_under\"                       \n",
       " [21] \"HTert_coldata\"                            \n",
       " [22] \"HTert_counts\"                             \n",
       " [23] \"HTert_dds\"                                \n",
       " [24] \"HTert_dds_f\"                              \n",
       " [25] \"HTert_diff_genes\"                         \n",
       " [26] \"HTert_ex20_counts\"                        \n",
       " [27] \"HTert_ex9_counts\"                         \n",
       " [28] \"HTert_ex9vex20_over\"                      \n",
       " [29] \"HTert_ex9vex20_res\"                       \n",
       " [30] \"HTert_ex9vex20_resSig_padj\"               \n",
       " [31] \"HTert_ex9vex20_under\"                     \n",
       " [32] \"HTert_filt\"                               \n",
       " [33] \"HTert_genotype\"                           \n",
       " [34] \"HTert_norm_sigset\"                        \n",
       " [35] \"HTert_normalized_counts_f\"                \n",
       " [36] \"HTert_normalized_counts_f_nonames\"        \n",
       " [37] \"HTert_rlog\"                               \n",
       " [38] \"HTert_sampleDistMatrix\"                   \n",
       " [39] \"HTert_samples\"                            \n",
       " [40] \"HTert_vst\"                                \n",
       " [41] \"HiC_overlap_venns\"                        \n",
       " [42] \"IRS2_exp\"                                 \n",
       " [43] \"Lineage_vec\"                              \n",
       " [44] \"MCF10ADists\"                              \n",
       " [45] \"MCF10A_H1047R_counts\"                     \n",
       " [46] \"MCF10A_WTvex20_over\"                      \n",
       " [47] \"MCF10A_WTvex20_res\"                       \n",
       " [48] \"MCF10A_WTvex20_resSig_padj\"               \n",
       " [49] \"MCF10A_WTvex20_under\"                     \n",
       " [50] \"MCF10A_WTvex9_over\"                       \n",
       " [51] \"MCF10A_WTvex9_res\"                        \n",
       " [52] \"MCF10A_WTvex9_resSig_padj\"                \n",
       " [53] \"MCF10A_WTvex9_under\"                      \n",
       " [54] \"MCF10A_coldata\"                           \n",
       " [55] \"MCF10A_counts\"                            \n",
       " [56] \"MCF10A_dds\"                               \n",
       " [57] \"MCF10A_dds_f\"                             \n",
       " [58] \"MCF10A_diff_genes\"                        \n",
       " [59] \"MCF10A_ex9vex20_over\"                     \n",
       " [60] \"MCF10A_ex9vex20_res\"                      \n",
       " [61] \"MCF10A_ex9vex20_resSig_padj\"              \n",
       " [62] \"MCF10A_ex9vex20_under\"                    \n",
       " [63] \"MCF10A_filt\"                              \n",
       " [64] \"MCF10A_genotype\"                          \n",
       " [65] \"MCF10A_norm_sigset\"                       \n",
       " [66] \"MCF10A_normalized_counts_f\"               \n",
       " [67] \"MCF10A_normalized_counts_f_nonames\"       \n",
       " [68] \"MCF10A_rlog\"                              \n",
       " [69] \"MCF10A_sampleDistMatrix\"                  \n",
       " [70] \"MCF10A_samples\"                           \n",
       " [71] \"MCF10A_vst\"                               \n",
       " [72] \"MCF7_corrected_hic_enriched\"              \n",
       " [73] \"MCF7_parental_hic_enriched\"               \n",
       " [74] \"MCF_7_counts\"                             \n",
       " [75] \"MCF_dual_E545K_counts\"                    \n",
       " [76] \"MCF_total_coldata\"                        \n",
       " [77] \"MCF_total_dds\"                            \n",
       " [78] \"MCF_total_dds_f\"                          \n",
       " [79] \"MCF_total_samples\"                        \n",
       " [80] \"MCF_total_treatment\"                      \n",
       " [81] \"MCF_total_vst\"                            \n",
       " [82] \"PIK3R1_exp\"                               \n",
       " [83] \"PRLR_exp\"                                 \n",
       " [84] \"T47D_counts\"                              \n",
       " [85] \"TGFB_DN\"                                  \n",
       " [86] \"TGFB_DN_genes\"                            \n",
       " [87] \"TGFB_DN_guides\"                           \n",
       " [88] \"TGFB_DN_targets\"                          \n",
       " [89] \"TGFB_UP\"                                  \n",
       " [90] \"TGFB_UP_genes\"                            \n",
       " [91] \"TGFB_UP_guides\"                           \n",
       " [92] \"TGFB_UP_path\"                             \n",
       " [93] \"TGFB_UP_targets\"                          \n",
       " [94] \"TGFB_guides\"                              \n",
       " [95] \"THRB_exp\"                                 \n",
       " [96] \"WT_expressed_E545K_interacting_overlap\"   \n",
       " [97] \"WT_expressed_WT_interacting_overlap\"      \n",
       " [98] \"WTvex20_GOBP\"                             \n",
       " [99] \"WTvex20_GOMF\"                             \n",
       "[100] \"WTvex20_KEGG\"                             \n",
       "[101] \"WTvex20_combined\"                         \n",
       "[102] \"WTvex20_combined_GOBP\"                    \n",
       "[103] \"WTvex20_combined_GOBP_nocan\"              \n",
       "[104] \"WTvex20_combined_nocan\"                   \n",
       "[105] \"WTvex20_genes_log2fc_1.5\"                 \n",
       "[106] \"WTvex20_genes_log2fc_1.5_entrez\"          \n",
       "[107] \"WTvex20_genes_log2fc_1.5_entrez_df\"       \n",
       "[108] \"WTvex20_genes_padj\"                       \n",
       "[109] \"WTvex20_genes_padj_entrez\"                \n",
       "[110] \"WTvex20_over_clean\"                       \n",
       "[111] \"WTvex20_over_clean_nocan\"                 \n",
       "[112] \"WTvex20_over_common\"                      \n",
       "[113] \"WTvex20_over_common_GOBP\"                 \n",
       "[114] \"WTvex20_over_common_GOBP_nocan\"           \n",
       "[115] \"WTvex20_over_common_GOMF\"                 \n",
       "[116] \"WTvex20_over_nocan\"                       \n",
       "[117] \"WTvex20_res\"                              \n",
       "[118] \"WTvex20_resSig_padj\"                      \n",
       "[119] \"WTvex20_resSig_padj_df\"                   \n",
       "[120] \"WTvex20_resSig_padj_ordered\"              \n",
       "[121] \"WTvex20_resSig_padj_ordered_df\"           \n",
       "[122] \"WTvex20_res_df\"                           \n",
       "[123] \"WTvex20_under_clean\"                      \n",
       "[124] \"WTvex20_under_clean_nocan\"                \n",
       "[125] \"WTvex20_under_common\"                     \n",
       "[126] \"WTvex20_under_common_GOBP\"                \n",
       "[127] \"WTvex20_under_common_GOBP_nocan\"          \n",
       "[128] \"WTvex20_under_common_GOMF\"                \n",
       "[129] \"WTvex20_under_nocan\"                      \n",
       "[130] \"WTvex9_Comparison_gene_list_df\"           \n",
       "[131] \"WTvex9_GOBP\"                              \n",
       "[132] \"WTvex9_GOMF\"                              \n",
       "[133] \"WTvex9_KEGG\"                              \n",
       "[134] \"WTvex9_combined\"                          \n",
       "[135] \"WTvex9_combined_GOBP\"                     \n",
       "[136] \"WTvex9_combined_GOBP_nocan\"               \n",
       "[137] \"WTvex9_combined_nocan\"                    \n",
       "[138] \"WTvex9_genes_log2fc_1.5\"                  \n",
       "[139] \"WTvex9_genes_log2fc_1.5_entrez\"           \n",
       "[140] \"WTvex9_genes_padj\"                        \n",
       "[141] \"WTvex9_genes_padj_entrez\"                 \n",
       "[142] \"WTvex9_over_clean\"                        \n",
       "[143] \"WTvex9_over_clean_nocan\"                  \n",
       "[144] \"WTvex9_over_common\"                       \n",
       "[145] \"WTvex9_over_common_GOBP\"                  \n",
       "[146] \"WTvex9_over_common_GOBP_nocan\"            \n",
       "[147] \"WTvex9_over_common_GOMF\"                  \n",
       "[148] \"WTvex9_over_common_KEGG\"                  \n",
       "[149] \"WTvex9_over_nocan\"                        \n",
       "[150] \"WTvex9_res\"                               \n",
       "[151] \"WTvex9_resSig_padj\"                       \n",
       "[152] \"WTvex9_resSig_padj_df\"                    \n",
       "[153] \"WTvex9_resSig_padj_ordered\"               \n",
       "[154] \"WTvex9_resSig_padj_ordered_df\"            \n",
       "[155] \"WTvex9_res_df\"                            \n",
       "[156] \"WTvex9_under_clean\"                       \n",
       "[157] \"WTvex9_under_clean_nocan\"                 \n",
       "[158] \"WTvex9_under_common\"                      \n",
       "[159] \"WTvex9_under_common_GOBP\"                 \n",
       "[160] \"WTvex9_under_common_GOBP_nocan\"           \n",
       "[161] \"WTvex9_under_common_GOMF\"                 \n",
       "[162] \"WTvex9_under_nocan\"                       \n",
       "[163] \"add.flag\"                                 \n",
       "[164] \"all_diff_genes\"                           \n",
       "[165] \"bp\"                                       \n",
       "[166] \"cancerDists\"                              \n",
       "[167] \"cancer_WTvex20_over\"                      \n",
       "[168] \"cancer_WTvex20_res\"                       \n",
       "[169] \"cancer_WTvex20_resSig_padj\"               \n",
       "[170] \"cancer_WTvex20_under\"                     \n",
       "[171] \"cancer_WTvex9_over\"                       \n",
       "[172] \"cancer_WTvex9_over_clean\"                 \n",
       "[173] \"cancer_WTvex9_over_symbols\"               \n",
       "[174] \"cancer_WTvex9_res\"                        \n",
       "[175] \"cancer_WTvex9_resSig_padj\"                \n",
       "[176] \"cancer_WTvex9_under\"                      \n",
       "[177] \"cancer_WTvex9_under_clean\"                \n",
       "[178] \"cancer_WTvex9_under_symbols\"              \n",
       "[179] \"cancer_coldata\"                           \n",
       "[180] \"cancer_counts\"                            \n",
       "[181] \"cancer_dds\"                               \n",
       "[182] \"cancer_dds_f\"                             \n",
       "[183] \"cancer_diff_genes\"                        \n",
       "[184] \"cancer_ex9vex20_over\"                     \n",
       "[185] \"cancer_ex9vex20_res\"                      \n",
       "[186] \"cancer_ex9vex20_resSig_padj\"              \n",
       "[187] \"cancer_ex9vex20_under\"                    \n",
       "[188] \"cancer_filt\"                              \n",
       "[189] \"cancer_genotype\"                          \n",
       "[190] \"cancer_norm_sigset\"                       \n",
       "[191] \"cancer_normalized_counts_f\"               \n",
       "[192] \"cancer_normalized_counts_f_nonames\"       \n",
       "[193] \"cancer_rlog\"                              \n",
       "[194] \"cancer_sampleDistMatrix\"                  \n",
       "[195] \"cancer_samples\"                           \n",
       "[196] \"cancer_vst\"                               \n",
       "[197] \"cleaningY\"                                \n",
       "[198] \"cleany\"                                   \n",
       "[199] \"cleany2\"                                  \n",
       "[200] \"cleany3\"                                  \n",
       "[201] \"clinsub\"                                  \n",
       "[202] \"cluster3\"                                 \n",
       "[203] \"cluster3_GOBP\"                            \n",
       "[204] \"cluster3_genes\"                           \n",
       "[205] \"cluster3_genes_clean\"                     \n",
       "[206] \"cluster9\"                                 \n",
       "[207] \"cluster9_GOBP\"                            \n",
       "[208] \"cluster9_genes\"                           \n",
       "[209] \"cluster9_genes_clean\"                     \n",
       "[210] \"cluster_df\"                               \n",
       "[211] \"clusters\"                                 \n",
       "[212] \"colMeta_df\"                               \n",
       "[213] \"colors\"                                   \n",
       "[214] \"counts_deseq_sva\"                         \n",
       "[215] \"cov\"                                      \n",
       "[216] \"dat\"                                      \n",
       "[217] \"datares_HC\"                               \n",
       "[218] \"dds\"                                      \n",
       "[219] \"dds_LRT\"                                  \n",
       "[220] \"ddssva\"                                   \n",
       "[221] \"diff_WTvex20\"                             \n",
       "[222] \"diff_WTvex20_sig\"                         \n",
       "[223] \"diff_WTvex9\"                              \n",
       "[224] \"diff_WTvex9_sig\"                          \n",
       "[225] \"diff_ex9vex20\"                            \n",
       "[226] \"diff_ex9vex20_sig\"                        \n",
       "[227] \"dups\"                                     \n",
       "[228] \"ens.str_WTvex20\"                          \n",
       "[229] \"ens.str_WTvex9\"                           \n",
       "[230] \"ens.str_ex9vex20\"                         \n",
       "[231] \"ex9vex20_GOBP\"                            \n",
       "[232] \"ex9vex20_GOMF\"                            \n",
       "[233] \"ex9vex20_KEGG\"                            \n",
       "[234] \"ex9vex20_combined\"                        \n",
       "[235] \"ex9vex20_combined_GOBP\"                   \n",
       "[236] \"ex9vex20_combined_GOBP_nocan\"             \n",
       "[237] \"ex9vex20_combined_nocan\"                  \n",
       "[238] \"ex9vex20_genes_log2fc_1.5\"                \n",
       "[239] \"ex9vex20_genes_log2fc_1.5_entrez\"         \n",
       "[240] \"ex9vex20_genes_log2fc_1.5_entrez_df\"      \n",
       "[241] \"ex9vex20_genes_padj\"                      \n",
       "[242] \"ex9vex20_genes_padj_entrez\"               \n",
       "[243] \"ex9vex20_over_clean\"                      \n",
       "[244] \"ex9vex20_over_clean_nocan\"                \n",
       "[245] \"ex9vex20_over_common\"                     \n",
       "[246] \"ex9vex20_over_common_GOBP\"                \n",
       "[247] \"ex9vex20_over_common_GOBP_nocan\"          \n",
       "[248] \"ex9vex20_over_common_GOMF\"                \n",
       "[249] \"ex9vex20_over_nocan\"                      \n",
       "[250] \"ex9vex20_res\"                             \n",
       "[251] \"ex9vex20_resSig_padj\"                     \n",
       "[252] \"ex9vex20_resSig_padj_df\"                  \n",
       "[253] \"ex9vex20_resSig_padj_ordered\"             \n",
       "[254] \"ex9vex20_resSig_padj_ordered_df\"          \n",
       "[255] \"ex9vex20_res_df\"                          \n",
       "[256] \"ex9vex20_under_clean\"                     \n",
       "[257] \"ex9vex20_under_clean_nocan\"               \n",
       "[258] \"ex9vex20_under_common\"                    \n",
       "[259] \"ex9vex20_under_common_GOBP\"               \n",
       "[260] \"ex9vex20_under_common_GOBP_nocan\"         \n",
       "[261] \"ex9vex20_under_common_GOMF\"               \n",
       "[262] \"ex9vex20_under_nocan\"                     \n",
       "[263] \"exampleSet\"                               \n",
       "[264] \"expression\"                               \n",
       "[265] \"filt\"                                     \n",
       "[266] \"fullGeneList\"                             \n",
       "[267] \"geneSetClustering\"                        \n",
       "[268] \"genes\"                                    \n",
       "[269] \"genes2\"                                   \n",
       "[270] \"geneset\"                                  \n",
       "[271] \"geno_comp\"                                \n",
       "[272] \"geno_comp_filt\"                           \n",
       "[273] \"good_names\"                               \n",
       "[274] \"gr\"                                       \n",
       "[275] \"heatmapres2.1\"                            \n",
       "[276] \"iSample\"                                  \n",
       "[277] \"idx\"                                      \n",
       "[278] \"kmax\"                                     \n",
       "[279] \"lfcShrink_df\"                             \n",
       "[280] \"lfcShrink_df_absvalue\"                    \n",
       "[281] \"lfcShrink_df_absvalue_filt\"               \n",
       "[282] \"list\"                                     \n",
       "[283] \"lp\"                                       \n",
       "[284] \"mart\"                                     \n",
       "[285] \"mat2\"                                     \n",
       "[286] \"mat2.1\"                                   \n",
       "[287] \"mn\"                                       \n",
       "[288] \"mod\"                                      \n",
       "[289] \"mod0\"                                     \n",
       "[290] \"myBreaks\"                                 \n",
       "[291] \"myC6\"                                     \n",
       "[292] \"myColor\"                                  \n",
       "[293] \"my_palette\"                               \n",
       "[294] \"myvars\"                                   \n",
       "[295] \"nSamples\"                                 \n",
       "[296] \"nas\"                                      \n",
       "[297] \"norm_subset_mat\"                          \n",
       "[298] \"norm_total_counts_sig_subset\"             \n",
       "[299] \"normalized_MCF_total_counts\"              \n",
       "[300] \"normalized_MCF_total_counts_f\"            \n",
       "[301] \"normalized_MCF_total_counts_f_nonames\"    \n",
       "[302] \"normalized_total_counts\"                  \n",
       "[303] \"normalized_total_counts_f\"                \n",
       "[304] \"normalized_total_counts_f_nonames\"        \n",
       "[305] \"normalized_total_counts_t\"                \n",
       "[306] \"number_col\"                               \n",
       "[307] \"p\"                                        \n",
       "[308] \"p_improved\"                               \n",
       "[309] \"paletteLength\"                            \n",
       "[310] \"pca\"                                      \n",
       "[311] \"prePathfindR_df\"                          \n",
       "[312] \"progressBar\"                              \n",
       "[313] \"range\"                                    \n",
       "[314] \"raw_total_counts_sig_subset\"              \n",
       "[315] \"reorder_cells\"                            \n",
       "[316] \"res\"                                      \n",
       "[317] \"resAshWTvex20\"                            \n",
       "[318] \"resAshWTvex9\"                             \n",
       "[319] \"resAshex9vex20\"                           \n",
       "[320] \"resCombo\"                                 \n",
       "[321] \"resLRT\"                                   \n",
       "[322] \"rld.sigtest2\"                             \n",
       "[323] \"rlog_dds\"                                 \n",
       "[324] \"sampleDistMatrix\"                         \n",
       "[325] \"sampleDists\"                              \n",
       "[326] \"sample_info\"                              \n",
       "[327] \"secondp\"                                  \n",
       "[328] \"selectGeneList\"                           \n",
       "[329] \"sets_of_int\"                              \n",
       "[330] \"sub\"                                      \n",
       "[331] \"svseq\"                                    \n",
       "[332] \"test\"                                     \n",
       "[333] \"test_df\"                                  \n",
       "[334] \"total_coldata\"                            \n",
       "[335] \"total_counts\"                             \n",
       "[336] \"total_counts_nonames\"                     \n",
       "[337] \"total_counts_t\"                           \n",
       "[338] \"total_dds\"                                \n",
       "[339] \"total_dds_f\"                              \n",
       "[340] \"total_rlog\"                               \n",
       "[341] \"total_samples\"                            \n",
       "[342] \"total_treatment\"                          \n",
       "[343] \"total_vst\"                                \n",
       "[344] \"venn_data\"                                \n",
       "[345] \"vsd\"                                      \n",
       "[346] \"wss\"                                      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "library(\"tidyverse\")\n",
    "library(\"org.Hs.eg.db\")\n",
    "library(\"ggrepel\")\n",
    "library(\"DESeq2\")\n",
    "library(\"vsn\")\n",
    "library(\"RColorBrewer\")\n",
    "library(\"AnnotationDbi\")\n",
    "library(\"org.Hs.eg.db\")\n",
    "library(\"ReportingTools\")\n",
    "library(\"sva\")\n",
    "library(\"GSVA\")\n",
    "library(\"GSEABase\")\n",
    "library(\"Biobase\")\n",
    "library(\"genefilter\")\n",
    "library(\"gplots\")\n",
    "library(\"pheatmap\")\n",
    "library(\"limma\")\n",
    "library(\"stats\")\n",
    "library(\"grid\")\n",
    "library(\"pathfindR\")\n",
    "library(\"magick\")\n",
    "setwd(\"/data/park_lab/adam/Total_RNA\")\n",
    "load(\"model_env.RData\")\n",
    "ls()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0b9cb4b7",
   "metadata": {},
   "source": [
    "This is my total raw counts file from my Total RNA-seq environment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9f4e61ab",
   "metadata": {},
   "outputs": [],
   "source": [
    "total_counts"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "25a6117a",
   "metadata": {},
   "source": [
    "Here we create an object that defines the genotypes and lineages for each of our samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4e475e8d",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "Lineage_vec <- c(rep(\"MCF10A\", 6),\n",
    "                  rep(\"cancer\", 6),\n",
    "                  rep(\"MCF10A\", 3),\n",
    "                  rep(\"HTert\", 9),\n",
    "                  rep(\"cancer\", 3))\n",
    "Lineage_vec\n",
    "Genotype_vec <- c(rep(\"ex9\", 3), rep(\"WT\", 6), rep(\"ex9\", 3), rep(\"ex20\", 3),\n",
    "                  rep(\"WT\", 3), rep(\"ex9\", 3), rep(\"ex20\", 6))\n",
    "Genotype_vec\n",
    "sample_info <- data.frame(Lineage_vec, Genotype_vec)\n",
    "rownames(sample_info) <- rownames(total_counts_t)\n",
    "sample_info$Combo <- as.factor(paste(sample_info$Lineage_vec, sample_info$Genotype_vec))\n",
    "names(sample_info) <- c(\"lineage\", \"genotype\", \"combo\")\n",
    "sample_info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "285711f7",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "mn <- apply(total_counts_t,1,mean)\n",
    "hist(log2(mn), breaks = 100)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "04a80d13",
   "metadata": {},
   "outputs": [],
   "source": [
    "hist(log10(mn), breaks = 100)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c8ebd098",
   "metadata": {},
   "source": [
    "Create a DESeq model with genotype and lineage as our variables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1ce7ec05",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "dds <- DESeqDataSetFromMatrix(countData = total_counts,\n",
    "                              colData = sample_info,\n",
    "                              design = ~lineage + genotype)\n",
    "str(dds)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a2d6a578",
   "metadata": {},
   "outputs": [],
   "source": [
    "vsd <- vst(dds, blind = FALSE)\n",
    "head(assay(vsd), 3)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "405e66bc",
   "metadata": {},
   "source": [
    "Here we create a heatmap that displays the similarities of our samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7e53f726",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "sampleDists <- dist(t(assay(vsd)))\n",
    "sampleDists\n",
    "sampleDistMatrix <- as.matrix( sampleDists )\n",
    "rownames( sampleDistMatrix ) <- vsd$lineage\n",
    "colnames(sampleDistMatrix) <- NULL\n",
    "colors <- colorRampPalette( rev(brewer.pal(9, \"Blues\")) )(255)\n",
    "p <- pheatmap(sampleDistMatrix,\n",
    "         clustering_distance_rows = sampleDists,\n",
    "         clustering_distance_cols = sampleDists,\n",
    "         col = colors)\n",
    "p"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "08cf8238",
   "metadata": {},
   "outputs": [],
   "source": [
    "library(repr)\n",
    "pdf(file = \"dist_heatmap.pdf\",\n",
    "   width = 8,\n",
    "   height = 8)\n",
    "options(repr.plot.width=8, repr.plot.height=8)\n",
    "p <- pheatmap(sampleDistMatrix,\n",
    "         clustering_distance_rows = sampleDists,\n",
    "         clustering_distance_cols = sampleDists,\n",
    "         col = colors)\n",
    "p\n",
    "dev.off"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "863434bc",
   "metadata": {},
   "source": [
    "Our PCA's show differences between our cell lines not defined by genotype, but rather by lineage"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "de82d44d",
   "metadata": {},
   "outputs": [],
   "source": [
    "plotPCA(vsd, intgroup = c(\"genotype\"))\n",
    "plotPCA(vsd, intgroup = c(\"lineage\"))\n",
    "plotPCA(vsd, intgroup = c(\"combo\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "366d5c13",
   "metadata": {},
   "outputs": [],
   "source": [
    "pdf(file=\"genotype.pca.pdf\")\n",
    "plotPCA(vsd, intgroup = c(\"genotype\"))\n",
    "dev.off()\n",
    "\n",
    "pdf(file=\"lineage.pca.pdf\")\n",
    "plotPCA(vsd, intgroup = c(\"lineage\"))\n",
    "dev.off()\n",
    "\n",
    "pdf(file=\"combo.pca.pdf\")\n",
    "plotPCA(vsd, intgroup = c(\"combo\"))\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "af52439d",
   "metadata": {},
   "outputs": [],
   "source": [
    "pca <- prcomp(total_counts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0b343eb6",
   "metadata": {},
   "outputs": [],
   "source": [
    "library(pca3d)\n",
    "gr <- sample_info$lineage\n",
    "options(rgl.useNULL = TRUE)\n",
    "pca3d(pca, group=gr)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "266eecc4",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "dds <- DESeq(dds)\n",
    "res <- results(dds)\n",
    "res <- res[order(res$padj),]\n",
    "head(res)\n",
    "\n",
    "plotMA(dds,ylim=c(-2,2),main=\"DESeq2\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "67f35dee",
   "metadata": {},
   "source": [
    "## GSVA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0c6d0cb5",
   "metadata": {},
   "outputs": [],
   "source": [
    "myC6 <- getGmt(\"/data/park_lab/adam/Total_RNA/c6.all.v7.5.1.entrez.gmt\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "78fac406",
   "metadata": {},
   "outputs": [],
   "source": [
    "library('limma')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0cc7d15e",
   "metadata": {},
   "outputs": [],
   "source": [
    "dat  <- counts(dds, normalized = TRUE)\n",
    "idx  <- rowMeans(dat) > 1\n",
    "dat  <- dat[idx, ]\n",
    "mod  <- model.matrix(~genotype, colData(dds))\n",
    "mod0 <- model.matrix(~1, colData(dds))\n",
    "svseq <- svaseq(dat, mod, mod0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "86485e5c",
   "metadata": {},
   "outputs": [],
   "source": [
    "ddssva <- dds\n",
    "ddssva$SV1 <- svseq$sv[,1]\n",
    "ddssva$SV2 <- svseq$sv[,2]\n",
    "ddssva$SV3 <- svseq$sv[,3]\n",
    "design(ddssva) <- ~SV1 + SV2 + SV3 + genotype\n",
    "ddssva <- DESeq(ddssva)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "56b4f407",
   "metadata": {},
   "outputs": [],
   "source": [
    "vsd_sva <- vst(ddssva, blind = FALSE)\n",
    "head(assay(vsd_sva), 3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "99a84593",
   "metadata": {},
   "outputs": [],
   "source": [
    "plotPCA(vsd_sva, intgroup = c(\"genotype\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e0f5adcb",
   "metadata": {},
   "outputs": [],
   "source": [
    "resCombo <- results(ddssva,\n",
    "                    contrast=c(\"genotype\", \"ex9\", \"ex20\"))\n",
    "\n",
    "head(resCombo)\n",
    "table(resCombo$padj < 0.05)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bb99eb22",
   "metadata": {},
   "outputs": [],
   "source": [
    "cov = cbind(svseq$sv[,1:3])\n",
    "counts_deseq_sva <- removeBatchEffect(dat, covariates = cov)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1f52f118",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "counts_deseq_sva"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7b184a95",
   "metadata": {},
   "outputs": [],
   "source": [
    "cleaningY = function(y, mod, svaobj) {\n",
    "X=cbind(mod,svaobj$sv)\n",
    "Hat=solve(t(X)%*%X)%*%t(X)\n",
    " beta=(Hat%*%t(y))\n",
    "P=ncol(mod)\n",
    "cleany=y-t(as.matrix(X[,-c(1:P)])%*%beta[-c(1:P),])\n",
    " return(cleany)\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b5543fd3",
   "metadata": {},
   "outputs": [],
   "source": [
    "cleany = cleaningY(dat,mod,svseq)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "954beace",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "cleany"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f4f30574",
   "metadata": {},
   "outputs": [],
   "source": [
    "good_names <- rownames(cleany)\n",
    "good_names <- tools::file_path_sans_ext(c(good_names))\n",
    "good_names\n",
    "rownames(cleany) <- good_names\n",
    "head(cleany)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b82c3bb6",
   "metadata": {},
   "outputs": [],
   "source": [
    "genes2 <- mapIds(org.Hs.eg.db,\n",
    "                     keys=rownames(cleany),\n",
    "                     column=\"ENTREZID\",\n",
    "                     keytype=\"ENSEMBL\",\n",
    "                     multiVals=\"first\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3a6c08e9",
   "metadata": {},
   "outputs": [],
   "source": [
    "rownames(cleany) = genes2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b0cce472",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "cleany"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ba847d0d",
   "metadata": {},
   "outputs": [],
   "source": [
    "nas=na.omit(rownames(cleany))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "507209ac",
   "metadata": {},
   "outputs": [],
   "source": [
    "cleany2sss=cleany[-na.action(nas),]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2e89c5c7",
   "metadata": {},
   "outputs": [],
   "source": [
    "dups <- which(duplicated(rownames(cleany2)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "47ba0527",
   "metadata": {},
   "outputs": [],
   "source": [
    "cleany3=cleany2[-dups,]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "77cb619f",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "cleany3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d21ea5a5",
   "metadata": {},
   "outputs": [],
   "source": [
    "colnames(cleany3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ab45b1c2",
   "metadata": {},
   "outputs": [],
   "source": [
    "sub = c(1,2,3,10,11,12,13,14,15,19,20,21,22,23,24,25,26,27)\n",
    "exampleSet <- ExpressionSet(assayData = cleany3[,sub])\n",
    "test = gsva(exprs(exampleSet), myC6, method = \"ssgsea\",\n",
    "            kcdf = \"Poisson\", verbose=TRUE,\n",
    "            parallel.sz=1, ssgsea.norm = F)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7e8fb516",
   "metadata": {},
   "outputs": [],
   "source": [
    "myvars <- apply(test,1, var,na.rm=TRUE) \n",
    "myvars <- sort(myvars,decreasing=TRUE) \n",
    "length(myvars)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7e9e2dd7",
   "metadata": {},
   "outputs": [],
   "source": [
    "expression <- test[names(myvars),] \n",
    "dim(expression)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5267e41c",
   "metadata": {},
   "outputs": [],
   "source": [
    "clinsub=sample_info[sub,]\n",
    "clinsub"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "21dbb100",
   "metadata": {},
   "outputs": [],
   "source": [
    "lp = as.factor(clinsub[,3])\n",
    "secondp = as.factor(clinsub[,2])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "209e81a5",
   "metadata": {},
   "outputs": [],
   "source": [
    "expression"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5cc0971b",
   "metadata": {},
   "outputs": [],
   "source": [
    "bp=brewer.pal(10,\"Spectral\")\n",
    "lp = as.factor(clinsub[,3])\n",
    "secondp = as.factor(clinsub[,2])\n",
    "#grid.newpage()\n",
    "pdf(\"~/E545KvH1047R.ssgsea.pdf\")\n",
    "my_palette <- colorRampPalette(c(\"purple\", \"white\", \"green\"))(n = 1000)\n",
    "heatmap.2(expression,\n",
    "          trace = \"none\",\n",
    "          col = my_palette,\n",
    "          margins = c(5,25),\n",
    "          labCol = \"\",\n",
    "          cexRow = .7, ColSideColors = bp[lp]\n",
    "          key = TRUE, scale = \"row\")\n",
    "legend(\"topright\",legend = unique(lp), fill = bp[unique(lp)])\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cee77163",
   "metadata": {},
   "outputs": [],
   "source": [
    "geneSetClustering <- hclust(as.dist(1-cor(t(expression), method=\"kendall\")),\n",
    "                            method=\"complete\")\n",
    "heatmap.2(expression,\n",
    "          trace = \"none\",\n",
    "          col = my_palette,\n",
    "          margins = c(5,25),\n",
    "          labCol = \"\",\n",
    "          cexRow = .7, ColSideColors = bp[lp],\n",
    "          key = TRUE, scale = \"row\",\n",
    "          Rowv=as.dendrogram(geneSetClustering))\n",
    "legend(\"topright\",legend = unique(lp), fill = bp[unique(lp)])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "775a5c09",
   "metadata": {},
   "outputs": [],
   "source": [
    "pdf(\"~/E545KvH1047R_clusteredy.ssgsea.pdf\")\n",
    "my_palette <- colorRampPalette(c(\"purple\", \"white\", \"green\"))(n = 1000)\n",
    "heatmap.2(expression,\n",
    "          trace = \"none\",\n",
    "          col = my_palette,\n",
    "          margins = c(5,25),\n",
    "          labCol = \"\",\n",
    "          cexRow = .7, ColSideColors = bp[lp],\n",
    "          key = TRUE, scale = \"row\",\n",
    "         Rowv=as.dendrogram(geneSetClustering))\n",
    "legend(\"topright\",legend = unique(lp), fill = bp[unique(lp)])\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f2c801da",
   "metadata": {},
   "outputs": [],
   "source": [
    "heatmap.2(expression,\n",
    "          trace = \"none\",\n",
    "          col = my_palette,\n",
    "          margins = c(5,25),\n",
    "          labCol = \"\",\n",
    "          cexRow = .7, ColSideColors = bp[lp],\n",
    "          key = TRUE, scale = \"row\",\n",
    "         )\n",
    "legend(\"topright\",legend = unique(lp), fill = bp[unique(lp)])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a0447595",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "CRISPR_guides <- read_tsv(file = \"MCF10A_guides.txt\")\n",
    "CRISPR_guides"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d767bb4b",
   "metadata": {},
   "outputs": [],
   "source": [
    "Gene_targets <- CRISPR_guides %>%\n",
    "                dplyr::select(Target.Gene.Symbol) %>%\n",
    "                unique() %>%\n",
    "                pull(Target.Gene.Symbol) %>%\n",
    "                sort()\n",
    "Gene_targets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "371bd599",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "test_df <- as.data.frame(test) %>%\n",
    "            tibble::rownames_to_column(\"PATHWAY\")\n",
    "number_col <- seq(1, 189, by=1)\n",
    "test_df$Number_ID <- number_col\n",
    "test_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "627ffe7d",
   "metadata": {},
   "outputs": [],
   "source": [
    "TGFB_UP <- test_df %>% \n",
    "            dplyr::filter(PATHWAY == \"TGFB_UP.V1_UP\")\n",
    "TGFB_DN <- test_df %>% \n",
    "            dplyr::filter(PATHWAY == \"TGFB_UP.V1_DN\")\n",
    "TGFB_UP\n",
    "TGFB_DN"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0d94ff74",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "TGFB_UP_genes <- myC6[[167]]@geneIds\n",
    "TGFB_DN_genes <- myC6[[67]]@geneIds\n",
    "str(TGFB_UP_genes)\n",
    "str(TGFB_DN_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "351c5cac",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "TGFB_DN_guides <- CRISPR_guides %>% \n",
    "        filter(Target.Gene.ID %in% TGFB_DN_genes)\n",
    "TGFB_DN_guides\n",
    "TGFB_DN_targets <- TGFB_DN_guides %>% \n",
    "                pull(Target.Gene.Symbol) %>%\n",
    "                unique() %>%\n",
    "                sort()\n",
    "TGFB_DN_targets\n",
    "TGFB_UP_guides <- CRISPR_guides %>% \n",
    "        filter(Target.Gene.ID %in% TGFB_UP_genes)\n",
    "TGFB_UP_guides\n",
    "TGFB_UP_targets <- TGFB_UP_guides %>% \n",
    "                pull(Target.Gene.Symbol) %>%\n",
    "                unique() %>%\n",
    "                sort()\n",
    "TGFB_UP_targets"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "acd95c99",
   "metadata": {},
   "source": [
    "### Heatmap Cleanup"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "10d55adc",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "expression"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "540e064a",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "colMeta_df <- data.frame(Genotype = c(rep(\"E545K\", 6), rep(\"H1047R\", 3), rep(\"E545K\", 3), rep(\"H1047R\", 6)), \n",
    "                      Lineage = c(rep(\"MCF10A\", 3), rep(\"MCF7\", 3), rep(\"MCF10A\", 3),\n",
    "                                 rep(\"HTert-IMEC\", 6), rep(\"T47D\", 3)),\n",
    "                      stringsAsFactors = F, \n",
    "                      row.names = colnames(expression))\n",
    "colMeta_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5c089ad6",
   "metadata": {},
   "outputs": [],
   "source": [
    "p <- pheatmap(expression,\n",
    "              clustering_distance_cols = \"euclidean\", \n",
    "              clustering_distance_rows = \"euclidean\", \n",
    "              annotation_col = colMeta_df,\n",
    "              fontsize_number = 2, \n",
    "              scale = \"row\", \n",
    "              cutree_cols = 2,\n",
    "              show_colnames = FALSE\n",
    "             )\n",
    "p"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "678e4436",
   "metadata": {
    "code_folding": [
     2
    ]
   },
   "outputs": [],
   "source": [
    "add.flag <- function(pheatmap,\n",
    "                     kept.labels,\n",
    "                     repel.degree) {\n",
    "\n",
    "  # repel.degree = number within [0, 1], which controls how much \n",
    "  #                space to allocate for repelling labels.\n",
    "  ## repel.degree = 0: spread out labels over existing range of kept labels\n",
    "  ## repel.degree = 1: spread out labels over the full y-axis\n",
    "\n",
    "  heatmap <- pheatmap$gtable\n",
    "\n",
    "  new.label <- heatmap$grobs[[which(heatmap$layout$name == \"row_names\")]] \n",
    "\n",
    "  # keep only labels in kept.labels, replace the rest with \"\"\n",
    "  new.label$label <- ifelse(new.label$label %in% kept.labels, \n",
    "                            new.label$label, \"\")\n",
    "\n",
    "  # calculate evenly spaced out y-axis positions\n",
    "  repelled.y <- function(d, d.select, k = repel.degree){\n",
    "    # d = vector of distances for labels\n",
    "    # d.select = vector of T/F for which labels are significant\n",
    "\n",
    "    # recursive function to get current label positions\n",
    "    # (note the unit is \"npc\" for all components of each distance)\n",
    "    strip.npc <- function(dd){\n",
    "      if(!\"unit.arithmetic\" %in% class(dd)) {\n",
    "        return(as.numeric(dd))\n",
    "      }\n",
    "\n",
    "      d1 <- strip.npc(dd$arg1)\n",
    "      d2 <- strip.npc(dd$arg2)\n",
    "      fn <- dd$fname\n",
    "      return(lazyeval::lazy_eval(paste(d1, fn, d2)))\n",
    "    }\n",
    "\n",
    "    full.range <- sapply(seq_along(d), function(i) strip.npc(d[i]))\n",
    "    selected.range <- sapply(seq_along(d[d.select]), function(i) strip.npc(d[d.select][i]))\n",
    "\n",
    "    return(unit(seq(from = max(selected.range) + k*(max(full.range) - max(selected.range)),\n",
    "                    to = min(selected.range) - k*(min(selected.range) - min(full.range)), \n",
    "                    length.out = sum(d.select)), \n",
    "                \"npc\"))\n",
    "  }\n",
    "  new.y.positions <- repelled.y(new.label$y,\n",
    "                                d.select = new.label$label != \"\")\n",
    "  new.flag <- segmentsGrob(x0 = new.label$x,\n",
    "                           x1 = new.label$x + unit(0.15, \"npc\"),\n",
    "                           y0 = new.label$y[new.label$label != \"\"],\n",
    "                           y1 = new.y.positions)\n",
    "\n",
    "  # shift position for selected labels\n",
    "  new.label$x <- new.label$x + unit(0.2, \"npc\")\n",
    "  new.label$y[new.label$label != \"\"] <- new.y.positions\n",
    "\n",
    "  # add flag to heatmap\n",
    "  heatmap <- gtable::gtable_add_grob(x = heatmap,\n",
    "                                   grobs = new.flag,\n",
    "                                   t = 4, \n",
    "                                   l = 4\n",
    "  )\n",
    "\n",
    "  # replace label positions in heatmap\n",
    "  heatmap$grobs[[which(heatmap$layout$name == \"row_names\")]] <- new.label\n",
    "\n",
    "  # plot result\n",
    "  grid.newpage()\n",
    "  grid.draw(heatmap)\n",
    "\n",
    "  # return a copy of the heatmap invisibly\n",
    "  invisible(heatmap)\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8a7df273",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "sort(rownames(expression))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0e12a046",
   "metadata": {},
   "outputs": [],
   "source": [
    "sets_of_int <- c(\"AKT_UP_MTOR_DN.V1_DN\", \"AKT_UP_MTOR_DN.V1_UP\",\n",
    "                \"AKT_UP.V1_DN\", \"AKT_UP.V1_UP\", \n",
    "                \"ATF2_S_UP.V1_UP\", \"ATF2_UP.V1_DN\", \"ATF2_UP.V1_UP\", \n",
    "                \"JNK_DN.V1_DN\", \"JNK_DN.V1_UP\", \n",
    "                \"PTEN_DN.V1_DN\", \"PTEN_DN.V1_UP\", \"PTEN_DN.V2_DN\", \"PTEN_DN.V2_UP\", \n",
    "                \"MTOR_UP.V1_DN\", \"MTOR_UP.V1_UP\", \n",
    "                \"MYC_UP.V1_DN\", \"MYC_UP.V1_UP\", \n",
    "                \"TGFB_UP.V1_DN\", \"TGFB_UP.V1_UP\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bc4ef29b",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "p_improved <- add.flag(p, sets_of_int, 0)\n",
    "p_improved"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f6eacb44",
   "metadata": {},
   "outputs": [],
   "source": [
    "ggsave(filename = \"GSVA_heatmap_clean.pdf\", plot = p_improved)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8e3c4725",
   "metadata": {},
   "source": [
    "## Diff Seq Analyses"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "572eba3b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'Intercept'</li><li>'SV1'</li><li>'SV2'</li><li>'SV3'</li><li>'genotype_ex9_vs_ex20'</li><li>'genotype_WT_vs_ex20'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'Intercept'\n",
       "\\item 'SV1'\n",
       "\\item 'SV2'\n",
       "\\item 'SV3'\n",
       "\\item 'genotype\\_ex9\\_vs\\_ex20'\n",
       "\\item 'genotype\\_WT\\_vs\\_ex20'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'Intercept'\n",
       "2. 'SV1'\n",
       "3. 'SV2'\n",
       "4. 'SV3'\n",
       "5. 'genotype_ex9_vs_ex20'\n",
       "6. 'genotype_WT_vs_ex20'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"Intercept\"            \"SV1\"                  \"SV2\"                 \n",
       "[4] \"SV3\"                  \"genotype_ex9_vs_ex20\" \"genotype_WT_vs_ex20\" "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "resultsNames(ddssva)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "befda8c8",
   "metadata": {},
   "source": [
    "### Genotype Direct Comparison"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "ad42edd4",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 4374  6</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>baseMean</th><th scope=col>log2FoldChange</th><th scope=col>lfcSE</th><th scope=col>stat</th><th scope=col>pvalue</th><th scope=col>padj</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>ENSG00000278791.1</th><td>    6.612653</td><td>-2.302249</td><td>0.4649189</td><td>-4.951937</td><td>7.347834e-07</td><td>7.714498e-06</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000229376.3</th><td>   12.801501</td><td>-1.756743</td><td>0.4287367</td><td>-4.097487</td><td>4.176587e-05</td><td>2.741731e-04</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000142609.18</th><td>   13.785815</td><td> 1.869731</td><td>0.3283950</td><td> 5.693544</td><td>1.244291e-08</td><td>1.983657e-07</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000187730.9</th><td>   29.824951</td><td> 2.674128</td><td>0.5126600</td><td> 5.216183</td><td>1.826478e-07</td><td>2.238148e-06</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000238164.6</th><td>   40.461046</td><td>-1.773661</td><td>0.4521381</td><td>-3.922831</td><td>8.751443e-05</td><td>5.237764e-04</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000157873.17</th><td>  232.385304</td><td>-1.795708</td><td>0.3865943</td><td>-4.644942</td><td>3.401715e-06</td><td>2.984972e-05</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000177133.11</th><td>   10.966659</td><td> 4.003949</td><td>0.8200935</td><td> 4.882308</td><td>1.048511e-06</td><td>1.060372e-05</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000142611.17</th><td>   68.509740</td><td> 4.330422</td><td>0.6370000</td><td> 6.798151</td><td>1.059707e-11</td><td>3.358566e-10</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000162591.16</th><td> 8307.720221</td><td> 1.723491</td><td>0.3207743</td><td> 5.372910</td><td>7.747601e-08</td><td>1.024647e-06</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000238290.1</th><td>   13.087750</td><td>-1.720946</td><td>0.5031363</td><td>-3.420436</td><td>6.252075e-04</td><td>2.887785e-03</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000131686.15</th><td>    3.342750</td><td> 2.932551</td><td>0.7919433</td><td> 3.702981</td><td>2.130811e-04</td><td>1.138600e-03</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000228526.7</th><td>  564.749075</td><td>-1.979453</td><td>0.3774731</td><td>-5.243958</td><td>1.571678e-07</td><td>1.957046e-06</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000234546.4</th><td>  243.063482</td><td>-2.144048</td><td>0.4451617</td><td>-4.816335</td><td>1.462193e-06</td><td>1.418474e-05</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000224340.1</th><td>    5.738009</td><td>-1.634055</td><td>0.5758284</td><td>-2.837747</td><td>4.543319e-03</td><td>1.570555e-02</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000199562.1</th><td>    3.426536</td><td>-1.986978</td><td>0.6490028</td><td>-3.061586</td><td>2.201677e-03</td><td>8.484277e-03</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000175262.14</th><td>    4.162315</td><td> 1.909322</td><td>0.6284430</td><td> 3.038179</td><td>2.380128e-03</td><td>9.067296e-03</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000235643.2</th><td>    2.064915</td><td> 3.126468</td><td>1.2402793</td><td> 2.520777</td><td>1.170960e-02</td><td>3.480761e-02</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000276470.1</th><td>   11.000890</td><td>-1.833650</td><td>0.4553776</td><td>-4.026657</td><td>5.657540e-05</td><td>3.585591e-04</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000224904.1</th><td>    1.592345</td><td> 3.174139</td><td>1.1585486</td><td> 2.739755</td><td>6.148504e-03</td><td>2.029518e-02</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000120949.15</th><td>    3.947415</td><td> 1.887533</td><td>0.6579427</td><td> 2.868841</td><td>4.119791e-03</td><td>1.445611e-02</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000162496.9</th><td>11434.244751</td><td> 2.824843</td><td>0.4017785</td><td> 7.030848</td><td>2.052821e-12</td><td>7.610343e-11</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000276830.1</th><td>   43.966641</td><td> 1.942730</td><td>0.5121412</td><td> 3.793349</td><td>1.486291e-04</td><td>8.353903e-04</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000272482.1</th><td>  143.835884</td><td> 3.415021</td><td>0.5378340</td><td> 6.349582</td><td>2.159009e-10</td><td>5.189214e-09</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000204518.2</th><td>    2.232951</td><td> 4.536725</td><td>1.8693370</td><td> 2.426917</td><td>1.522776e-02</td><td>4.332557e-02</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000188984.12</th><td>    1.468852</td><td> 3.029707</td><td>1.2788664</td><td> 2.369057</td><td>1.783353e-02</td><td>4.952245e-02</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000116726.4</th><td>    1.454513</td><td> 3.169252</td><td>1.0848809</td><td> 2.921290</td><td>3.485850e-03</td><td>1.253581e-02</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000116721.9</th><td>    1.807958</td><td> 2.697376</td><td>1.1147098</td><td> 2.419801</td><td>1.552900e-02</td><td>4.402101e-02</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000204481.8</th><td>    1.509265</td><td> 3.547712</td><td>1.2913436</td><td> 2.747303</td><td>6.008764e-03</td><td>1.989338e-02</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000162494.6</th><td>    7.509142</td><td> 3.410109</td><td>1.0467941</td><td> 3.257670</td><td>1.123311e-03</td><td>4.788098e-03</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000233485.1</th><td>    4.901759</td><td> 2.376155</td><td>0.9123343</td><td> 2.604479</td><td>9.201411e-03</td><td>2.845367e-02</td></tr>\n",
       "\t<tr><th scope=row></th><td></td><td></td><td></td><td></td><td></td><td></td></tr>\n",
       "\t<tr><th scope=row>ENSG00000170965.10</th><td>   50.049865</td><td> 1.870657</td><td>0.2107735</td><td> 8.875201</td><td>6.980729e-19</td><td>7.825434e-17</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000181433.9</th><td>    1.490064</td><td> 3.543457</td><td>1.0233818</td><td> 3.462498</td><td>5.351859e-04</td><td>2.520895e-03</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000022267.19</th><td> 8862.367219</td><td> 5.307562</td><td>0.7194787</td><td> 7.376955</td><td>1.619511e-13</td><td>7.459257e-12</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000129680.16</th><td> 3226.925586</td><td> 3.086494</td><td>0.3935205</td><td> 7.843286</td><td>4.389063e-15</td><td>2.668323e-13</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000156920.11</th><td>    8.299286</td><td> 1.858064</td><td>0.6503765</td><td> 2.856905</td><td>4.277933e-03</td><td>1.492267e-02</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000156925.12</th><td>   16.933753</td><td> 2.849854</td><td>0.6358960</td><td> 4.481635</td><td>7.407320e-06</td><td>5.967258e-05</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000101977.21</th><td>  667.205436</td><td>-2.171844</td><td>0.6405926</td><td>-3.390367</td><td>6.979912e-04</td><td>3.184054e-03</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000230707.1</th><td>    9.118160</td><td> 2.417701</td><td>0.8632671</td><td> 2.800641</td><td>5.100124e-03</td><td>1.732644e-02</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000185985.10</th><td>    9.849136</td><td> 6.203665</td><td>0.7712065</td><td> 8.044104</td><td>8.687899e-16</td><td>6.095050e-14</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000155966.14</th><td>  294.549288</td><td> 1.696034</td><td>0.5106201</td><td> 3.321519</td><td>8.952893e-04</td><td>3.944784e-03</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000147381.11</th><td>    1.345848</td><td> 4.422309</td><td>1.6587917</td><td> 2.665982</td><td>7.676379e-03</td><td>2.447546e-02</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000102287.19</th><td>  625.538353</td><td>-1.870973</td><td>0.3926244</td><td>-4.765299</td><td>1.885741e-06</td><td>1.766043e-05</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000011677.13</th><td>   58.713057</td><td>-3.059954</td><td>0.4504043</td><td>-6.793793</td><td>1.092232e-11</td><td>3.448751e-10</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000268089.3</th><td>   52.199893</td><td>-2.370338</td><td>0.7735584</td><td>-3.064200</td><td>2.182530e-03</td><td>8.425886e-03</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000221867.9</th><td>   83.448408</td><td>-6.231697</td><td>2.1609190</td><td>-2.883818</td><td>3.928859e-03</td><td>1.388850e-02</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000198883.12</th><td>    4.107412</td><td> 1.936476</td><td>0.7373340</td><td> 2.626321</td><td>8.631349e-03</td><td>2.700792e-02</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000235961.5</th><td>   27.783377</td><td> 3.445541</td><td>0.8527000</td><td> 4.040742</td><td>5.328237e-05</td><td>3.398747e-04</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000189420.8</th><td>    9.553255</td><td> 1.790793</td><td>0.6911955</td><td> 2.590864</td><td>9.573538e-03</td><td>2.941123e-02</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000182492.16</th><td> 3831.186718</td><td> 2.708571</td><td>0.9559595</td><td> 2.833353</td><td>4.606248e-03</td><td>1.588550e-02</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000067842.17</th><td>    7.671835</td><td> 2.253908</td><td>0.6990900</td><td> 3.224060</td><td>1.263867e-03</td><td>5.301962e-03</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000130829.18</th><td>  883.727537</td><td> 3.014168</td><td>0.4570148</td><td> 6.595340</td><td>4.242830e-11</td><td>1.182802e-09</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000007350.17</th><td>    2.189215</td><td> 3.841916</td><td>1.1526952</td><td> 3.332985</td><td>8.591965e-04</td><td>3.806733e-03</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000285018.1</th><td>  118.616157</td><td> 1.807665</td><td>0.4912794</td><td> 3.679505</td><td>2.336869e-04</td><td>1.235310e-03</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000160211.19</th><td>31539.118406</td><td> 1.729240</td><td>0.1657735</td><td>10.431344</td><td>1.783474e-25</td><td>5.211925e-23</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000160219.12</th><td>  189.036114</td><td> 1.988261</td><td>0.2614332</td><td> 7.605235</td><td>2.843871e-14</td><td>1.517829e-12</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000155961.5</th><td>   54.860647</td><td> 2.253466</td><td>0.8531395</td><td> 2.641380</td><td>8.256902e-03</td><td>2.603259e-02</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000286217.2</th><td>    1.416493</td><td> 3.678475</td><td>1.2151377</td><td> 3.027208</td><td>2.468237e-03</td><td>9.346657e-03</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000114374.13</th><td>    2.291132</td><td> 3.784364</td><td>1.2238852</td><td> 3.092091</td><td>1.987521e-03</td><td>7.780784e-03</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000012817.15</th><td>    2.051723</td><td> 3.321876</td><td>1.1093026</td><td> 2.994562</td><td>2.748391e-03</td><td>1.024097e-02</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000211459.2</th><td> 4123.886138</td><td>-1.530304</td><td>0.3409141</td><td>-4.488825</td><td>7.161696e-06</td><td>5.784783e-05</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 4374  6\n",
       "\\begin{tabular}{r|llllll}\n",
       "  & baseMean & log2FoldChange & lfcSE & stat & pvalue & padj\\\\\n",
       "  & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tENSG00000278791.1 &     6.612653 & -2.302249 & 0.4649189 & -4.951937 & 7.347834e-07 & 7.714498e-06\\\\\n",
       "\tENSG00000229376.3 &    12.801501 & -1.756743 & 0.4287367 & -4.097487 & 4.176587e-05 & 2.741731e-04\\\\\n",
       "\tENSG00000142609.18 &    13.785815 &  1.869731 & 0.3283950 &  5.693544 & 1.244291e-08 & 1.983657e-07\\\\\n",
       "\tENSG00000187730.9 &    29.824951 &  2.674128 & 0.5126600 &  5.216183 & 1.826478e-07 & 2.238148e-06\\\\\n",
       "\tENSG00000238164.6 &    40.461046 & -1.773661 & 0.4521381 & -3.922831 & 8.751443e-05 & 5.237764e-04\\\\\n",
       "\tENSG00000157873.17 &   232.385304 & -1.795708 & 0.3865943 & -4.644942 & 3.401715e-06 & 2.984972e-05\\\\\n",
       "\tENSG00000177133.11 &    10.966659 &  4.003949 & 0.8200935 &  4.882308 & 1.048511e-06 & 1.060372e-05\\\\\n",
       "\tENSG00000142611.17 &    68.509740 &  4.330422 & 0.6370000 &  6.798151 & 1.059707e-11 & 3.358566e-10\\\\\n",
       "\tENSG00000162591.16 &  8307.720221 &  1.723491 & 0.3207743 &  5.372910 & 7.747601e-08 & 1.024647e-06\\\\\n",
       "\tENSG00000238290.1 &    13.087750 & -1.720946 & 0.5031363 & -3.420436 & 6.252075e-04 & 2.887785e-03\\\\\n",
       "\tENSG00000131686.15 &     3.342750 &  2.932551 & 0.7919433 &  3.702981 & 2.130811e-04 & 1.138600e-03\\\\\n",
       "\tENSG00000228526.7 &   564.749075 & -1.979453 & 0.3774731 & -5.243958 & 1.571678e-07 & 1.957046e-06\\\\\n",
       "\tENSG00000234546.4 &   243.063482 & -2.144048 & 0.4451617 & -4.816335 & 1.462193e-06 & 1.418474e-05\\\\\n",
       "\tENSG00000224340.1 &     5.738009 & -1.634055 & 0.5758284 & -2.837747 & 4.543319e-03 & 1.570555e-02\\\\\n",
       "\tENSG00000199562.1 &     3.426536 & -1.986978 & 0.6490028 & -3.061586 & 2.201677e-03 & 8.484277e-03\\\\\n",
       "\tENSG00000175262.14 &     4.162315 &  1.909322 & 0.6284430 &  3.038179 & 2.380128e-03 & 9.067296e-03\\\\\n",
       "\tENSG00000235643.2 &     2.064915 &  3.126468 & 1.2402793 &  2.520777 & 1.170960e-02 & 3.480761e-02\\\\\n",
       "\tENSG00000276470.1 &    11.000890 & -1.833650 & 0.4553776 & -4.026657 & 5.657540e-05 & 3.585591e-04\\\\\n",
       "\tENSG00000224904.1 &     1.592345 &  3.174139 & 1.1585486 &  2.739755 & 6.148504e-03 & 2.029518e-02\\\\\n",
       "\tENSG00000120949.15 &     3.947415 &  1.887533 & 0.6579427 &  2.868841 & 4.119791e-03 & 1.445611e-02\\\\\n",
       "\tENSG00000162496.9 & 11434.244751 &  2.824843 & 0.4017785 &  7.030848 & 2.052821e-12 & 7.610343e-11\\\\\n",
       "\tENSG00000276830.1 &    43.966641 &  1.942730 & 0.5121412 &  3.793349 & 1.486291e-04 & 8.353903e-04\\\\\n",
       "\tENSG00000272482.1 &   143.835884 &  3.415021 & 0.5378340 &  6.349582 & 2.159009e-10 & 5.189214e-09\\\\\n",
       "\tENSG00000204518.2 &     2.232951 &  4.536725 & 1.8693370 &  2.426917 & 1.522776e-02 & 4.332557e-02\\\\\n",
       "\tENSG00000188984.12 &     1.468852 &  3.029707 & 1.2788664 &  2.369057 & 1.783353e-02 & 4.952245e-02\\\\\n",
       "\tENSG00000116726.4 &     1.454513 &  3.169252 & 1.0848809 &  2.921290 & 3.485850e-03 & 1.253581e-02\\\\\n",
       "\tENSG00000116721.9 &     1.807958 &  2.697376 & 1.1147098 &  2.419801 & 1.552900e-02 & 4.402101e-02\\\\\n",
       "\tENSG00000204481.8 &     1.509265 &  3.547712 & 1.2913436 &  2.747303 & 6.008764e-03 & 1.989338e-02\\\\\n",
       "\tENSG00000162494.6 &     7.509142 &  3.410109 & 1.0467941 &  3.257670 & 1.123311e-03 & 4.788098e-03\\\\\n",
       "\tENSG00000233485.1 &     4.901759 &  2.376155 & 0.9123343 &  2.604479 & 9.201411e-03 & 2.845367e-02\\\\\n",
       "\t &  &  &  &  &  & \\\\\n",
       "\tENSG00000170965.10 &    50.049865 &  1.870657 & 0.2107735 &  8.875201 & 6.980729e-19 & 7.825434e-17\\\\\n",
       "\tENSG00000181433.9 &     1.490064 &  3.543457 & 1.0233818 &  3.462498 & 5.351859e-04 & 2.520895e-03\\\\\n",
       "\tENSG00000022267.19 &  8862.367219 &  5.307562 & 0.7194787 &  7.376955 & 1.619511e-13 & 7.459257e-12\\\\\n",
       "\tENSG00000129680.16 &  3226.925586 &  3.086494 & 0.3935205 &  7.843286 & 4.389063e-15 & 2.668323e-13\\\\\n",
       "\tENSG00000156920.11 &     8.299286 &  1.858064 & 0.6503765 &  2.856905 & 4.277933e-03 & 1.492267e-02\\\\\n",
       "\tENSG00000156925.12 &    16.933753 &  2.849854 & 0.6358960 &  4.481635 & 7.407320e-06 & 5.967258e-05\\\\\n",
       "\tENSG00000101977.21 &   667.205436 & -2.171844 & 0.6405926 & -3.390367 & 6.979912e-04 & 3.184054e-03\\\\\n",
       "\tENSG00000230707.1 &     9.118160 &  2.417701 & 0.8632671 &  2.800641 & 5.100124e-03 & 1.732644e-02\\\\\n",
       "\tENSG00000185985.10 &     9.849136 &  6.203665 & 0.7712065 &  8.044104 & 8.687899e-16 & 6.095050e-14\\\\\n",
       "\tENSG00000155966.14 &   294.549288 &  1.696034 & 0.5106201 &  3.321519 & 8.952893e-04 & 3.944784e-03\\\\\n",
       "\tENSG00000147381.11 &     1.345848 &  4.422309 & 1.6587917 &  2.665982 & 7.676379e-03 & 2.447546e-02\\\\\n",
       "\tENSG00000102287.19 &   625.538353 & -1.870973 & 0.3926244 & -4.765299 & 1.885741e-06 & 1.766043e-05\\\\\n",
       "\tENSG00000011677.13 &    58.713057 & -3.059954 & 0.4504043 & -6.793793 & 1.092232e-11 & 3.448751e-10\\\\\n",
       "\tENSG00000268089.3 &    52.199893 & -2.370338 & 0.7735584 & -3.064200 & 2.182530e-03 & 8.425886e-03\\\\\n",
       "\tENSG00000221867.9 &    83.448408 & -6.231697 & 2.1609190 & -2.883818 & 3.928859e-03 & 1.388850e-02\\\\\n",
       "\tENSG00000198883.12 &     4.107412 &  1.936476 & 0.7373340 &  2.626321 & 8.631349e-03 & 2.700792e-02\\\\\n",
       "\tENSG00000235961.5 &    27.783377 &  3.445541 & 0.8527000 &  4.040742 & 5.328237e-05 & 3.398747e-04\\\\\n",
       "\tENSG00000189420.8 &     9.553255 &  1.790793 & 0.6911955 &  2.590864 & 9.573538e-03 & 2.941123e-02\\\\\n",
       "\tENSG00000182492.16 &  3831.186718 &  2.708571 & 0.9559595 &  2.833353 & 4.606248e-03 & 1.588550e-02\\\\\n",
       "\tENSG00000067842.17 &     7.671835 &  2.253908 & 0.6990900 &  3.224060 & 1.263867e-03 & 5.301962e-03\\\\\n",
       "\tENSG00000130829.18 &   883.727537 &  3.014168 & 0.4570148 &  6.595340 & 4.242830e-11 & 1.182802e-09\\\\\n",
       "\tENSG00000007350.17 &     2.189215 &  3.841916 & 1.1526952 &  3.332985 & 8.591965e-04 & 3.806733e-03\\\\\n",
       "\tENSG00000285018.1 &   118.616157 &  1.807665 & 0.4912794 &  3.679505 & 2.336869e-04 & 1.235310e-03\\\\\n",
       "\tENSG00000160211.19 & 31539.118406 &  1.729240 & 0.1657735 & 10.431344 & 1.783474e-25 & 5.211925e-23\\\\\n",
       "\tENSG00000160219.12 &   189.036114 &  1.988261 & 0.2614332 &  7.605235 & 2.843871e-14 & 1.517829e-12\\\\\n",
       "\tENSG00000155961.5 &    54.860647 &  2.253466 & 0.8531395 &  2.641380 & 8.256902e-03 & 2.603259e-02\\\\\n",
       "\tENSG00000286217.2 &     1.416493 &  3.678475 & 1.2151377 &  3.027208 & 2.468237e-03 & 9.346657e-03\\\\\n",
       "\tENSG00000114374.13 &     2.291132 &  3.784364 & 1.2238852 &  3.092091 & 1.987521e-03 & 7.780784e-03\\\\\n",
       "\tENSG00000012817.15 &     2.051723 &  3.321876 & 1.1093026 &  2.994562 & 2.748391e-03 & 1.024097e-02\\\\\n",
       "\tENSG00000211459.2 &  4123.886138 & -1.530304 & 0.3409141 & -4.488825 & 7.161696e-06 & 5.784783e-05\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 4374  6\n",
       "\n",
       "| <!--/--> | baseMean &lt;dbl&gt; | log2FoldChange &lt;dbl&gt; | lfcSE &lt;dbl&gt; | stat &lt;dbl&gt; | pvalue &lt;dbl&gt; | padj &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "| ENSG00000278791.1 |     6.612653 | -2.302249 | 0.4649189 | -4.951937 | 7.347834e-07 | 7.714498e-06 |\n",
       "| ENSG00000229376.3 |    12.801501 | -1.756743 | 0.4287367 | -4.097487 | 4.176587e-05 | 2.741731e-04 |\n",
       "| ENSG00000142609.18 |    13.785815 |  1.869731 | 0.3283950 |  5.693544 | 1.244291e-08 | 1.983657e-07 |\n",
       "| ENSG00000187730.9 |    29.824951 |  2.674128 | 0.5126600 |  5.216183 | 1.826478e-07 | 2.238148e-06 |\n",
       "| ENSG00000238164.6 |    40.461046 | -1.773661 | 0.4521381 | -3.922831 | 8.751443e-05 | 5.237764e-04 |\n",
       "| ENSG00000157873.17 |   232.385304 | -1.795708 | 0.3865943 | -4.644942 | 3.401715e-06 | 2.984972e-05 |\n",
       "| ENSG00000177133.11 |    10.966659 |  4.003949 | 0.8200935 |  4.882308 | 1.048511e-06 | 1.060372e-05 |\n",
       "| ENSG00000142611.17 |    68.509740 |  4.330422 | 0.6370000 |  6.798151 | 1.059707e-11 | 3.358566e-10 |\n",
       "| ENSG00000162591.16 |  8307.720221 |  1.723491 | 0.3207743 |  5.372910 | 7.747601e-08 | 1.024647e-06 |\n",
       "| ENSG00000238290.1 |    13.087750 | -1.720946 | 0.5031363 | -3.420436 | 6.252075e-04 | 2.887785e-03 |\n",
       "| ENSG00000131686.15 |     3.342750 |  2.932551 | 0.7919433 |  3.702981 | 2.130811e-04 | 1.138600e-03 |\n",
       "| ENSG00000228526.7 |   564.749075 | -1.979453 | 0.3774731 | -5.243958 | 1.571678e-07 | 1.957046e-06 |\n",
       "| ENSG00000234546.4 |   243.063482 | -2.144048 | 0.4451617 | -4.816335 | 1.462193e-06 | 1.418474e-05 |\n",
       "| ENSG00000224340.1 |     5.738009 | -1.634055 | 0.5758284 | -2.837747 | 4.543319e-03 | 1.570555e-02 |\n",
       "| ENSG00000199562.1 |     3.426536 | -1.986978 | 0.6490028 | -3.061586 | 2.201677e-03 | 8.484277e-03 |\n",
       "| ENSG00000175262.14 |     4.162315 |  1.909322 | 0.6284430 |  3.038179 | 2.380128e-03 | 9.067296e-03 |\n",
       "| ENSG00000235643.2 |     2.064915 |  3.126468 | 1.2402793 |  2.520777 | 1.170960e-02 | 3.480761e-02 |\n",
       "| ENSG00000276470.1 |    11.000890 | -1.833650 | 0.4553776 | -4.026657 | 5.657540e-05 | 3.585591e-04 |\n",
       "| ENSG00000224904.1 |     1.592345 |  3.174139 | 1.1585486 |  2.739755 | 6.148504e-03 | 2.029518e-02 |\n",
       "| ENSG00000120949.15 |     3.947415 |  1.887533 | 0.6579427 |  2.868841 | 4.119791e-03 | 1.445611e-02 |\n",
       "| ENSG00000162496.9 | 11434.244751 |  2.824843 | 0.4017785 |  7.030848 | 2.052821e-12 | 7.610343e-11 |\n",
       "| ENSG00000276830.1 |    43.966641 |  1.942730 | 0.5121412 |  3.793349 | 1.486291e-04 | 8.353903e-04 |\n",
       "| ENSG00000272482.1 |   143.835884 |  3.415021 | 0.5378340 |  6.349582 | 2.159009e-10 | 5.189214e-09 |\n",
       "| ENSG00000204518.2 |     2.232951 |  4.536725 | 1.8693370 |  2.426917 | 1.522776e-02 | 4.332557e-02 |\n",
       "| ENSG00000188984.12 |     1.468852 |  3.029707 | 1.2788664 |  2.369057 | 1.783353e-02 | 4.952245e-02 |\n",
       "| ENSG00000116726.4 |     1.454513 |  3.169252 | 1.0848809 |  2.921290 | 3.485850e-03 | 1.253581e-02 |\n",
       "| ENSG00000116721.9 |     1.807958 |  2.697376 | 1.1147098 |  2.419801 | 1.552900e-02 | 4.402101e-02 |\n",
       "| ENSG00000204481.8 |     1.509265 |  3.547712 | 1.2913436 |  2.747303 | 6.008764e-03 | 1.989338e-02 |\n",
       "| ENSG00000162494.6 |     7.509142 |  3.410109 | 1.0467941 |  3.257670 | 1.123311e-03 | 4.788098e-03 |\n",
       "| ENSG00000233485.1 |     4.901759 |  2.376155 | 0.9123343 |  2.604479 | 9.201411e-03 | 2.845367e-02 |\n",
       "|  |  |  |  |  |  |  |\n",
       "| ENSG00000170965.10 |    50.049865 |  1.870657 | 0.2107735 |  8.875201 | 6.980729e-19 | 7.825434e-17 |\n",
       "| ENSG00000181433.9 |     1.490064 |  3.543457 | 1.0233818 |  3.462498 | 5.351859e-04 | 2.520895e-03 |\n",
       "| ENSG00000022267.19 |  8862.367219 |  5.307562 | 0.7194787 |  7.376955 | 1.619511e-13 | 7.459257e-12 |\n",
       "| ENSG00000129680.16 |  3226.925586 |  3.086494 | 0.3935205 |  7.843286 | 4.389063e-15 | 2.668323e-13 |\n",
       "| ENSG00000156920.11 |     8.299286 |  1.858064 | 0.6503765 |  2.856905 | 4.277933e-03 | 1.492267e-02 |\n",
       "| ENSG00000156925.12 |    16.933753 |  2.849854 | 0.6358960 |  4.481635 | 7.407320e-06 | 5.967258e-05 |\n",
       "| ENSG00000101977.21 |   667.205436 | -2.171844 | 0.6405926 | -3.390367 | 6.979912e-04 | 3.184054e-03 |\n",
       "| ENSG00000230707.1 |     9.118160 |  2.417701 | 0.8632671 |  2.800641 | 5.100124e-03 | 1.732644e-02 |\n",
       "| ENSG00000185985.10 |     9.849136 |  6.203665 | 0.7712065 |  8.044104 | 8.687899e-16 | 6.095050e-14 |\n",
       "| ENSG00000155966.14 |   294.549288 |  1.696034 | 0.5106201 |  3.321519 | 8.952893e-04 | 3.944784e-03 |\n",
       "| ENSG00000147381.11 |     1.345848 |  4.422309 | 1.6587917 |  2.665982 | 7.676379e-03 | 2.447546e-02 |\n",
       "| ENSG00000102287.19 |   625.538353 | -1.870973 | 0.3926244 | -4.765299 | 1.885741e-06 | 1.766043e-05 |\n",
       "| ENSG00000011677.13 |    58.713057 | -3.059954 | 0.4504043 | -6.793793 | 1.092232e-11 | 3.448751e-10 |\n",
       "| ENSG00000268089.3 |    52.199893 | -2.370338 | 0.7735584 | -3.064200 | 2.182530e-03 | 8.425886e-03 |\n",
       "| ENSG00000221867.9 |    83.448408 | -6.231697 | 2.1609190 | -2.883818 | 3.928859e-03 | 1.388850e-02 |\n",
       "| ENSG00000198883.12 |     4.107412 |  1.936476 | 0.7373340 |  2.626321 | 8.631349e-03 | 2.700792e-02 |\n",
       "| ENSG00000235961.5 |    27.783377 |  3.445541 | 0.8527000 |  4.040742 | 5.328237e-05 | 3.398747e-04 |\n",
       "| ENSG00000189420.8 |     9.553255 |  1.790793 | 0.6911955 |  2.590864 | 9.573538e-03 | 2.941123e-02 |\n",
       "| ENSG00000182492.16 |  3831.186718 |  2.708571 | 0.9559595 |  2.833353 | 4.606248e-03 | 1.588550e-02 |\n",
       "| ENSG00000067842.17 |     7.671835 |  2.253908 | 0.6990900 |  3.224060 | 1.263867e-03 | 5.301962e-03 |\n",
       "| ENSG00000130829.18 |   883.727537 |  3.014168 | 0.4570148 |  6.595340 | 4.242830e-11 | 1.182802e-09 |\n",
       "| ENSG00000007350.17 |     2.189215 |  3.841916 | 1.1526952 |  3.332985 | 8.591965e-04 | 3.806733e-03 |\n",
       "| ENSG00000285018.1 |   118.616157 |  1.807665 | 0.4912794 |  3.679505 | 2.336869e-04 | 1.235310e-03 |\n",
       "| ENSG00000160211.19 | 31539.118406 |  1.729240 | 0.1657735 | 10.431344 | 1.783474e-25 | 5.211925e-23 |\n",
       "| ENSG00000160219.12 |   189.036114 |  1.988261 | 0.2614332 |  7.605235 | 2.843871e-14 | 1.517829e-12 |\n",
       "| ENSG00000155961.5 |    54.860647 |  2.253466 | 0.8531395 |  2.641380 | 8.256902e-03 | 2.603259e-02 |\n",
       "| ENSG00000286217.2 |     1.416493 |  3.678475 | 1.2151377 |  3.027208 | 2.468237e-03 | 9.346657e-03 |\n",
       "| ENSG00000114374.13 |     2.291132 |  3.784364 | 1.2238852 |  3.092091 | 1.987521e-03 | 7.780784e-03 |\n",
       "| ENSG00000012817.15 |     2.051723 |  3.321876 | 1.1093026 |  2.994562 | 2.748391e-03 | 1.024097e-02 |\n",
       "| ENSG00000211459.2 |  4123.886138 | -1.530304 | 0.3409141 | -4.488825 | 7.161696e-06 | 5.784783e-05 |\n",
       "\n"
      ],
      "text/plain": [
       "                   baseMean     log2FoldChange lfcSE     stat      pvalue      \n",
       "ENSG00000278791.1      6.612653 -2.302249      0.4649189 -4.951937 7.347834e-07\n",
       "ENSG00000229376.3     12.801501 -1.756743      0.4287367 -4.097487 4.176587e-05\n",
       "ENSG00000142609.18    13.785815  1.869731      0.3283950  5.693544 1.244291e-08\n",
       "ENSG00000187730.9     29.824951  2.674128      0.5126600  5.216183 1.826478e-07\n",
       "ENSG00000238164.6     40.461046 -1.773661      0.4521381 -3.922831 8.751443e-05\n",
       "ENSG00000157873.17   232.385304 -1.795708      0.3865943 -4.644942 3.401715e-06\n",
       "ENSG00000177133.11    10.966659  4.003949      0.8200935  4.882308 1.048511e-06\n",
       "ENSG00000142611.17    68.509740  4.330422      0.6370000  6.798151 1.059707e-11\n",
       "ENSG00000162591.16  8307.720221  1.723491      0.3207743  5.372910 7.747601e-08\n",
       "ENSG00000238290.1     13.087750 -1.720946      0.5031363 -3.420436 6.252075e-04\n",
       "ENSG00000131686.15     3.342750  2.932551      0.7919433  3.702981 2.130811e-04\n",
       "ENSG00000228526.7    564.749075 -1.979453      0.3774731 -5.243958 1.571678e-07\n",
       "ENSG00000234546.4    243.063482 -2.144048      0.4451617 -4.816335 1.462193e-06\n",
       "ENSG00000224340.1      5.738009 -1.634055      0.5758284 -2.837747 4.543319e-03\n",
       "ENSG00000199562.1      3.426536 -1.986978      0.6490028 -3.061586 2.201677e-03\n",
       "ENSG00000175262.14     4.162315  1.909322      0.6284430  3.038179 2.380128e-03\n",
       "ENSG00000235643.2      2.064915  3.126468      1.2402793  2.520777 1.170960e-02\n",
       "ENSG00000276470.1     11.000890 -1.833650      0.4553776 -4.026657 5.657540e-05\n",
       "ENSG00000224904.1      1.592345  3.174139      1.1585486  2.739755 6.148504e-03\n",
       "ENSG00000120949.15     3.947415  1.887533      0.6579427  2.868841 4.119791e-03\n",
       "ENSG00000162496.9  11434.244751  2.824843      0.4017785  7.030848 2.052821e-12\n",
       "ENSG00000276830.1     43.966641  1.942730      0.5121412  3.793349 1.486291e-04\n",
       "ENSG00000272482.1    143.835884  3.415021      0.5378340  6.349582 2.159009e-10\n",
       "ENSG00000204518.2      2.232951  4.536725      1.8693370  2.426917 1.522776e-02\n",
       "ENSG00000188984.12     1.468852  3.029707      1.2788664  2.369057 1.783353e-02\n",
       "ENSG00000116726.4      1.454513  3.169252      1.0848809  2.921290 3.485850e-03\n",
       "ENSG00000116721.9      1.807958  2.697376      1.1147098  2.419801 1.552900e-02\n",
       "ENSG00000204481.8      1.509265  3.547712      1.2913436  2.747303 6.008764e-03\n",
       "ENSG00000162494.6      7.509142  3.410109      1.0467941  3.257670 1.123311e-03\n",
       "ENSG00000233485.1      4.901759  2.376155      0.9123343  2.604479 9.201411e-03\n",
       "<U+22EE>             <U+22EE>     <U+22EE>       <U+22EE>  <U+22EE>  <U+22EE>    \n",
       "ENSG00000170965.10    50.049865  1.870657      0.2107735  8.875201 6.980729e-19\n",
       "ENSG00000181433.9      1.490064  3.543457      1.0233818  3.462498 5.351859e-04\n",
       "ENSG00000022267.19  8862.367219  5.307562      0.7194787  7.376955 1.619511e-13\n",
       "ENSG00000129680.16  3226.925586  3.086494      0.3935205  7.843286 4.389063e-15\n",
       "ENSG00000156920.11     8.299286  1.858064      0.6503765  2.856905 4.277933e-03\n",
       "ENSG00000156925.12    16.933753  2.849854      0.6358960  4.481635 7.407320e-06\n",
       "ENSG00000101977.21   667.205436 -2.171844      0.6405926 -3.390367 6.979912e-04\n",
       "ENSG00000230707.1      9.118160  2.417701      0.8632671  2.800641 5.100124e-03\n",
       "ENSG00000185985.10     9.849136  6.203665      0.7712065  8.044104 8.687899e-16\n",
       "ENSG00000155966.14   294.549288  1.696034      0.5106201  3.321519 8.952893e-04\n",
       "ENSG00000147381.11     1.345848  4.422309      1.6587917  2.665982 7.676379e-03\n",
       "ENSG00000102287.19   625.538353 -1.870973      0.3926244 -4.765299 1.885741e-06\n",
       "ENSG00000011677.13    58.713057 -3.059954      0.4504043 -6.793793 1.092232e-11\n",
       "ENSG00000268089.3     52.199893 -2.370338      0.7735584 -3.064200 2.182530e-03\n",
       "ENSG00000221867.9     83.448408 -6.231697      2.1609190 -2.883818 3.928859e-03\n",
       "ENSG00000198883.12     4.107412  1.936476      0.7373340  2.626321 8.631349e-03\n",
       "ENSG00000235961.5     27.783377  3.445541      0.8527000  4.040742 5.328237e-05\n",
       "ENSG00000189420.8      9.553255  1.790793      0.6911955  2.590864 9.573538e-03\n",
       "ENSG00000182492.16  3831.186718  2.708571      0.9559595  2.833353 4.606248e-03\n",
       "ENSG00000067842.17     7.671835  2.253908      0.6990900  3.224060 1.263867e-03\n",
       "ENSG00000130829.18   883.727537  3.014168      0.4570148  6.595340 4.242830e-11\n",
       "ENSG00000007350.17     2.189215  3.841916      1.1526952  3.332985 8.591965e-04\n",
       "ENSG00000285018.1    118.616157  1.807665      0.4912794  3.679505 2.336869e-04\n",
       "ENSG00000160211.19 31539.118406  1.729240      0.1657735 10.431344 1.783474e-25\n",
       "ENSG00000160219.12   189.036114  1.988261      0.2614332  7.605235 2.843871e-14\n",
       "ENSG00000155961.5     54.860647  2.253466      0.8531395  2.641380 8.256902e-03\n",
       "ENSG00000286217.2      1.416493  3.678475      1.2151377  3.027208 2.468237e-03\n",
       "ENSG00000114374.13     2.291132  3.784364      1.2238852  3.092091 1.987521e-03\n",
       "ENSG00000012817.15     2.051723  3.321876      1.1093026  2.994562 2.748391e-03\n",
       "ENSG00000211459.2   4123.886138 -1.530304      0.3409141 -4.488825 7.161696e-06\n",
       "                   padj        \n",
       "ENSG00000278791.1  7.714498e-06\n",
       "ENSG00000229376.3  2.741731e-04\n",
       "ENSG00000142609.18 1.983657e-07\n",
       "ENSG00000187730.9  2.238148e-06\n",
       "ENSG00000238164.6  5.237764e-04\n",
       "ENSG00000157873.17 2.984972e-05\n",
       "ENSG00000177133.11 1.060372e-05\n",
       "ENSG00000142611.17 3.358566e-10\n",
       "ENSG00000162591.16 1.024647e-06\n",
       "ENSG00000238290.1  2.887785e-03\n",
       "ENSG00000131686.15 1.138600e-03\n",
       "ENSG00000228526.7  1.957046e-06\n",
       "ENSG00000234546.4  1.418474e-05\n",
       "ENSG00000224340.1  1.570555e-02\n",
       "ENSG00000199562.1  8.484277e-03\n",
       "ENSG00000175262.14 9.067296e-03\n",
       "ENSG00000235643.2  3.480761e-02\n",
       "ENSG00000276470.1  3.585591e-04\n",
       "ENSG00000224904.1  2.029518e-02\n",
       "ENSG00000120949.15 1.445611e-02\n",
       "ENSG00000162496.9  7.610343e-11\n",
       "ENSG00000276830.1  8.353903e-04\n",
       "ENSG00000272482.1  5.189214e-09\n",
       "ENSG00000204518.2  4.332557e-02\n",
       "ENSG00000188984.12 4.952245e-02\n",
       "ENSG00000116726.4  1.253581e-02\n",
       "ENSG00000116721.9  4.402101e-02\n",
       "ENSG00000204481.8  1.989338e-02\n",
       "ENSG00000162494.6  4.788098e-03\n",
       "ENSG00000233485.1  2.845367e-02\n",
       "<U+22EE>             <U+22EE>    \n",
       "ENSG00000170965.10 7.825434e-17\n",
       "ENSG00000181433.9  2.520895e-03\n",
       "ENSG00000022267.19 7.459257e-12\n",
       "ENSG00000129680.16 2.668323e-13\n",
       "ENSG00000156920.11 1.492267e-02\n",
       "ENSG00000156925.12 5.967258e-05\n",
       "ENSG00000101977.21 3.184054e-03\n",
       "ENSG00000230707.1  1.732644e-02\n",
       "ENSG00000185985.10 6.095050e-14\n",
       "ENSG00000155966.14 3.944784e-03\n",
       "ENSG00000147381.11 2.447546e-02\n",
       "ENSG00000102287.19 1.766043e-05\n",
       "ENSG00000011677.13 3.448751e-10\n",
       "ENSG00000268089.3  8.425886e-03\n",
       "ENSG00000221867.9  1.388850e-02\n",
       "ENSG00000198883.12 2.700792e-02\n",
       "ENSG00000235961.5  3.398747e-04\n",
       "ENSG00000189420.8  2.941123e-02\n",
       "ENSG00000182492.16 1.588550e-02\n",
       "ENSG00000067842.17 5.301962e-03\n",
       "ENSG00000130829.18 1.182802e-09\n",
       "ENSG00000007350.17 3.806733e-03\n",
       "ENSG00000285018.1  1.235310e-03\n",
       "ENSG00000160211.19 5.211925e-23\n",
       "ENSG00000160219.12 1.517829e-12\n",
       "ENSG00000155961.5  2.603259e-02\n",
       "ENSG00000286217.2  9.346657e-03\n",
       "ENSG00000114374.13 7.780784e-03\n",
       "ENSG00000012817.15 1.024097e-02\n",
       "ENSG00000211459.2  5.784783e-05"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "geno_comp <- results(ddssva, contrast = c(\"genotype\", \"ex9\", \"ex20\"))\n",
    "geno_comp <- as.data.frame(geno_comp)\n",
    "geno_comp_filt <- geno_comp %>%\n",
    "  filter(log2FoldChange < -1.5 | log2FoldChange > 1.5) %>%\n",
    "  filter(padj < 0.05)\n",
    "geno_comp_filt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "e30adf3f",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 4374  3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>GeneID</th><th scope=col>log2FoldChange</th><th scope=col>padj</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>ENSG00000278791.1 </td><td>-2.302249</td><td>7.714498e-06</td></tr>\n",
       "\t<tr><td>ENSG00000229376.3 </td><td>-1.756743</td><td>2.741731e-04</td></tr>\n",
       "\t<tr><td>ENSG00000142609.18</td><td> 1.869731</td><td>1.983657e-07</td></tr>\n",
       "\t<tr><td>ENSG00000187730.9 </td><td> 2.674128</td><td>2.238148e-06</td></tr>\n",
       "\t<tr><td>ENSG00000238164.6 </td><td>-1.773661</td><td>5.237764e-04</td></tr>\n",
       "\t<tr><td>ENSG00000157873.17</td><td>-1.795708</td><td>2.984972e-05</td></tr>\n",
       "\t<tr><td>ENSG00000177133.11</td><td> 4.003949</td><td>1.060372e-05</td></tr>\n",
       "\t<tr><td>ENSG00000142611.17</td><td> 4.330422</td><td>3.358566e-10</td></tr>\n",
       "\t<tr><td>ENSG00000162591.16</td><td> 1.723491</td><td>1.024647e-06</td></tr>\n",
       "\t<tr><td>ENSG00000238290.1 </td><td>-1.720946</td><td>2.887785e-03</td></tr>\n",
       "\t<tr><td>ENSG00000131686.15</td><td> 2.932551</td><td>1.138600e-03</td></tr>\n",
       "\t<tr><td>ENSG00000228526.7 </td><td>-1.979453</td><td>1.957046e-06</td></tr>\n",
       "\t<tr><td>ENSG00000234546.4 </td><td>-2.144048</td><td>1.418474e-05</td></tr>\n",
       "\t<tr><td>ENSG00000224340.1 </td><td>-1.634055</td><td>1.570555e-02</td></tr>\n",
       "\t<tr><td>ENSG00000199562.1 </td><td>-1.986978</td><td>8.484277e-03</td></tr>\n",
       "\t<tr><td>ENSG00000175262.14</td><td> 1.909322</td><td>9.067296e-03</td></tr>\n",
       "\t<tr><td>ENSG00000235643.2 </td><td> 3.126468</td><td>3.480761e-02</td></tr>\n",
       "\t<tr><td>ENSG00000276470.1 </td><td>-1.833650</td><td>3.585591e-04</td></tr>\n",
       "\t<tr><td>ENSG00000224904.1 </td><td> 3.174139</td><td>2.029518e-02</td></tr>\n",
       "\t<tr><td>ENSG00000120949.15</td><td> 1.887533</td><td>1.445611e-02</td></tr>\n",
       "\t<tr><td>ENSG00000162496.9 </td><td> 2.824843</td><td>7.610343e-11</td></tr>\n",
       "\t<tr><td>ENSG00000276830.1 </td><td> 1.942730</td><td>8.353903e-04</td></tr>\n",
       "\t<tr><td>ENSG00000272482.1 </td><td> 3.415021</td><td>5.189214e-09</td></tr>\n",
       "\t<tr><td>ENSG00000204518.2 </td><td> 4.536725</td><td>4.332557e-02</td></tr>\n",
       "\t<tr><td>ENSG00000188984.12</td><td> 3.029707</td><td>4.952245e-02</td></tr>\n",
       "\t<tr><td>ENSG00000116726.4 </td><td> 3.169252</td><td>1.253581e-02</td></tr>\n",
       "\t<tr><td>ENSG00000116721.9 </td><td> 2.697376</td><td>4.402101e-02</td></tr>\n",
       "\t<tr><td>ENSG00000204481.8 </td><td> 3.547712</td><td>1.989338e-02</td></tr>\n",
       "\t<tr><td>ENSG00000162494.6 </td><td> 3.410109</td><td>4.788098e-03</td></tr>\n",
       "\t<tr><td>ENSG00000233485.1 </td><td> 2.376155</td><td>2.845367e-02</td></tr>\n",
       "\t<tr><td></td><td></td><td></td></tr>\n",
       "\t<tr><td>ENSG00000170965.10</td><td> 1.870657</td><td>7.825434e-17</td></tr>\n",
       "\t<tr><td>ENSG00000181433.9 </td><td> 3.543457</td><td>2.520895e-03</td></tr>\n",
       "\t<tr><td>ENSG00000022267.19</td><td> 5.307562</td><td>7.459257e-12</td></tr>\n",
       "\t<tr><td>ENSG00000129680.16</td><td> 3.086494</td><td>2.668323e-13</td></tr>\n",
       "\t<tr><td>ENSG00000156920.11</td><td> 1.858064</td><td>1.492267e-02</td></tr>\n",
       "\t<tr><td>ENSG00000156925.12</td><td> 2.849854</td><td>5.967258e-05</td></tr>\n",
       "\t<tr><td>ENSG00000101977.21</td><td>-2.171844</td><td>3.184054e-03</td></tr>\n",
       "\t<tr><td>ENSG00000230707.1 </td><td> 2.417701</td><td>1.732644e-02</td></tr>\n",
       "\t<tr><td>ENSG00000185985.10</td><td> 6.203665</td><td>6.095050e-14</td></tr>\n",
       "\t<tr><td>ENSG00000155966.14</td><td> 1.696034</td><td>3.944784e-03</td></tr>\n",
       "\t<tr><td>ENSG00000147381.11</td><td> 4.422309</td><td>2.447546e-02</td></tr>\n",
       "\t<tr><td>ENSG00000102287.19</td><td>-1.870973</td><td>1.766043e-05</td></tr>\n",
       "\t<tr><td>ENSG00000011677.13</td><td>-3.059954</td><td>3.448751e-10</td></tr>\n",
       "\t<tr><td>ENSG00000268089.3 </td><td>-2.370338</td><td>8.425886e-03</td></tr>\n",
       "\t<tr><td>ENSG00000221867.9 </td><td>-6.231697</td><td>1.388850e-02</td></tr>\n",
       "\t<tr><td>ENSG00000198883.12</td><td> 1.936476</td><td>2.700792e-02</td></tr>\n",
       "\t<tr><td>ENSG00000235961.5 </td><td> 3.445541</td><td>3.398747e-04</td></tr>\n",
       "\t<tr><td>ENSG00000189420.8 </td><td> 1.790793</td><td>2.941123e-02</td></tr>\n",
       "\t<tr><td>ENSG00000182492.16</td><td> 2.708571</td><td>1.588550e-02</td></tr>\n",
       "\t<tr><td>ENSG00000067842.17</td><td> 2.253908</td><td>5.301962e-03</td></tr>\n",
       "\t<tr><td>ENSG00000130829.18</td><td> 3.014168</td><td>1.182802e-09</td></tr>\n",
       "\t<tr><td>ENSG00000007350.17</td><td> 3.841916</td><td>3.806733e-03</td></tr>\n",
       "\t<tr><td>ENSG00000285018.1 </td><td> 1.807665</td><td>1.235310e-03</td></tr>\n",
       "\t<tr><td>ENSG00000160211.19</td><td> 1.729240</td><td>5.211925e-23</td></tr>\n",
       "\t<tr><td>ENSG00000160219.12</td><td> 1.988261</td><td>1.517829e-12</td></tr>\n",
       "\t<tr><td>ENSG00000155961.5 </td><td> 2.253466</td><td>2.603259e-02</td></tr>\n",
       "\t<tr><td>ENSG00000286217.2 </td><td> 3.678475</td><td>9.346657e-03</td></tr>\n",
       "\t<tr><td>ENSG00000114374.13</td><td> 3.784364</td><td>7.780784e-03</td></tr>\n",
       "\t<tr><td>ENSG00000012817.15</td><td> 3.321876</td><td>1.024097e-02</td></tr>\n",
       "\t<tr><td>ENSG00000211459.2 </td><td>-1.530304</td><td>5.784783e-05</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 4374  3\n",
       "\\begin{tabular}{lll}\n",
       " GeneID & log2FoldChange & padj\\\\\n",
       " <chr> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t ENSG00000278791.1  & -2.302249 & 7.714498e-06\\\\\n",
       "\t ENSG00000229376.3  & -1.756743 & 2.741731e-04\\\\\n",
       "\t ENSG00000142609.18 &  1.869731 & 1.983657e-07\\\\\n",
       "\t ENSG00000187730.9  &  2.674128 & 2.238148e-06\\\\\n",
       "\t ENSG00000238164.6  & -1.773661 & 5.237764e-04\\\\\n",
       "\t ENSG00000157873.17 & -1.795708 & 2.984972e-05\\\\\n",
       "\t ENSG00000177133.11 &  4.003949 & 1.060372e-05\\\\\n",
       "\t ENSG00000142611.17 &  4.330422 & 3.358566e-10\\\\\n",
       "\t ENSG00000162591.16 &  1.723491 & 1.024647e-06\\\\\n",
       "\t ENSG00000238290.1  & -1.720946 & 2.887785e-03\\\\\n",
       "\t ENSG00000131686.15 &  2.932551 & 1.138600e-03\\\\\n",
       "\t ENSG00000228526.7  & -1.979453 & 1.957046e-06\\\\\n",
       "\t ENSG00000234546.4  & -2.144048 & 1.418474e-05\\\\\n",
       "\t ENSG00000224340.1  & -1.634055 & 1.570555e-02\\\\\n",
       "\t ENSG00000199562.1  & -1.986978 & 8.484277e-03\\\\\n",
       "\t ENSG00000175262.14 &  1.909322 & 9.067296e-03\\\\\n",
       "\t ENSG00000235643.2  &  3.126468 & 3.480761e-02\\\\\n",
       "\t ENSG00000276470.1  & -1.833650 & 3.585591e-04\\\\\n",
       "\t ENSG00000224904.1  &  3.174139 & 2.029518e-02\\\\\n",
       "\t ENSG00000120949.15 &  1.887533 & 1.445611e-02\\\\\n",
       "\t ENSG00000162496.9  &  2.824843 & 7.610343e-11\\\\\n",
       "\t ENSG00000276830.1  &  1.942730 & 8.353903e-04\\\\\n",
       "\t ENSG00000272482.1  &  3.415021 & 5.189214e-09\\\\\n",
       "\t ENSG00000204518.2  &  4.536725 & 4.332557e-02\\\\\n",
       "\t ENSG00000188984.12 &  3.029707 & 4.952245e-02\\\\\n",
       "\t ENSG00000116726.4  &  3.169252 & 1.253581e-02\\\\\n",
       "\t ENSG00000116721.9  &  2.697376 & 4.402101e-02\\\\\n",
       "\t ENSG00000204481.8  &  3.547712 & 1.989338e-02\\\\\n",
       "\t ENSG00000162494.6  &  3.410109 & 4.788098e-03\\\\\n",
       "\t ENSG00000233485.1  &  2.376155 & 2.845367e-02\\\\\n",
       "\t  &  & \\\\\n",
       "\t ENSG00000170965.10 &  1.870657 & 7.825434e-17\\\\\n",
       "\t ENSG00000181433.9  &  3.543457 & 2.520895e-03\\\\\n",
       "\t ENSG00000022267.19 &  5.307562 & 7.459257e-12\\\\\n",
       "\t ENSG00000129680.16 &  3.086494 & 2.668323e-13\\\\\n",
       "\t ENSG00000156920.11 &  1.858064 & 1.492267e-02\\\\\n",
       "\t ENSG00000156925.12 &  2.849854 & 5.967258e-05\\\\\n",
       "\t ENSG00000101977.21 & -2.171844 & 3.184054e-03\\\\\n",
       "\t ENSG00000230707.1  &  2.417701 & 1.732644e-02\\\\\n",
       "\t ENSG00000185985.10 &  6.203665 & 6.095050e-14\\\\\n",
       "\t ENSG00000155966.14 &  1.696034 & 3.944784e-03\\\\\n",
       "\t ENSG00000147381.11 &  4.422309 & 2.447546e-02\\\\\n",
       "\t ENSG00000102287.19 & -1.870973 & 1.766043e-05\\\\\n",
       "\t ENSG00000011677.13 & -3.059954 & 3.448751e-10\\\\\n",
       "\t ENSG00000268089.3  & -2.370338 & 8.425886e-03\\\\\n",
       "\t ENSG00000221867.9  & -6.231697 & 1.388850e-02\\\\\n",
       "\t ENSG00000198883.12 &  1.936476 & 2.700792e-02\\\\\n",
       "\t ENSG00000235961.5  &  3.445541 & 3.398747e-04\\\\\n",
       "\t ENSG00000189420.8  &  1.790793 & 2.941123e-02\\\\\n",
       "\t ENSG00000182492.16 &  2.708571 & 1.588550e-02\\\\\n",
       "\t ENSG00000067842.17 &  2.253908 & 5.301962e-03\\\\\n",
       "\t ENSG00000130829.18 &  3.014168 & 1.182802e-09\\\\\n",
       "\t ENSG00000007350.17 &  3.841916 & 3.806733e-03\\\\\n",
       "\t ENSG00000285018.1  &  1.807665 & 1.235310e-03\\\\\n",
       "\t ENSG00000160211.19 &  1.729240 & 5.211925e-23\\\\\n",
       "\t ENSG00000160219.12 &  1.988261 & 1.517829e-12\\\\\n",
       "\t ENSG00000155961.5  &  2.253466 & 2.603259e-02\\\\\n",
       "\t ENSG00000286217.2  &  3.678475 & 9.346657e-03\\\\\n",
       "\t ENSG00000114374.13 &  3.784364 & 7.780784e-03\\\\\n",
       "\t ENSG00000012817.15 &  3.321876 & 1.024097e-02\\\\\n",
       "\t ENSG00000211459.2  & -1.530304 & 5.784783e-05\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 4374  3\n",
       "\n",
       "| GeneID &lt;chr&gt; | log2FoldChange &lt;dbl&gt; | padj &lt;dbl&gt; |\n",
       "|---|---|---|\n",
       "| ENSG00000278791.1  | -2.302249 | 7.714498e-06 |\n",
       "| ENSG00000229376.3  | -1.756743 | 2.741731e-04 |\n",
       "| ENSG00000142609.18 |  1.869731 | 1.983657e-07 |\n",
       "| ENSG00000187730.9  |  2.674128 | 2.238148e-06 |\n",
       "| ENSG00000238164.6  | -1.773661 | 5.237764e-04 |\n",
       "| ENSG00000157873.17 | -1.795708 | 2.984972e-05 |\n",
       "| ENSG00000177133.11 |  4.003949 | 1.060372e-05 |\n",
       "| ENSG00000142611.17 |  4.330422 | 3.358566e-10 |\n",
       "| ENSG00000162591.16 |  1.723491 | 1.024647e-06 |\n",
       "| ENSG00000238290.1  | -1.720946 | 2.887785e-03 |\n",
       "| ENSG00000131686.15 |  2.932551 | 1.138600e-03 |\n",
       "| ENSG00000228526.7  | -1.979453 | 1.957046e-06 |\n",
       "| ENSG00000234546.4  | -2.144048 | 1.418474e-05 |\n",
       "| ENSG00000224340.1  | -1.634055 | 1.570555e-02 |\n",
       "| ENSG00000199562.1  | -1.986978 | 8.484277e-03 |\n",
       "| ENSG00000175262.14 |  1.909322 | 9.067296e-03 |\n",
       "| ENSG00000235643.2  |  3.126468 | 3.480761e-02 |\n",
       "| ENSG00000276470.1  | -1.833650 | 3.585591e-04 |\n",
       "| ENSG00000224904.1  |  3.174139 | 2.029518e-02 |\n",
       "| ENSG00000120949.15 |  1.887533 | 1.445611e-02 |\n",
       "| ENSG00000162496.9  |  2.824843 | 7.610343e-11 |\n",
       "| ENSG00000276830.1  |  1.942730 | 8.353903e-04 |\n",
       "| ENSG00000272482.1  |  3.415021 | 5.189214e-09 |\n",
       "| ENSG00000204518.2  |  4.536725 | 4.332557e-02 |\n",
       "| ENSG00000188984.12 |  3.029707 | 4.952245e-02 |\n",
       "| ENSG00000116726.4  |  3.169252 | 1.253581e-02 |\n",
       "| ENSG00000116721.9  |  2.697376 | 4.402101e-02 |\n",
       "| ENSG00000204481.8  |  3.547712 | 1.989338e-02 |\n",
       "| ENSG00000162494.6  |  3.410109 | 4.788098e-03 |\n",
       "| ENSG00000233485.1  |  2.376155 | 2.845367e-02 |\n",
       "|  |  |  |\n",
       "| ENSG00000170965.10 |  1.870657 | 7.825434e-17 |\n",
       "| ENSG00000181433.9  |  3.543457 | 2.520895e-03 |\n",
       "| ENSG00000022267.19 |  5.307562 | 7.459257e-12 |\n",
       "| ENSG00000129680.16 |  3.086494 | 2.668323e-13 |\n",
       "| ENSG00000156920.11 |  1.858064 | 1.492267e-02 |\n",
       "| ENSG00000156925.12 |  2.849854 | 5.967258e-05 |\n",
       "| ENSG00000101977.21 | -2.171844 | 3.184054e-03 |\n",
       "| ENSG00000230707.1  |  2.417701 | 1.732644e-02 |\n",
       "| ENSG00000185985.10 |  6.203665 | 6.095050e-14 |\n",
       "| ENSG00000155966.14 |  1.696034 | 3.944784e-03 |\n",
       "| ENSG00000147381.11 |  4.422309 | 2.447546e-02 |\n",
       "| ENSG00000102287.19 | -1.870973 | 1.766043e-05 |\n",
       "| ENSG00000011677.13 | -3.059954 | 3.448751e-10 |\n",
       "| ENSG00000268089.3  | -2.370338 | 8.425886e-03 |\n",
       "| ENSG00000221867.9  | -6.231697 | 1.388850e-02 |\n",
       "| ENSG00000198883.12 |  1.936476 | 2.700792e-02 |\n",
       "| ENSG00000235961.5  |  3.445541 | 3.398747e-04 |\n",
       "| ENSG00000189420.8  |  1.790793 | 2.941123e-02 |\n",
       "| ENSG00000182492.16 |  2.708571 | 1.588550e-02 |\n",
       "| ENSG00000067842.17 |  2.253908 | 5.301962e-03 |\n",
       "| ENSG00000130829.18 |  3.014168 | 1.182802e-09 |\n",
       "| ENSG00000007350.17 |  3.841916 | 3.806733e-03 |\n",
       "| ENSG00000285018.1  |  1.807665 | 1.235310e-03 |\n",
       "| ENSG00000160211.19 |  1.729240 | 5.211925e-23 |\n",
       "| ENSG00000160219.12 |  1.988261 | 1.517829e-12 |\n",
       "| ENSG00000155961.5  |  2.253466 | 2.603259e-02 |\n",
       "| ENSG00000286217.2  |  3.678475 | 9.346657e-03 |\n",
       "| ENSG00000114374.13 |  3.784364 | 7.780784e-03 |\n",
       "| ENSG00000012817.15 |  3.321876 | 1.024097e-02 |\n",
       "| ENSG00000211459.2  | -1.530304 | 5.784783e-05 |\n",
       "\n"
      ],
      "text/plain": [
       "       GeneID             log2FoldChange padj        \n",
       "1      ENSG00000278791.1  -2.302249      7.714498e-06\n",
       "2      ENSG00000229376.3  -1.756743      2.741731e-04\n",
       "3      ENSG00000142609.18  1.869731      1.983657e-07\n",
       "4      ENSG00000187730.9   2.674128      2.238148e-06\n",
       "5      ENSG00000238164.6  -1.773661      5.237764e-04\n",
       "6      ENSG00000157873.17 -1.795708      2.984972e-05\n",
       "7      ENSG00000177133.11  4.003949      1.060372e-05\n",
       "8      ENSG00000142611.17  4.330422      3.358566e-10\n",
       "9      ENSG00000162591.16  1.723491      1.024647e-06\n",
       "10     ENSG00000238290.1  -1.720946      2.887785e-03\n",
       "11     ENSG00000131686.15  2.932551      1.138600e-03\n",
       "12     ENSG00000228526.7  -1.979453      1.957046e-06\n",
       "13     ENSG00000234546.4  -2.144048      1.418474e-05\n",
       "14     ENSG00000224340.1  -1.634055      1.570555e-02\n",
       "15     ENSG00000199562.1  -1.986978      8.484277e-03\n",
       "16     ENSG00000175262.14  1.909322      9.067296e-03\n",
       "17     ENSG00000235643.2   3.126468      3.480761e-02\n",
       "18     ENSG00000276470.1  -1.833650      3.585591e-04\n",
       "19     ENSG00000224904.1   3.174139      2.029518e-02\n",
       "20     ENSG00000120949.15  1.887533      1.445611e-02\n",
       "21     ENSG00000162496.9   2.824843      7.610343e-11\n",
       "22     ENSG00000276830.1   1.942730      8.353903e-04\n",
       "23     ENSG00000272482.1   3.415021      5.189214e-09\n",
       "24     ENSG00000204518.2   4.536725      4.332557e-02\n",
       "25     ENSG00000188984.12  3.029707      4.952245e-02\n",
       "26     ENSG00000116726.4   3.169252      1.253581e-02\n",
       "27     ENSG00000116721.9   2.697376      4.402101e-02\n",
       "28     ENSG00000204481.8   3.547712      1.989338e-02\n",
       "29     ENSG00000162494.6   3.410109      4.788098e-03\n",
       "30     ENSG00000233485.1   2.376155      2.845367e-02\n",
       "<U+22EE> <U+22EE>           <U+22EE>       <U+22EE>    \n",
       "4345   ENSG00000170965.10  1.870657      7.825434e-17\n",
       "4346   ENSG00000181433.9   3.543457      2.520895e-03\n",
       "4347   ENSG00000022267.19  5.307562      7.459257e-12\n",
       "4348   ENSG00000129680.16  3.086494      2.668323e-13\n",
       "4349   ENSG00000156920.11  1.858064      1.492267e-02\n",
       "4350   ENSG00000156925.12  2.849854      5.967258e-05\n",
       "4351   ENSG00000101977.21 -2.171844      3.184054e-03\n",
       "4352   ENSG00000230707.1   2.417701      1.732644e-02\n",
       "4353   ENSG00000185985.10  6.203665      6.095050e-14\n",
       "4354   ENSG00000155966.14  1.696034      3.944784e-03\n",
       "4355   ENSG00000147381.11  4.422309      2.447546e-02\n",
       "4356   ENSG00000102287.19 -1.870973      1.766043e-05\n",
       "4357   ENSG00000011677.13 -3.059954      3.448751e-10\n",
       "4358   ENSG00000268089.3  -2.370338      8.425886e-03\n",
       "4359   ENSG00000221867.9  -6.231697      1.388850e-02\n",
       "4360   ENSG00000198883.12  1.936476      2.700792e-02\n",
       "4361   ENSG00000235961.5   3.445541      3.398747e-04\n",
       "4362   ENSG00000189420.8   1.790793      2.941123e-02\n",
       "4363   ENSG00000182492.16  2.708571      1.588550e-02\n",
       "4364   ENSG00000067842.17  2.253908      5.301962e-03\n",
       "4365   ENSG00000130829.18  3.014168      1.182802e-09\n",
       "4366   ENSG00000007350.17  3.841916      3.806733e-03\n",
       "4367   ENSG00000285018.1   1.807665      1.235310e-03\n",
       "4368   ENSG00000160211.19  1.729240      5.211925e-23\n",
       "4369   ENSG00000160219.12  1.988261      1.517829e-12\n",
       "4370   ENSG00000155961.5   2.253466      2.603259e-02\n",
       "4371   ENSG00000286217.2   3.678475      9.346657e-03\n",
       "4372   ENSG00000114374.13  3.784364      7.780784e-03\n",
       "4373   ENSG00000012817.15  3.321876      1.024097e-02\n",
       "4374   ENSG00000211459.2  -1.530304      5.784783e-05"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "prePathfindR_df <- geno_comp_filt %>%\n",
    "                    rownames_to_column(\"GeneID\") %>%\n",
    "                    dplyr::select(c(\"GeneID\", \"log2FoldChange\", \"padj\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "cd574f1d",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 4367  3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>hgnc_symbol</th><th scope=col>log2FoldChange</th><th scope=col>padj</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>CFH    </td><td> 2.000816</td><td>2.186195e-07</td></tr>\n",
       "\t<tr><td>AOC1   </td><td>-2.576799</td><td>5.349419e-03</td></tr>\n",
       "\t<tr><td>WNT16  </td><td> 3.088030</td><td>1.045158e-06</td></tr>\n",
       "\t<tr><td>TFPI   </td><td> 1.910179</td><td>6.260387e-10</td></tr>\n",
       "\t<tr><td>SLC7A2 </td><td> 3.866093</td><td>2.657544e-24</td></tr>\n",
       "\t<tr><td>ABCB5  </td><td> 1.632194</td><td>3.167950e-02</td></tr>\n",
       "\t<tr><td>SKAP2  </td><td> 2.119249</td><td>2.514852e-11</td></tr>\n",
       "\t<tr><td>MPO    </td><td> 2.904284</td><td>1.020554e-03</td></tr>\n",
       "\t<tr><td>SOX8   </td><td> 3.605927</td><td>2.937395e-05</td></tr>\n",
       "\t<tr><td>TMEM98 </td><td>-2.766673</td><td>3.800028e-08</td></tr>\n",
       "\t<tr><td>ABCC8  </td><td> 3.675937</td><td>2.870305e-15</td></tr>\n",
       "\t<tr><td>DLX6   </td><td> 5.108520</td><td>4.016659e-07</td></tr>\n",
       "\t<tr><td>USH1C  </td><td> 1.834305</td><td>1.744765e-02</td></tr>\n",
       "\t<tr><td>PRSS21 </td><td> 1.535662</td><td>1.595898e-03</td></tr>\n",
       "\t<tr><td>MATK   </td><td> 3.243889</td><td>4.458763e-12</td></tr>\n",
       "\t<tr><td>SCN4A  </td><td> 3.322588</td><td>5.280402e-08</td></tr>\n",
       "\t<tr><td>TKTL1  </td><td> 3.841916</td><td>3.806733e-03</td></tr>\n",
       "\t<tr><td>IL32   </td><td>-1.759436</td><td>2.298351e-07</td></tr>\n",
       "\t<tr><td>PAX7   </td><td> 3.863516</td><td>2.236451e-04</td></tr>\n",
       "\t<tr><td>FMO1   </td><td> 1.880446</td><td>1.680261e-05</td></tr>\n",
       "\t<tr><td>SLC6A7 </td><td> 2.498799</td><td>1.310032e-02</td></tr>\n",
       "\t<tr><td>GABRA3 </td><td>-3.059954</td><td>3.448751e-10</td></tr>\n",
       "\t<tr><td>SEMA3B </td><td> 1.570285</td><td>3.656304e-07</td></tr>\n",
       "\t<tr><td>LTF    </td><td>-1.656888</td><td>2.354533e-03</td></tr>\n",
       "\t<tr><td>NR1H4  </td><td> 4.332383</td><td>8.178697e-05</td></tr>\n",
       "\t<tr><td>ALOX5  </td><td>-2.172160</td><td>1.932958e-03</td></tr>\n",
       "\t<tr><td>KDM5D  </td><td> 3.321876</td><td>1.024097e-02</td></tr>\n",
       "\t<tr><td>SLC7A14</td><td> 2.238593</td><td>1.155544e-02</td></tr>\n",
       "\t<tr><td>IGF1   </td><td>-2.633200</td><td>5.468134e-05</td></tr>\n",
       "\t<tr><td>CNTN1  </td><td> 1.634947</td><td>6.128467e-08</td></tr>\n",
       "\t<tr><td></td><td></td><td></td></tr>\n",
       "\t<tr><td>       </td><td> 2.400793</td><td>6.245499e-04</td></tr>\n",
       "\t<tr><td>       </td><td> 3.579139</td><td>1.716630e-03</td></tr>\n",
       "\t<tr><td>       </td><td> 3.577967</td><td>9.964412e-03</td></tr>\n",
       "\t<tr><td>       </td><td> 2.966998</td><td>6.134898e-04</td></tr>\n",
       "\t<tr><td>       </td><td> 4.177529</td><td>1.881429e-06</td></tr>\n",
       "\t<tr><td>       </td><td> 2.256452</td><td>1.198140e-03</td></tr>\n",
       "\t<tr><td>       </td><td> 2.935816</td><td>1.260149e-02</td></tr>\n",
       "\t<tr><td>       </td><td> 2.890514</td><td>1.169877e-02</td></tr>\n",
       "\t<tr><td>       </td><td>-1.970837</td><td>4.347253e-02</td></tr>\n",
       "\t<tr><td>       </td><td> 2.506229</td><td>4.712007e-02</td></tr>\n",
       "\t<tr><td>       </td><td>-2.465417</td><td>6.209511e-05</td></tr>\n",
       "\t<tr><td>       </td><td> 1.722960</td><td>3.203485e-04</td></tr>\n",
       "\t<tr><td>       </td><td> 3.249567</td><td>1.666688e-03</td></tr>\n",
       "\t<tr><td>       </td><td> 2.689127</td><td>1.273936e-02</td></tr>\n",
       "\t<tr><td>       </td><td> 4.528986</td><td>1.630010e-06</td></tr>\n",
       "\t<tr><td>       </td><td> 1.751495</td><td>6.059852e-09</td></tr>\n",
       "\t<tr><td>       </td><td> 4.192971</td><td>5.044360e-04</td></tr>\n",
       "\t<tr><td>       </td><td> 3.358857</td><td>2.061833e-05</td></tr>\n",
       "\t<tr><td>       </td><td> 1.844520</td><td>3.014890e-05</td></tr>\n",
       "\t<tr><td>       </td><td>-3.001874</td><td>2.762255e-06</td></tr>\n",
       "\t<tr><td>       </td><td>-1.882020</td><td>3.528529e-04</td></tr>\n",
       "\t<tr><td>       </td><td>-2.016672</td><td>4.233078e-02</td></tr>\n",
       "\t<tr><td>       </td><td>-1.652164</td><td>8.799479e-03</td></tr>\n",
       "\t<tr><td>       </td><td> 2.678990</td><td>4.062568e-02</td></tr>\n",
       "\t<tr><td>       </td><td> 3.343057</td><td>1.712519e-02</td></tr>\n",
       "\t<tr><td>       </td><td> 4.877521</td><td>1.574210e-04</td></tr>\n",
       "\t<tr><td>       </td><td> 4.517112</td><td>9.530196e-03</td></tr>\n",
       "\t<tr><td>       </td><td> 1.548025</td><td>6.029857e-04</td></tr>\n",
       "\t<tr><td>FAM106C</td><td>-1.610026</td><td>3.309201e-02</td></tr>\n",
       "\t<tr><td>       </td><td> 3.263966</td><td>9.363160e-09</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 4367  3\n",
       "\\begin{tabular}{lll}\n",
       " hgnc\\_symbol & log2FoldChange & padj\\\\\n",
       " <chr> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t CFH     &  2.000816 & 2.186195e-07\\\\\n",
       "\t AOC1    & -2.576799 & 5.349419e-03\\\\\n",
       "\t WNT16   &  3.088030 & 1.045158e-06\\\\\n",
       "\t TFPI    &  1.910179 & 6.260387e-10\\\\\n",
       "\t SLC7A2  &  3.866093 & 2.657544e-24\\\\\n",
       "\t ABCB5   &  1.632194 & 3.167950e-02\\\\\n",
       "\t SKAP2   &  2.119249 & 2.514852e-11\\\\\n",
       "\t MPO     &  2.904284 & 1.020554e-03\\\\\n",
       "\t SOX8    &  3.605927 & 2.937395e-05\\\\\n",
       "\t TMEM98  & -2.766673 & 3.800028e-08\\\\\n",
       "\t ABCC8   &  3.675937 & 2.870305e-15\\\\\n",
       "\t DLX6    &  5.108520 & 4.016659e-07\\\\\n",
       "\t USH1C   &  1.834305 & 1.744765e-02\\\\\n",
       "\t PRSS21  &  1.535662 & 1.595898e-03\\\\\n",
       "\t MATK    &  3.243889 & 4.458763e-12\\\\\n",
       "\t SCN4A   &  3.322588 & 5.280402e-08\\\\\n",
       "\t TKTL1   &  3.841916 & 3.806733e-03\\\\\n",
       "\t IL32    & -1.759436 & 2.298351e-07\\\\\n",
       "\t PAX7    &  3.863516 & 2.236451e-04\\\\\n",
       "\t FMO1    &  1.880446 & 1.680261e-05\\\\\n",
       "\t SLC6A7  &  2.498799 & 1.310032e-02\\\\\n",
       "\t GABRA3  & -3.059954 & 3.448751e-10\\\\\n",
       "\t SEMA3B  &  1.570285 & 3.656304e-07\\\\\n",
       "\t LTF     & -1.656888 & 2.354533e-03\\\\\n",
       "\t NR1H4   &  4.332383 & 8.178697e-05\\\\\n",
       "\t ALOX5   & -2.172160 & 1.932958e-03\\\\\n",
       "\t KDM5D   &  3.321876 & 1.024097e-02\\\\\n",
       "\t SLC7A14 &  2.238593 & 1.155544e-02\\\\\n",
       "\t IGF1    & -2.633200 & 5.468134e-05\\\\\n",
       "\t CNTN1   &  1.634947 & 6.128467e-08\\\\\n",
       "\t  &  & \\\\\n",
       "\t         &  2.400793 & 6.245499e-04\\\\\n",
       "\t         &  3.579139 & 1.716630e-03\\\\\n",
       "\t         &  3.577967 & 9.964412e-03\\\\\n",
       "\t         &  2.966998 & 6.134898e-04\\\\\n",
       "\t         &  4.177529 & 1.881429e-06\\\\\n",
       "\t         &  2.256452 & 1.198140e-03\\\\\n",
       "\t         &  2.935816 & 1.260149e-02\\\\\n",
       "\t         &  2.890514 & 1.169877e-02\\\\\n",
       "\t         & -1.970837 & 4.347253e-02\\\\\n",
       "\t         &  2.506229 & 4.712007e-02\\\\\n",
       "\t         & -2.465417 & 6.209511e-05\\\\\n",
       "\t         &  1.722960 & 3.203485e-04\\\\\n",
       "\t         &  3.249567 & 1.666688e-03\\\\\n",
       "\t         &  2.689127 & 1.273936e-02\\\\\n",
       "\t         &  4.528986 & 1.630010e-06\\\\\n",
       "\t         &  1.751495 & 6.059852e-09\\\\\n",
       "\t         &  4.192971 & 5.044360e-04\\\\\n",
       "\t         &  3.358857 & 2.061833e-05\\\\\n",
       "\t         &  1.844520 & 3.014890e-05\\\\\n",
       "\t         & -3.001874 & 2.762255e-06\\\\\n",
       "\t         & -1.882020 & 3.528529e-04\\\\\n",
       "\t         & -2.016672 & 4.233078e-02\\\\\n",
       "\t         & -1.652164 & 8.799479e-03\\\\\n",
       "\t         &  2.678990 & 4.062568e-02\\\\\n",
       "\t         &  3.343057 & 1.712519e-02\\\\\n",
       "\t         &  4.877521 & 1.574210e-04\\\\\n",
       "\t         &  4.517112 & 9.530196e-03\\\\\n",
       "\t         &  1.548025 & 6.029857e-04\\\\\n",
       "\t FAM106C & -1.610026 & 3.309201e-02\\\\\n",
       "\t         &  3.263966 & 9.363160e-09\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 4367  3\n",
       "\n",
       "| hgnc_symbol &lt;chr&gt; | log2FoldChange &lt;dbl&gt; | padj &lt;dbl&gt; |\n",
       "|---|---|---|\n",
       "| CFH     |  2.000816 | 2.186195e-07 |\n",
       "| AOC1    | -2.576799 | 5.349419e-03 |\n",
       "| WNT16   |  3.088030 | 1.045158e-06 |\n",
       "| TFPI    |  1.910179 | 6.260387e-10 |\n",
       "| SLC7A2  |  3.866093 | 2.657544e-24 |\n",
       "| ABCB5   |  1.632194 | 3.167950e-02 |\n",
       "| SKAP2   |  2.119249 | 2.514852e-11 |\n",
       "| MPO     |  2.904284 | 1.020554e-03 |\n",
       "| SOX8    |  3.605927 | 2.937395e-05 |\n",
       "| TMEM98  | -2.766673 | 3.800028e-08 |\n",
       "| ABCC8   |  3.675937 | 2.870305e-15 |\n",
       "| DLX6    |  5.108520 | 4.016659e-07 |\n",
       "| USH1C   |  1.834305 | 1.744765e-02 |\n",
       "| PRSS21  |  1.535662 | 1.595898e-03 |\n",
       "| MATK    |  3.243889 | 4.458763e-12 |\n",
       "| SCN4A   |  3.322588 | 5.280402e-08 |\n",
       "| TKTL1   |  3.841916 | 3.806733e-03 |\n",
       "| IL32    | -1.759436 | 2.298351e-07 |\n",
       "| PAX7    |  3.863516 | 2.236451e-04 |\n",
       "| FMO1    |  1.880446 | 1.680261e-05 |\n",
       "| SLC6A7  |  2.498799 | 1.310032e-02 |\n",
       "| GABRA3  | -3.059954 | 3.448751e-10 |\n",
       "| SEMA3B  |  1.570285 | 3.656304e-07 |\n",
       "| LTF     | -1.656888 | 2.354533e-03 |\n",
       "| NR1H4   |  4.332383 | 8.178697e-05 |\n",
       "| ALOX5   | -2.172160 | 1.932958e-03 |\n",
       "| KDM5D   |  3.321876 | 1.024097e-02 |\n",
       "| SLC7A14 |  2.238593 | 1.155544e-02 |\n",
       "| IGF1    | -2.633200 | 5.468134e-05 |\n",
       "| CNTN1   |  1.634947 | 6.128467e-08 |\n",
       "|  |  |  |\n",
       "| <!----> |  2.400793 | 6.245499e-04 |\n",
       "| <!----> |  3.579139 | 1.716630e-03 |\n",
       "| <!----> |  3.577967 | 9.964412e-03 |\n",
       "| <!----> |  2.966998 | 6.134898e-04 |\n",
       "| <!----> |  4.177529 | 1.881429e-06 |\n",
       "| <!----> |  2.256452 | 1.198140e-03 |\n",
       "| <!----> |  2.935816 | 1.260149e-02 |\n",
       "| <!----> |  2.890514 | 1.169877e-02 |\n",
       "| <!----> | -1.970837 | 4.347253e-02 |\n",
       "| <!----> |  2.506229 | 4.712007e-02 |\n",
       "| <!----> | -2.465417 | 6.209511e-05 |\n",
       "| <!----> |  1.722960 | 3.203485e-04 |\n",
       "| <!----> |  3.249567 | 1.666688e-03 |\n",
       "| <!----> |  2.689127 | 1.273936e-02 |\n",
       "| <!----> |  4.528986 | 1.630010e-06 |\n",
       "| <!----> |  1.751495 | 6.059852e-09 |\n",
       "| <!----> |  4.192971 | 5.044360e-04 |\n",
       "| <!----> |  3.358857 | 2.061833e-05 |\n",
       "| <!----> |  1.844520 | 3.014890e-05 |\n",
       "| <!----> | -3.001874 | 2.762255e-06 |\n",
       "| <!----> | -1.882020 | 3.528529e-04 |\n",
       "| <!----> | -2.016672 | 4.233078e-02 |\n",
       "| <!----> | -1.652164 | 8.799479e-03 |\n",
       "| <!----> |  2.678990 | 4.062568e-02 |\n",
       "| <!----> |  3.343057 | 1.712519e-02 |\n",
       "| <!----> |  4.877521 | 1.574210e-04 |\n",
       "| <!----> |  4.517112 | 9.530196e-03 |\n",
       "| <!----> |  1.548025 | 6.029857e-04 |\n",
       "| FAM106C | -1.610026 | 3.309201e-02 |\n",
       "| <!----> |  3.263966 | 9.363160e-09 |\n",
       "\n"
      ],
      "text/plain": [
       "       hgnc_symbol log2FoldChange padj        \n",
       "1      CFH          2.000816      2.186195e-07\n",
       "2      AOC1        -2.576799      5.349419e-03\n",
       "3      WNT16        3.088030      1.045158e-06\n",
       "4      TFPI         1.910179      6.260387e-10\n",
       "5      SLC7A2       3.866093      2.657544e-24\n",
       "6      ABCB5        1.632194      3.167950e-02\n",
       "7      SKAP2        2.119249      2.514852e-11\n",
       "8      MPO          2.904284      1.020554e-03\n",
       "9      SOX8         3.605927      2.937395e-05\n",
       "10     TMEM98      -2.766673      3.800028e-08\n",
       "11     ABCC8        3.675937      2.870305e-15\n",
       "12     DLX6         5.108520      4.016659e-07\n",
       "13     USH1C        1.834305      1.744765e-02\n",
       "14     PRSS21       1.535662      1.595898e-03\n",
       "15     MATK         3.243889      4.458763e-12\n",
       "16     SCN4A        3.322588      5.280402e-08\n",
       "17     TKTL1        3.841916      3.806733e-03\n",
       "18     IL32        -1.759436      2.298351e-07\n",
       "19     PAX7         3.863516      2.236451e-04\n",
       "20     FMO1         1.880446      1.680261e-05\n",
       "21     SLC6A7       2.498799      1.310032e-02\n",
       "22     GABRA3      -3.059954      3.448751e-10\n",
       "23     SEMA3B       1.570285      3.656304e-07\n",
       "24     LTF         -1.656888      2.354533e-03\n",
       "25     NR1H4        4.332383      8.178697e-05\n",
       "26     ALOX5       -2.172160      1.932958e-03\n",
       "27     KDM5D        3.321876      1.024097e-02\n",
       "28     SLC7A14      2.238593      1.155544e-02\n",
       "29     IGF1        -2.633200      5.468134e-05\n",
       "30     CNTN1        1.634947      6.128467e-08\n",
       "<U+22EE> <U+22EE>    <U+22EE>       <U+22EE>    \n",
       "4338                2.400793      6.245499e-04\n",
       "4339                3.579139      1.716630e-03\n",
       "4340                3.577967      9.964412e-03\n",
       "4341                2.966998      6.134898e-04\n",
       "4342                4.177529      1.881429e-06\n",
       "4343                2.256452      1.198140e-03\n",
       "4344                2.935816      1.260149e-02\n",
       "4345                2.890514      1.169877e-02\n",
       "4346               -1.970837      4.347253e-02\n",
       "4347                2.506229      4.712007e-02\n",
       "4348               -2.465417      6.209511e-05\n",
       "4349                1.722960      3.203485e-04\n",
       "4350                3.249567      1.666688e-03\n",
       "4351                2.689127      1.273936e-02\n",
       "4352                4.528986      1.630010e-06\n",
       "4353                1.751495      6.059852e-09\n",
       "4354                4.192971      5.044360e-04\n",
       "4355                3.358857      2.061833e-05\n",
       "4356                1.844520      3.014890e-05\n",
       "4357               -3.001874      2.762255e-06\n",
       "4358               -1.882020      3.528529e-04\n",
       "4359               -2.016672      4.233078e-02\n",
       "4360               -1.652164      8.799479e-03\n",
       "4361                2.678990      4.062568e-02\n",
       "4362                3.343057      1.712519e-02\n",
       "4363                4.877521      1.574210e-04\n",
       "4364                4.517112      9.530196e-03\n",
       "4365                1.548025      6.029857e-04\n",
       "4366   FAM106C     -1.610026      3.309201e-02\n",
       "4367                3.263966      9.363160e-09"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "prePathfindR_df\n",
    "write_tsv(prePathfindR_df, file = \"GenoDirectComp_SVA.txt\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "9d45a5cd",
   "metadata": {},
   "outputs": [],
   "source": [
    "library(biomaRt)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "c7790a52",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 4367  4</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>GeneID</th><th scope=col>log2FoldChange</th><th scope=col>padj</th><th scope=col>hgnc_symbol</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>ENSG00000000971</td><td> 2.000816</td><td>2.186195e-07</td><td>CFH    </td></tr>\n",
       "\t<tr><td>ENSG00000002726</td><td>-2.576799</td><td>5.349419e-03</td><td>AOC1   </td></tr>\n",
       "\t<tr><td>ENSG00000002745</td><td> 3.088030</td><td>1.045158e-06</td><td>WNT16  </td></tr>\n",
       "\t<tr><td>ENSG00000003436</td><td> 1.910179</td><td>6.260387e-10</td><td>TFPI   </td></tr>\n",
       "\t<tr><td>ENSG00000003989</td><td> 3.866093</td><td>2.657544e-24</td><td>SLC7A2 </td></tr>\n",
       "\t<tr><td>ENSG00000004846</td><td> 1.632194</td><td>3.167950e-02</td><td>ABCB5  </td></tr>\n",
       "\t<tr><td>ENSG00000005020</td><td> 2.119249</td><td>2.514852e-11</td><td>SKAP2  </td></tr>\n",
       "\t<tr><td>ENSG00000005381</td><td> 2.904284</td><td>1.020554e-03</td><td>MPO    </td></tr>\n",
       "\t<tr><td>ENSG00000005513</td><td> 3.605927</td><td>2.937395e-05</td><td>SOX8   </td></tr>\n",
       "\t<tr><td>ENSG00000006042</td><td>-2.766673</td><td>3.800028e-08</td><td>TMEM98 </td></tr>\n",
       "\t<tr><td>ENSG00000006071</td><td> 3.675937</td><td>2.870305e-15</td><td>ABCC8  </td></tr>\n",
       "\t<tr><td>ENSG00000006377</td><td> 5.108520</td><td>4.016659e-07</td><td>DLX6   </td></tr>\n",
       "\t<tr><td>ENSG00000006611</td><td> 1.834305</td><td>1.744765e-02</td><td>USH1C  </td></tr>\n",
       "\t<tr><td>ENSG00000007038</td><td> 1.535662</td><td>1.595898e-03</td><td>PRSS21 </td></tr>\n",
       "\t<tr><td>ENSG00000007264</td><td> 3.243889</td><td>4.458763e-12</td><td>MATK   </td></tr>\n",
       "\t<tr><td>ENSG00000007314</td><td> 3.322588</td><td>5.280402e-08</td><td>SCN4A  </td></tr>\n",
       "\t<tr><td>ENSG00000007350</td><td> 3.841916</td><td>3.806733e-03</td><td>TKTL1  </td></tr>\n",
       "\t<tr><td>ENSG00000008517</td><td>-1.759436</td><td>2.298351e-07</td><td>IL32   </td></tr>\n",
       "\t<tr><td>ENSG00000009709</td><td> 3.863516</td><td>2.236451e-04</td><td>PAX7   </td></tr>\n",
       "\t<tr><td>ENSG00000010932</td><td> 1.880446</td><td>1.680261e-05</td><td>FMO1   </td></tr>\n",
       "\t<tr><td>ENSG00000011083</td><td> 2.498799</td><td>1.310032e-02</td><td>SLC6A7 </td></tr>\n",
       "\t<tr><td>ENSG00000011677</td><td>-3.059954</td><td>3.448751e-10</td><td>GABRA3 </td></tr>\n",
       "\t<tr><td>ENSG00000012171</td><td> 1.570285</td><td>3.656304e-07</td><td>SEMA3B </td></tr>\n",
       "\t<tr><td>ENSG00000012223</td><td>-1.656888</td><td>2.354533e-03</td><td>LTF    </td></tr>\n",
       "\t<tr><td>ENSG00000012504</td><td> 4.332383</td><td>8.178697e-05</td><td>NR1H4  </td></tr>\n",
       "\t<tr><td>ENSG00000012779</td><td>-2.172160</td><td>1.932958e-03</td><td>ALOX5  </td></tr>\n",
       "\t<tr><td>ENSG00000012817</td><td> 3.321876</td><td>1.024097e-02</td><td>KDM5D  </td></tr>\n",
       "\t<tr><td>ENSG00000013293</td><td> 2.238593</td><td>1.155544e-02</td><td>SLC7A14</td></tr>\n",
       "\t<tr><td>ENSG00000017427</td><td>-2.633200</td><td>5.468134e-05</td><td>IGF1   </td></tr>\n",
       "\t<tr><td>ENSG00000018236</td><td> 1.634947</td><td>6.128467e-08</td><td>CNTN1  </td></tr>\n",
       "\t<tr><td></td><td></td><td></td><td></td></tr>\n",
       "\t<tr><td>ENSG00000287886</td><td> 2.400793</td><td>6.245499e-04</td><td>       </td></tr>\n",
       "\t<tr><td>ENSG00000287887</td><td> 3.579139</td><td>1.716630e-03</td><td>       </td></tr>\n",
       "\t<tr><td>ENSG00000287898</td><td> 3.577967</td><td>9.964412e-03</td><td>       </td></tr>\n",
       "\t<tr><td>ENSG00000287899</td><td> 2.966998</td><td>6.134898e-04</td><td>       </td></tr>\n",
       "\t<tr><td>ENSG00000287900</td><td> 4.177529</td><td>1.881429e-06</td><td>       </td></tr>\n",
       "\t<tr><td>ENSG00000287901</td><td> 2.256452</td><td>1.198140e-03</td><td>       </td></tr>\n",
       "\t<tr><td>ENSG00000287913</td><td> 2.935816</td><td>1.260149e-02</td><td>       </td></tr>\n",
       "\t<tr><td>ENSG00000287923</td><td> 2.890514</td><td>1.169877e-02</td><td>       </td></tr>\n",
       "\t<tr><td>ENSG00000287945</td><td>-1.970837</td><td>4.347253e-02</td><td>       </td></tr>\n",
       "\t<tr><td>ENSG00000287946</td><td> 2.506229</td><td>4.712007e-02</td><td>       </td></tr>\n",
       "\t<tr><td>ENSG00000287963</td><td>-2.465417</td><td>6.209511e-05</td><td>       </td></tr>\n",
       "\t<tr><td>ENSG00000287979</td><td> 1.722960</td><td>3.203485e-04</td><td>       </td></tr>\n",
       "\t<tr><td>ENSG00000287996</td><td> 3.249567</td><td>1.666688e-03</td><td>       </td></tr>\n",
       "\t<tr><td>ENSG00000288004</td><td> 2.689127</td><td>1.273936e-02</td><td>       </td></tr>\n",
       "\t<tr><td>ENSG00000288011</td><td> 4.528986</td><td>1.630010e-06</td><td>       </td></tr>\n",
       "\t<tr><td>ENSG00000288018</td><td> 1.751495</td><td>6.059852e-09</td><td>       </td></tr>\n",
       "\t<tr><td>ENSG00000288029</td><td> 4.192971</td><td>5.044360e-04</td><td>       </td></tr>\n",
       "\t<tr><td>ENSG00000288046</td><td> 3.358857</td><td>2.061833e-05</td><td>       </td></tr>\n",
       "\t<tr><td>ENSG00000288068</td><td> 1.844520</td><td>3.014890e-05</td><td>       </td></tr>\n",
       "\t<tr><td>ENSG00000288073</td><td>-3.001874</td><td>2.762255e-06</td><td>       </td></tr>\n",
       "\t<tr><td>ENSG00000288076</td><td>-1.882020</td><td>3.528529e-04</td><td>       </td></tr>\n",
       "\t<tr><td>ENSG00000288077</td><td>-2.016672</td><td>4.233078e-02</td><td>       </td></tr>\n",
       "\t<tr><td>ENSG00000288079</td><td>-1.652164</td><td>8.799479e-03</td><td>       </td></tr>\n",
       "\t<tr><td>ENSG00000288086</td><td> 2.678990</td><td>4.062568e-02</td><td>       </td></tr>\n",
       "\t<tr><td>ENSG00000288091</td><td> 3.343057</td><td>1.712519e-02</td><td>       </td></tr>\n",
       "\t<tr><td>ENSG00000288100</td><td> 4.877521</td><td>1.574210e-04</td><td>       </td></tr>\n",
       "\t<tr><td>ENSG00000288105</td><td> 4.517112</td><td>9.530196e-03</td><td>       </td></tr>\n",
       "\t<tr><td>ENSG00000288108</td><td> 1.548025</td><td>6.029857e-04</td><td>       </td></tr>\n",
       "\t<tr><td>ENSG00000288235</td><td>-1.610026</td><td>3.309201e-02</td><td>FAM106C</td></tr>\n",
       "\t<tr><td>ENSG00000288520</td><td> 3.263966</td><td>9.363160e-09</td><td>       </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 4367  4\n",
       "\\begin{tabular}{llll}\n",
       " GeneID & log2FoldChange & padj & hgnc\\_symbol\\\\\n",
       " <chr> & <dbl> & <dbl> & <chr>\\\\\n",
       "\\hline\n",
       "\t ENSG00000000971 &  2.000816 & 2.186195e-07 & CFH    \\\\\n",
       "\t ENSG00000002726 & -2.576799 & 5.349419e-03 & AOC1   \\\\\n",
       "\t ENSG00000002745 &  3.088030 & 1.045158e-06 & WNT16  \\\\\n",
       "\t ENSG00000003436 &  1.910179 & 6.260387e-10 & TFPI   \\\\\n",
       "\t ENSG00000003989 &  3.866093 & 2.657544e-24 & SLC7A2 \\\\\n",
       "\t ENSG00000004846 &  1.632194 & 3.167950e-02 & ABCB5  \\\\\n",
       "\t ENSG00000005020 &  2.119249 & 2.514852e-11 & SKAP2  \\\\\n",
       "\t ENSG00000005381 &  2.904284 & 1.020554e-03 & MPO    \\\\\n",
       "\t ENSG00000005513 &  3.605927 & 2.937395e-05 & SOX8   \\\\\n",
       "\t ENSG00000006042 & -2.766673 & 3.800028e-08 & TMEM98 \\\\\n",
       "\t ENSG00000006071 &  3.675937 & 2.870305e-15 & ABCC8  \\\\\n",
       "\t ENSG00000006377 &  5.108520 & 4.016659e-07 & DLX6   \\\\\n",
       "\t ENSG00000006611 &  1.834305 & 1.744765e-02 & USH1C  \\\\\n",
       "\t ENSG00000007038 &  1.535662 & 1.595898e-03 & PRSS21 \\\\\n",
       "\t ENSG00000007264 &  3.243889 & 4.458763e-12 & MATK   \\\\\n",
       "\t ENSG00000007314 &  3.322588 & 5.280402e-08 & SCN4A  \\\\\n",
       "\t ENSG00000007350 &  3.841916 & 3.806733e-03 & TKTL1  \\\\\n",
       "\t ENSG00000008517 & -1.759436 & 2.298351e-07 & IL32   \\\\\n",
       "\t ENSG00000009709 &  3.863516 & 2.236451e-04 & PAX7   \\\\\n",
       "\t ENSG00000010932 &  1.880446 & 1.680261e-05 & FMO1   \\\\\n",
       "\t ENSG00000011083 &  2.498799 & 1.310032e-02 & SLC6A7 \\\\\n",
       "\t ENSG00000011677 & -3.059954 & 3.448751e-10 & GABRA3 \\\\\n",
       "\t ENSG00000012171 &  1.570285 & 3.656304e-07 & SEMA3B \\\\\n",
       "\t ENSG00000012223 & -1.656888 & 2.354533e-03 & LTF    \\\\\n",
       "\t ENSG00000012504 &  4.332383 & 8.178697e-05 & NR1H4  \\\\\n",
       "\t ENSG00000012779 & -2.172160 & 1.932958e-03 & ALOX5  \\\\\n",
       "\t ENSG00000012817 &  3.321876 & 1.024097e-02 & KDM5D  \\\\\n",
       "\t ENSG00000013293 &  2.238593 & 1.155544e-02 & SLC7A14\\\\\n",
       "\t ENSG00000017427 & -2.633200 & 5.468134e-05 & IGF1   \\\\\n",
       "\t ENSG00000018236 &  1.634947 & 6.128467e-08 & CNTN1  \\\\\n",
       "\t  &  &  & \\\\\n",
       "\t ENSG00000287886 &  2.400793 & 6.245499e-04 &        \\\\\n",
       "\t ENSG00000287887 &  3.579139 & 1.716630e-03 &        \\\\\n",
       "\t ENSG00000287898 &  3.577967 & 9.964412e-03 &        \\\\\n",
       "\t ENSG00000287899 &  2.966998 & 6.134898e-04 &        \\\\\n",
       "\t ENSG00000287900 &  4.177529 & 1.881429e-06 &        \\\\\n",
       "\t ENSG00000287901 &  2.256452 & 1.198140e-03 &        \\\\\n",
       "\t ENSG00000287913 &  2.935816 & 1.260149e-02 &        \\\\\n",
       "\t ENSG00000287923 &  2.890514 & 1.169877e-02 &        \\\\\n",
       "\t ENSG00000287945 & -1.970837 & 4.347253e-02 &        \\\\\n",
       "\t ENSG00000287946 &  2.506229 & 4.712007e-02 &        \\\\\n",
       "\t ENSG00000287963 & -2.465417 & 6.209511e-05 &        \\\\\n",
       "\t ENSG00000287979 &  1.722960 & 3.203485e-04 &        \\\\\n",
       "\t ENSG00000287996 &  3.249567 & 1.666688e-03 &        \\\\\n",
       "\t ENSG00000288004 &  2.689127 & 1.273936e-02 &        \\\\\n",
       "\t ENSG00000288011 &  4.528986 & 1.630010e-06 &        \\\\\n",
       "\t ENSG00000288018 &  1.751495 & 6.059852e-09 &        \\\\\n",
       "\t ENSG00000288029 &  4.192971 & 5.044360e-04 &        \\\\\n",
       "\t ENSG00000288046 &  3.358857 & 2.061833e-05 &        \\\\\n",
       "\t ENSG00000288068 &  1.844520 & 3.014890e-05 &        \\\\\n",
       "\t ENSG00000288073 & -3.001874 & 2.762255e-06 &        \\\\\n",
       "\t ENSG00000288076 & -1.882020 & 3.528529e-04 &        \\\\\n",
       "\t ENSG00000288077 & -2.016672 & 4.233078e-02 &        \\\\\n",
       "\t ENSG00000288079 & -1.652164 & 8.799479e-03 &        \\\\\n",
       "\t ENSG00000288086 &  2.678990 & 4.062568e-02 &        \\\\\n",
       "\t ENSG00000288091 &  3.343057 & 1.712519e-02 &        \\\\\n",
       "\t ENSG00000288100 &  4.877521 & 1.574210e-04 &        \\\\\n",
       "\t ENSG00000288105 &  4.517112 & 9.530196e-03 &        \\\\\n",
       "\t ENSG00000288108 &  1.548025 & 6.029857e-04 &        \\\\\n",
       "\t ENSG00000288235 & -1.610026 & 3.309201e-02 & FAM106C\\\\\n",
       "\t ENSG00000288520 &  3.263966 & 9.363160e-09 &        \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 4367  4\n",
       "\n",
       "| GeneID &lt;chr&gt; | log2FoldChange &lt;dbl&gt; | padj &lt;dbl&gt; | hgnc_symbol &lt;chr&gt; |\n",
       "|---|---|---|---|\n",
       "| ENSG00000000971 |  2.000816 | 2.186195e-07 | CFH     |\n",
       "| ENSG00000002726 | -2.576799 | 5.349419e-03 | AOC1    |\n",
       "| ENSG00000002745 |  3.088030 | 1.045158e-06 | WNT16   |\n",
       "| ENSG00000003436 |  1.910179 | 6.260387e-10 | TFPI    |\n",
       "| ENSG00000003989 |  3.866093 | 2.657544e-24 | SLC7A2  |\n",
       "| ENSG00000004846 |  1.632194 | 3.167950e-02 | ABCB5   |\n",
       "| ENSG00000005020 |  2.119249 | 2.514852e-11 | SKAP2   |\n",
       "| ENSG00000005381 |  2.904284 | 1.020554e-03 | MPO     |\n",
       "| ENSG00000005513 |  3.605927 | 2.937395e-05 | SOX8    |\n",
       "| ENSG00000006042 | -2.766673 | 3.800028e-08 | TMEM98  |\n",
       "| ENSG00000006071 |  3.675937 | 2.870305e-15 | ABCC8   |\n",
       "| ENSG00000006377 |  5.108520 | 4.016659e-07 | DLX6    |\n",
       "| ENSG00000006611 |  1.834305 | 1.744765e-02 | USH1C   |\n",
       "| ENSG00000007038 |  1.535662 | 1.595898e-03 | PRSS21  |\n",
       "| ENSG00000007264 |  3.243889 | 4.458763e-12 | MATK    |\n",
       "| ENSG00000007314 |  3.322588 | 5.280402e-08 | SCN4A   |\n",
       "| ENSG00000007350 |  3.841916 | 3.806733e-03 | TKTL1   |\n",
       "| ENSG00000008517 | -1.759436 | 2.298351e-07 | IL32    |\n",
       "| ENSG00000009709 |  3.863516 | 2.236451e-04 | PAX7    |\n",
       "| ENSG00000010932 |  1.880446 | 1.680261e-05 | FMO1    |\n",
       "| ENSG00000011083 |  2.498799 | 1.310032e-02 | SLC6A7  |\n",
       "| ENSG00000011677 | -3.059954 | 3.448751e-10 | GABRA3  |\n",
       "| ENSG00000012171 |  1.570285 | 3.656304e-07 | SEMA3B  |\n",
       "| ENSG00000012223 | -1.656888 | 2.354533e-03 | LTF     |\n",
       "| ENSG00000012504 |  4.332383 | 8.178697e-05 | NR1H4   |\n",
       "| ENSG00000012779 | -2.172160 | 1.932958e-03 | ALOX5   |\n",
       "| ENSG00000012817 |  3.321876 | 1.024097e-02 | KDM5D   |\n",
       "| ENSG00000013293 |  2.238593 | 1.155544e-02 | SLC7A14 |\n",
       "| ENSG00000017427 | -2.633200 | 5.468134e-05 | IGF1    |\n",
       "| ENSG00000018236 |  1.634947 | 6.128467e-08 | CNTN1   |\n",
       "|  |  |  |  |\n",
       "| ENSG00000287886 |  2.400793 | 6.245499e-04 | <!----> |\n",
       "| ENSG00000287887 |  3.579139 | 1.716630e-03 | <!----> |\n",
       "| ENSG00000287898 |  3.577967 | 9.964412e-03 | <!----> |\n",
       "| ENSG00000287899 |  2.966998 | 6.134898e-04 | <!----> |\n",
       "| ENSG00000287900 |  4.177529 | 1.881429e-06 | <!----> |\n",
       "| ENSG00000287901 |  2.256452 | 1.198140e-03 | <!----> |\n",
       "| ENSG00000287913 |  2.935816 | 1.260149e-02 | <!----> |\n",
       "| ENSG00000287923 |  2.890514 | 1.169877e-02 | <!----> |\n",
       "| ENSG00000287945 | -1.970837 | 4.347253e-02 | <!----> |\n",
       "| ENSG00000287946 |  2.506229 | 4.712007e-02 | <!----> |\n",
       "| ENSG00000287963 | -2.465417 | 6.209511e-05 | <!----> |\n",
       "| ENSG00000287979 |  1.722960 | 3.203485e-04 | <!----> |\n",
       "| ENSG00000287996 |  3.249567 | 1.666688e-03 | <!----> |\n",
       "| ENSG00000288004 |  2.689127 | 1.273936e-02 | <!----> |\n",
       "| ENSG00000288011 |  4.528986 | 1.630010e-06 | <!----> |\n",
       "| ENSG00000288018 |  1.751495 | 6.059852e-09 | <!----> |\n",
       "| ENSG00000288029 |  4.192971 | 5.044360e-04 | <!----> |\n",
       "| ENSG00000288046 |  3.358857 | 2.061833e-05 | <!----> |\n",
       "| ENSG00000288068 |  1.844520 | 3.014890e-05 | <!----> |\n",
       "| ENSG00000288073 | -3.001874 | 2.762255e-06 | <!----> |\n",
       "| ENSG00000288076 | -1.882020 | 3.528529e-04 | <!----> |\n",
       "| ENSG00000288077 | -2.016672 | 4.233078e-02 | <!----> |\n",
       "| ENSG00000288079 | -1.652164 | 8.799479e-03 | <!----> |\n",
       "| ENSG00000288086 |  2.678990 | 4.062568e-02 | <!----> |\n",
       "| ENSG00000288091 |  3.343057 | 1.712519e-02 | <!----> |\n",
       "| ENSG00000288100 |  4.877521 | 1.574210e-04 | <!----> |\n",
       "| ENSG00000288105 |  4.517112 | 9.530196e-03 | <!----> |\n",
       "| ENSG00000288108 |  1.548025 | 6.029857e-04 | <!----> |\n",
       "| ENSG00000288235 | -1.610026 | 3.309201e-02 | FAM106C |\n",
       "| ENSG00000288520 |  3.263966 | 9.363160e-09 | <!----> |\n",
       "\n"
      ],
      "text/plain": [
       "       GeneID          log2FoldChange padj         hgnc_symbol\n",
       "1      ENSG00000000971  2.000816      2.186195e-07 CFH        \n",
       "2      ENSG00000002726 -2.576799      5.349419e-03 AOC1       \n",
       "3      ENSG00000002745  3.088030      1.045158e-06 WNT16      \n",
       "4      ENSG00000003436  1.910179      6.260387e-10 TFPI       \n",
       "5      ENSG00000003989  3.866093      2.657544e-24 SLC7A2     \n",
       "6      ENSG00000004846  1.632194      3.167950e-02 ABCB5      \n",
       "7      ENSG00000005020  2.119249      2.514852e-11 SKAP2      \n",
       "8      ENSG00000005381  2.904284      1.020554e-03 MPO        \n",
       "9      ENSG00000005513  3.605927      2.937395e-05 SOX8       \n",
       "10     ENSG00000006042 -2.766673      3.800028e-08 TMEM98     \n",
       "11     ENSG00000006071  3.675937      2.870305e-15 ABCC8      \n",
       "12     ENSG00000006377  5.108520      4.016659e-07 DLX6       \n",
       "13     ENSG00000006611  1.834305      1.744765e-02 USH1C      \n",
       "14     ENSG00000007038  1.535662      1.595898e-03 PRSS21     \n",
       "15     ENSG00000007264  3.243889      4.458763e-12 MATK       \n",
       "16     ENSG00000007314  3.322588      5.280402e-08 SCN4A      \n",
       "17     ENSG00000007350  3.841916      3.806733e-03 TKTL1      \n",
       "18     ENSG00000008517 -1.759436      2.298351e-07 IL32       \n",
       "19     ENSG00000009709  3.863516      2.236451e-04 PAX7       \n",
       "20     ENSG00000010932  1.880446      1.680261e-05 FMO1       \n",
       "21     ENSG00000011083  2.498799      1.310032e-02 SLC6A7     \n",
       "22     ENSG00000011677 -3.059954      3.448751e-10 GABRA3     \n",
       "23     ENSG00000012171  1.570285      3.656304e-07 SEMA3B     \n",
       "24     ENSG00000012223 -1.656888      2.354533e-03 LTF        \n",
       "25     ENSG00000012504  4.332383      8.178697e-05 NR1H4      \n",
       "26     ENSG00000012779 -2.172160      1.932958e-03 ALOX5      \n",
       "27     ENSG00000012817  3.321876      1.024097e-02 KDM5D      \n",
       "28     ENSG00000013293  2.238593      1.155544e-02 SLC7A14    \n",
       "29     ENSG00000017427 -2.633200      5.468134e-05 IGF1       \n",
       "30     ENSG00000018236  1.634947      6.128467e-08 CNTN1      \n",
       "<U+22EE> <U+22EE>        <U+22EE>       <U+22EE>     <U+22EE>   \n",
       "4338   ENSG00000287886  2.400793      6.245499e-04            \n",
       "4339   ENSG00000287887  3.579139      1.716630e-03            \n",
       "4340   ENSG00000287898  3.577967      9.964412e-03            \n",
       "4341   ENSG00000287899  2.966998      6.134898e-04            \n",
       "4342   ENSG00000287900  4.177529      1.881429e-06            \n",
       "4343   ENSG00000287901  2.256452      1.198140e-03            \n",
       "4344   ENSG00000287913  2.935816      1.260149e-02            \n",
       "4345   ENSG00000287923  2.890514      1.169877e-02            \n",
       "4346   ENSG00000287945 -1.970837      4.347253e-02            \n",
       "4347   ENSG00000287946  2.506229      4.712007e-02            \n",
       "4348   ENSG00000287963 -2.465417      6.209511e-05            \n",
       "4349   ENSG00000287979  1.722960      3.203485e-04            \n",
       "4350   ENSG00000287996  3.249567      1.666688e-03            \n",
       "4351   ENSG00000288004  2.689127      1.273936e-02            \n",
       "4352   ENSG00000288011  4.528986      1.630010e-06            \n",
       "4353   ENSG00000288018  1.751495      6.059852e-09            \n",
       "4354   ENSG00000288029  4.192971      5.044360e-04            \n",
       "4355   ENSG00000288046  3.358857      2.061833e-05            \n",
       "4356   ENSG00000288068  1.844520      3.014890e-05            \n",
       "4357   ENSG00000288073 -3.001874      2.762255e-06            \n",
       "4358   ENSG00000288076 -1.882020      3.528529e-04            \n",
       "4359   ENSG00000288077 -2.016672      4.233078e-02            \n",
       "4360   ENSG00000288079 -1.652164      8.799479e-03            \n",
       "4361   ENSG00000288086  2.678990      4.062568e-02            \n",
       "4362   ENSG00000288091  3.343057      1.712519e-02            \n",
       "4363   ENSG00000288100  4.877521      1.574210e-04            \n",
       "4364   ENSG00000288105  4.517112      9.530196e-03            \n",
       "4365   ENSG00000288108  1.548025      6.029857e-04            \n",
       "4366   ENSG00000288235 -1.610026      3.309201e-02 FAM106C    \n",
       "4367   ENSG00000288520  3.263966      9.363160e-09            "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#add gene symbols\n",
    "prePathfindR_df$GeneID <- prePathfindR_df$GeneID %>% \n",
    "  tools::file_path_sans_ext()\n",
    "mart <- useDataset(\"hsapiens_gene_ensembl\", useMart(\"ensembl\"))\n",
    "genes <- prePathfindR_df$GeneID\n",
    "G_list <- getBM(filters= \"ensembl_gene_id\",\n",
    "                attributes= c(\"ensembl_gene_id\",\"hgnc_symbol\"),\n",
    "                values=genes,mart= mart,\n",
    "               useCache = FALSE)\n",
    "prePathfindR_df <- merge(prePathfindR_df,\n",
    "                               G_list,\n",
    "                               by.x=\"GeneID\",\n",
    "                               by.y=\"ensembl_gene_id\")\n",
    "prePathfindR_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "e6f69060",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 4367  3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>hgnc_symbol</th><th scope=col>log2FoldChange</th><th scope=col>padj</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>CFH    </td><td> 2.000816</td><td>2.186195e-07</td></tr>\n",
       "\t<tr><td>AOC1   </td><td>-2.576799</td><td>5.349419e-03</td></tr>\n",
       "\t<tr><td>WNT16  </td><td> 3.088030</td><td>1.045158e-06</td></tr>\n",
       "\t<tr><td>TFPI   </td><td> 1.910179</td><td>6.260387e-10</td></tr>\n",
       "\t<tr><td>SLC7A2 </td><td> 3.866093</td><td>2.657544e-24</td></tr>\n",
       "\t<tr><td>ABCB5  </td><td> 1.632194</td><td>3.167950e-02</td></tr>\n",
       "\t<tr><td>SKAP2  </td><td> 2.119249</td><td>2.514852e-11</td></tr>\n",
       "\t<tr><td>MPO    </td><td> 2.904284</td><td>1.020554e-03</td></tr>\n",
       "\t<tr><td>SOX8   </td><td> 3.605927</td><td>2.937395e-05</td></tr>\n",
       "\t<tr><td>TMEM98 </td><td>-2.766673</td><td>3.800028e-08</td></tr>\n",
       "\t<tr><td>ABCC8  </td><td> 3.675937</td><td>2.870305e-15</td></tr>\n",
       "\t<tr><td>DLX6   </td><td> 5.108520</td><td>4.016659e-07</td></tr>\n",
       "\t<tr><td>USH1C  </td><td> 1.834305</td><td>1.744765e-02</td></tr>\n",
       "\t<tr><td>PRSS21 </td><td> 1.535662</td><td>1.595898e-03</td></tr>\n",
       "\t<tr><td>MATK   </td><td> 3.243889</td><td>4.458763e-12</td></tr>\n",
       "\t<tr><td>SCN4A  </td><td> 3.322588</td><td>5.280402e-08</td></tr>\n",
       "\t<tr><td>TKTL1  </td><td> 3.841916</td><td>3.806733e-03</td></tr>\n",
       "\t<tr><td>IL32   </td><td>-1.759436</td><td>2.298351e-07</td></tr>\n",
       "\t<tr><td>PAX7   </td><td> 3.863516</td><td>2.236451e-04</td></tr>\n",
       "\t<tr><td>FMO1   </td><td> 1.880446</td><td>1.680261e-05</td></tr>\n",
       "\t<tr><td>SLC6A7 </td><td> 2.498799</td><td>1.310032e-02</td></tr>\n",
       "\t<tr><td>GABRA3 </td><td>-3.059954</td><td>3.448751e-10</td></tr>\n",
       "\t<tr><td>SEMA3B </td><td> 1.570285</td><td>3.656304e-07</td></tr>\n",
       "\t<tr><td>LTF    </td><td>-1.656888</td><td>2.354533e-03</td></tr>\n",
       "\t<tr><td>NR1H4  </td><td> 4.332383</td><td>8.178697e-05</td></tr>\n",
       "\t<tr><td>ALOX5  </td><td>-2.172160</td><td>1.932958e-03</td></tr>\n",
       "\t<tr><td>KDM5D  </td><td> 3.321876</td><td>1.024097e-02</td></tr>\n",
       "\t<tr><td>SLC7A14</td><td> 2.238593</td><td>1.155544e-02</td></tr>\n",
       "\t<tr><td>IGF1   </td><td>-2.633200</td><td>5.468134e-05</td></tr>\n",
       "\t<tr><td>CNTN1  </td><td> 1.634947</td><td>6.128467e-08</td></tr>\n",
       "\t<tr><td></td><td></td><td></td></tr>\n",
       "\t<tr><td>       </td><td> 2.400793</td><td>6.245499e-04</td></tr>\n",
       "\t<tr><td>       </td><td> 3.579139</td><td>1.716630e-03</td></tr>\n",
       "\t<tr><td>       </td><td> 3.577967</td><td>9.964412e-03</td></tr>\n",
       "\t<tr><td>       </td><td> 2.966998</td><td>6.134898e-04</td></tr>\n",
       "\t<tr><td>       </td><td> 4.177529</td><td>1.881429e-06</td></tr>\n",
       "\t<tr><td>       </td><td> 2.256452</td><td>1.198140e-03</td></tr>\n",
       "\t<tr><td>       </td><td> 2.935816</td><td>1.260149e-02</td></tr>\n",
       "\t<tr><td>       </td><td> 2.890514</td><td>1.169877e-02</td></tr>\n",
       "\t<tr><td>       </td><td>-1.970837</td><td>4.347253e-02</td></tr>\n",
       "\t<tr><td>       </td><td> 2.506229</td><td>4.712007e-02</td></tr>\n",
       "\t<tr><td>       </td><td>-2.465417</td><td>6.209511e-05</td></tr>\n",
       "\t<tr><td>       </td><td> 1.722960</td><td>3.203485e-04</td></tr>\n",
       "\t<tr><td>       </td><td> 3.249567</td><td>1.666688e-03</td></tr>\n",
       "\t<tr><td>       </td><td> 2.689127</td><td>1.273936e-02</td></tr>\n",
       "\t<tr><td>       </td><td> 4.528986</td><td>1.630010e-06</td></tr>\n",
       "\t<tr><td>       </td><td> 1.751495</td><td>6.059852e-09</td></tr>\n",
       "\t<tr><td>       </td><td> 4.192971</td><td>5.044360e-04</td></tr>\n",
       "\t<tr><td>       </td><td> 3.358857</td><td>2.061833e-05</td></tr>\n",
       "\t<tr><td>       </td><td> 1.844520</td><td>3.014890e-05</td></tr>\n",
       "\t<tr><td>       </td><td>-3.001874</td><td>2.762255e-06</td></tr>\n",
       "\t<tr><td>       </td><td>-1.882020</td><td>3.528529e-04</td></tr>\n",
       "\t<tr><td>       </td><td>-2.016672</td><td>4.233078e-02</td></tr>\n",
       "\t<tr><td>       </td><td>-1.652164</td><td>8.799479e-03</td></tr>\n",
       "\t<tr><td>       </td><td> 2.678990</td><td>4.062568e-02</td></tr>\n",
       "\t<tr><td>       </td><td> 3.343057</td><td>1.712519e-02</td></tr>\n",
       "\t<tr><td>       </td><td> 4.877521</td><td>1.574210e-04</td></tr>\n",
       "\t<tr><td>       </td><td> 4.517112</td><td>9.530196e-03</td></tr>\n",
       "\t<tr><td>       </td><td> 1.548025</td><td>6.029857e-04</td></tr>\n",
       "\t<tr><td>FAM106C</td><td>-1.610026</td><td>3.309201e-02</td></tr>\n",
       "\t<tr><td>       </td><td> 3.263966</td><td>9.363160e-09</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 4367  3\n",
       "\\begin{tabular}{lll}\n",
       " hgnc\\_symbol & log2FoldChange & padj\\\\\n",
       " <chr> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t CFH     &  2.000816 & 2.186195e-07\\\\\n",
       "\t AOC1    & -2.576799 & 5.349419e-03\\\\\n",
       "\t WNT16   &  3.088030 & 1.045158e-06\\\\\n",
       "\t TFPI    &  1.910179 & 6.260387e-10\\\\\n",
       "\t SLC7A2  &  3.866093 & 2.657544e-24\\\\\n",
       "\t ABCB5   &  1.632194 & 3.167950e-02\\\\\n",
       "\t SKAP2   &  2.119249 & 2.514852e-11\\\\\n",
       "\t MPO     &  2.904284 & 1.020554e-03\\\\\n",
       "\t SOX8    &  3.605927 & 2.937395e-05\\\\\n",
       "\t TMEM98  & -2.766673 & 3.800028e-08\\\\\n",
       "\t ABCC8   &  3.675937 & 2.870305e-15\\\\\n",
       "\t DLX6    &  5.108520 & 4.016659e-07\\\\\n",
       "\t USH1C   &  1.834305 & 1.744765e-02\\\\\n",
       "\t PRSS21  &  1.535662 & 1.595898e-03\\\\\n",
       "\t MATK    &  3.243889 & 4.458763e-12\\\\\n",
       "\t SCN4A   &  3.322588 & 5.280402e-08\\\\\n",
       "\t TKTL1   &  3.841916 & 3.806733e-03\\\\\n",
       "\t IL32    & -1.759436 & 2.298351e-07\\\\\n",
       "\t PAX7    &  3.863516 & 2.236451e-04\\\\\n",
       "\t FMO1    &  1.880446 & 1.680261e-05\\\\\n",
       "\t SLC6A7  &  2.498799 & 1.310032e-02\\\\\n",
       "\t GABRA3  & -3.059954 & 3.448751e-10\\\\\n",
       "\t SEMA3B  &  1.570285 & 3.656304e-07\\\\\n",
       "\t LTF     & -1.656888 & 2.354533e-03\\\\\n",
       "\t NR1H4   &  4.332383 & 8.178697e-05\\\\\n",
       "\t ALOX5   & -2.172160 & 1.932958e-03\\\\\n",
       "\t KDM5D   &  3.321876 & 1.024097e-02\\\\\n",
       "\t SLC7A14 &  2.238593 & 1.155544e-02\\\\\n",
       "\t IGF1    & -2.633200 & 5.468134e-05\\\\\n",
       "\t CNTN1   &  1.634947 & 6.128467e-08\\\\\n",
       "\t  &  & \\\\\n",
       "\t         &  2.400793 & 6.245499e-04\\\\\n",
       "\t         &  3.579139 & 1.716630e-03\\\\\n",
       "\t         &  3.577967 & 9.964412e-03\\\\\n",
       "\t         &  2.966998 & 6.134898e-04\\\\\n",
       "\t         &  4.177529 & 1.881429e-06\\\\\n",
       "\t         &  2.256452 & 1.198140e-03\\\\\n",
       "\t         &  2.935816 & 1.260149e-02\\\\\n",
       "\t         &  2.890514 & 1.169877e-02\\\\\n",
       "\t         & -1.970837 & 4.347253e-02\\\\\n",
       "\t         &  2.506229 & 4.712007e-02\\\\\n",
       "\t         & -2.465417 & 6.209511e-05\\\\\n",
       "\t         &  1.722960 & 3.203485e-04\\\\\n",
       "\t         &  3.249567 & 1.666688e-03\\\\\n",
       "\t         &  2.689127 & 1.273936e-02\\\\\n",
       "\t         &  4.528986 & 1.630010e-06\\\\\n",
       "\t         &  1.751495 & 6.059852e-09\\\\\n",
       "\t         &  4.192971 & 5.044360e-04\\\\\n",
       "\t         &  3.358857 & 2.061833e-05\\\\\n",
       "\t         &  1.844520 & 3.014890e-05\\\\\n",
       "\t         & -3.001874 & 2.762255e-06\\\\\n",
       "\t         & -1.882020 & 3.528529e-04\\\\\n",
       "\t         & -2.016672 & 4.233078e-02\\\\\n",
       "\t         & -1.652164 & 8.799479e-03\\\\\n",
       "\t         &  2.678990 & 4.062568e-02\\\\\n",
       "\t         &  3.343057 & 1.712519e-02\\\\\n",
       "\t         &  4.877521 & 1.574210e-04\\\\\n",
       "\t         &  4.517112 & 9.530196e-03\\\\\n",
       "\t         &  1.548025 & 6.029857e-04\\\\\n",
       "\t FAM106C & -1.610026 & 3.309201e-02\\\\\n",
       "\t         &  3.263966 & 9.363160e-09\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 4367  3\n",
       "\n",
       "| hgnc_symbol &lt;chr&gt; | log2FoldChange &lt;dbl&gt; | padj &lt;dbl&gt; |\n",
       "|---|---|---|\n",
       "| CFH     |  2.000816 | 2.186195e-07 |\n",
       "| AOC1    | -2.576799 | 5.349419e-03 |\n",
       "| WNT16   |  3.088030 | 1.045158e-06 |\n",
       "| TFPI    |  1.910179 | 6.260387e-10 |\n",
       "| SLC7A2  |  3.866093 | 2.657544e-24 |\n",
       "| ABCB5   |  1.632194 | 3.167950e-02 |\n",
       "| SKAP2   |  2.119249 | 2.514852e-11 |\n",
       "| MPO     |  2.904284 | 1.020554e-03 |\n",
       "| SOX8    |  3.605927 | 2.937395e-05 |\n",
       "| TMEM98  | -2.766673 | 3.800028e-08 |\n",
       "| ABCC8   |  3.675937 | 2.870305e-15 |\n",
       "| DLX6    |  5.108520 | 4.016659e-07 |\n",
       "| USH1C   |  1.834305 | 1.744765e-02 |\n",
       "| PRSS21  |  1.535662 | 1.595898e-03 |\n",
       "| MATK    |  3.243889 | 4.458763e-12 |\n",
       "| SCN4A   |  3.322588 | 5.280402e-08 |\n",
       "| TKTL1   |  3.841916 | 3.806733e-03 |\n",
       "| IL32    | -1.759436 | 2.298351e-07 |\n",
       "| PAX7    |  3.863516 | 2.236451e-04 |\n",
       "| FMO1    |  1.880446 | 1.680261e-05 |\n",
       "| SLC6A7  |  2.498799 | 1.310032e-02 |\n",
       "| GABRA3  | -3.059954 | 3.448751e-10 |\n",
       "| SEMA3B  |  1.570285 | 3.656304e-07 |\n",
       "| LTF     | -1.656888 | 2.354533e-03 |\n",
       "| NR1H4   |  4.332383 | 8.178697e-05 |\n",
       "| ALOX5   | -2.172160 | 1.932958e-03 |\n",
       "| KDM5D   |  3.321876 | 1.024097e-02 |\n",
       "| SLC7A14 |  2.238593 | 1.155544e-02 |\n",
       "| IGF1    | -2.633200 | 5.468134e-05 |\n",
       "| CNTN1   |  1.634947 | 6.128467e-08 |\n",
       "|  |  |  |\n",
       "| <!----> |  2.400793 | 6.245499e-04 |\n",
       "| <!----> |  3.579139 | 1.716630e-03 |\n",
       "| <!----> |  3.577967 | 9.964412e-03 |\n",
       "| <!----> |  2.966998 | 6.134898e-04 |\n",
       "| <!----> |  4.177529 | 1.881429e-06 |\n",
       "| <!----> |  2.256452 | 1.198140e-03 |\n",
       "| <!----> |  2.935816 | 1.260149e-02 |\n",
       "| <!----> |  2.890514 | 1.169877e-02 |\n",
       "| <!----> | -1.970837 | 4.347253e-02 |\n",
       "| <!----> |  2.506229 | 4.712007e-02 |\n",
       "| <!----> | -2.465417 | 6.209511e-05 |\n",
       "| <!----> |  1.722960 | 3.203485e-04 |\n",
       "| <!----> |  3.249567 | 1.666688e-03 |\n",
       "| <!----> |  2.689127 | 1.273936e-02 |\n",
       "| <!----> |  4.528986 | 1.630010e-06 |\n",
       "| <!----> |  1.751495 | 6.059852e-09 |\n",
       "| <!----> |  4.192971 | 5.044360e-04 |\n",
       "| <!----> |  3.358857 | 2.061833e-05 |\n",
       "| <!----> |  1.844520 | 3.014890e-05 |\n",
       "| <!----> | -3.001874 | 2.762255e-06 |\n",
       "| <!----> | -1.882020 | 3.528529e-04 |\n",
       "| <!----> | -2.016672 | 4.233078e-02 |\n",
       "| <!----> | -1.652164 | 8.799479e-03 |\n",
       "| <!----> |  2.678990 | 4.062568e-02 |\n",
       "| <!----> |  3.343057 | 1.712519e-02 |\n",
       "| <!----> |  4.877521 | 1.574210e-04 |\n",
       "| <!----> |  4.517112 | 9.530196e-03 |\n",
       "| <!----> |  1.548025 | 6.029857e-04 |\n",
       "| FAM106C | -1.610026 | 3.309201e-02 |\n",
       "| <!----> |  3.263966 | 9.363160e-09 |\n",
       "\n"
      ],
      "text/plain": [
       "       hgnc_symbol log2FoldChange padj        \n",
       "1      CFH          2.000816      2.186195e-07\n",
       "2      AOC1        -2.576799      5.349419e-03\n",
       "3      WNT16        3.088030      1.045158e-06\n",
       "4      TFPI         1.910179      6.260387e-10\n",
       "5      SLC7A2       3.866093      2.657544e-24\n",
       "6      ABCB5        1.632194      3.167950e-02\n",
       "7      SKAP2        2.119249      2.514852e-11\n",
       "8      MPO          2.904284      1.020554e-03\n",
       "9      SOX8         3.605927      2.937395e-05\n",
       "10     TMEM98      -2.766673      3.800028e-08\n",
       "11     ABCC8        3.675937      2.870305e-15\n",
       "12     DLX6         5.108520      4.016659e-07\n",
       "13     USH1C        1.834305      1.744765e-02\n",
       "14     PRSS21       1.535662      1.595898e-03\n",
       "15     MATK         3.243889      4.458763e-12\n",
       "16     SCN4A        3.322588      5.280402e-08\n",
       "17     TKTL1        3.841916      3.806733e-03\n",
       "18     IL32        -1.759436      2.298351e-07\n",
       "19     PAX7         3.863516      2.236451e-04\n",
       "20     FMO1         1.880446      1.680261e-05\n",
       "21     SLC6A7       2.498799      1.310032e-02\n",
       "22     GABRA3      -3.059954      3.448751e-10\n",
       "23     SEMA3B       1.570285      3.656304e-07\n",
       "24     LTF         -1.656888      2.354533e-03\n",
       "25     NR1H4        4.332383      8.178697e-05\n",
       "26     ALOX5       -2.172160      1.932958e-03\n",
       "27     KDM5D        3.321876      1.024097e-02\n",
       "28     SLC7A14      2.238593      1.155544e-02\n",
       "29     IGF1        -2.633200      5.468134e-05\n",
       "30     CNTN1        1.634947      6.128467e-08\n",
       "<U+22EE> <U+22EE>    <U+22EE>       <U+22EE>    \n",
       "4338                2.400793      6.245499e-04\n",
       "4339                3.579139      1.716630e-03\n",
       "4340                3.577967      9.964412e-03\n",
       "4341                2.966998      6.134898e-04\n",
       "4342                4.177529      1.881429e-06\n",
       "4343                2.256452      1.198140e-03\n",
       "4344                2.935816      1.260149e-02\n",
       "4345                2.890514      1.169877e-02\n",
       "4346               -1.970837      4.347253e-02\n",
       "4347                2.506229      4.712007e-02\n",
       "4348               -2.465417      6.209511e-05\n",
       "4349                1.722960      3.203485e-04\n",
       "4350                3.249567      1.666688e-03\n",
       "4351                2.689127      1.273936e-02\n",
       "4352                4.528986      1.630010e-06\n",
       "4353                1.751495      6.059852e-09\n",
       "4354                4.192971      5.044360e-04\n",
       "4355                3.358857      2.061833e-05\n",
       "4356                1.844520      3.014890e-05\n",
       "4357               -3.001874      2.762255e-06\n",
       "4358               -1.882020      3.528529e-04\n",
       "4359               -2.016672      4.233078e-02\n",
       "4360               -1.652164      8.799479e-03\n",
       "4361                2.678990      4.062568e-02\n",
       "4362                3.343057      1.712519e-02\n",
       "4363                4.877521      1.574210e-04\n",
       "4364                4.517112      9.530196e-03\n",
       "4365                1.548025      6.029857e-04\n",
       "4366   FAM106C     -1.610026      3.309201e-02\n",
       "4367                3.263966      9.363160e-09"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "prePathfindR_df <- prePathfindR_df %>%\n",
    "                dplyr::select(c(\"hgnc_symbol\", \"log2FoldChange\", \"padj\"))\n",
    "prePathfindR_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "7eb8e5cc",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 4367  3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>hgnc_symbol</th><th scope=col>log2FoldChange</th><th scope=col>padj</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>CFH    </td><td> 2.000816</td><td>2.186195e-07</td></tr>\n",
       "\t<tr><td>AOC1   </td><td>-2.576799</td><td>5.349419e-03</td></tr>\n",
       "\t<tr><td>WNT16  </td><td> 3.088030</td><td>1.045158e-06</td></tr>\n",
       "\t<tr><td>TFPI   </td><td> 1.910179</td><td>6.260387e-10</td></tr>\n",
       "\t<tr><td>SLC7A2 </td><td> 3.866093</td><td>2.657544e-24</td></tr>\n",
       "\t<tr><td>ABCB5  </td><td> 1.632194</td><td>3.167950e-02</td></tr>\n",
       "\t<tr><td>SKAP2  </td><td> 2.119249</td><td>2.514852e-11</td></tr>\n",
       "\t<tr><td>MPO    </td><td> 2.904284</td><td>1.020554e-03</td></tr>\n",
       "\t<tr><td>SOX8   </td><td> 3.605927</td><td>2.937395e-05</td></tr>\n",
       "\t<tr><td>TMEM98 </td><td>-2.766673</td><td>3.800028e-08</td></tr>\n",
       "\t<tr><td>ABCC8  </td><td> 3.675937</td><td>2.870305e-15</td></tr>\n",
       "\t<tr><td>DLX6   </td><td> 5.108520</td><td>4.016659e-07</td></tr>\n",
       "\t<tr><td>USH1C  </td><td> 1.834305</td><td>1.744765e-02</td></tr>\n",
       "\t<tr><td>PRSS21 </td><td> 1.535662</td><td>1.595898e-03</td></tr>\n",
       "\t<tr><td>MATK   </td><td> 3.243889</td><td>4.458763e-12</td></tr>\n",
       "\t<tr><td>SCN4A  </td><td> 3.322588</td><td>5.280402e-08</td></tr>\n",
       "\t<tr><td>TKTL1  </td><td> 3.841916</td><td>3.806733e-03</td></tr>\n",
       "\t<tr><td>IL32   </td><td>-1.759436</td><td>2.298351e-07</td></tr>\n",
       "\t<tr><td>PAX7   </td><td> 3.863516</td><td>2.236451e-04</td></tr>\n",
       "\t<tr><td>FMO1   </td><td> 1.880446</td><td>1.680261e-05</td></tr>\n",
       "\t<tr><td>SLC6A7 </td><td> 2.498799</td><td>1.310032e-02</td></tr>\n",
       "\t<tr><td>GABRA3 </td><td>-3.059954</td><td>3.448751e-10</td></tr>\n",
       "\t<tr><td>SEMA3B </td><td> 1.570285</td><td>3.656304e-07</td></tr>\n",
       "\t<tr><td>LTF    </td><td>-1.656888</td><td>2.354533e-03</td></tr>\n",
       "\t<tr><td>NR1H4  </td><td> 4.332383</td><td>8.178697e-05</td></tr>\n",
       "\t<tr><td>ALOX5  </td><td>-2.172160</td><td>1.932958e-03</td></tr>\n",
       "\t<tr><td>KDM5D  </td><td> 3.321876</td><td>1.024097e-02</td></tr>\n",
       "\t<tr><td>SLC7A14</td><td> 2.238593</td><td>1.155544e-02</td></tr>\n",
       "\t<tr><td>IGF1   </td><td>-2.633200</td><td>5.468134e-05</td></tr>\n",
       "\t<tr><td>CNTN1  </td><td> 1.634947</td><td>6.128467e-08</td></tr>\n",
       "\t<tr><td></td><td></td><td></td></tr>\n",
       "\t<tr><td>       </td><td> 2.400793</td><td>6.245499e-04</td></tr>\n",
       "\t<tr><td>       </td><td> 3.579139</td><td>1.716630e-03</td></tr>\n",
       "\t<tr><td>       </td><td> 3.577967</td><td>9.964412e-03</td></tr>\n",
       "\t<tr><td>       </td><td> 2.966998</td><td>6.134898e-04</td></tr>\n",
       "\t<tr><td>       </td><td> 4.177529</td><td>1.881429e-06</td></tr>\n",
       "\t<tr><td>       </td><td> 2.256452</td><td>1.198140e-03</td></tr>\n",
       "\t<tr><td>       </td><td> 2.935816</td><td>1.260149e-02</td></tr>\n",
       "\t<tr><td>       </td><td> 2.890514</td><td>1.169877e-02</td></tr>\n",
       "\t<tr><td>       </td><td>-1.970837</td><td>4.347253e-02</td></tr>\n",
       "\t<tr><td>       </td><td> 2.506229</td><td>4.712007e-02</td></tr>\n",
       "\t<tr><td>       </td><td>-2.465417</td><td>6.209511e-05</td></tr>\n",
       "\t<tr><td>       </td><td> 1.722960</td><td>3.203485e-04</td></tr>\n",
       "\t<tr><td>       </td><td> 3.249567</td><td>1.666688e-03</td></tr>\n",
       "\t<tr><td>       </td><td> 2.689127</td><td>1.273936e-02</td></tr>\n",
       "\t<tr><td>       </td><td> 4.528986</td><td>1.630010e-06</td></tr>\n",
       "\t<tr><td>       </td><td> 1.751495</td><td>6.059852e-09</td></tr>\n",
       "\t<tr><td>       </td><td> 4.192971</td><td>5.044360e-04</td></tr>\n",
       "\t<tr><td>       </td><td> 3.358857</td><td>2.061833e-05</td></tr>\n",
       "\t<tr><td>       </td><td> 1.844520</td><td>3.014890e-05</td></tr>\n",
       "\t<tr><td>       </td><td>-3.001874</td><td>2.762255e-06</td></tr>\n",
       "\t<tr><td>       </td><td>-1.882020</td><td>3.528529e-04</td></tr>\n",
       "\t<tr><td>       </td><td>-2.016672</td><td>4.233078e-02</td></tr>\n",
       "\t<tr><td>       </td><td>-1.652164</td><td>8.799479e-03</td></tr>\n",
       "\t<tr><td>       </td><td> 2.678990</td><td>4.062568e-02</td></tr>\n",
       "\t<tr><td>       </td><td> 3.343057</td><td>1.712519e-02</td></tr>\n",
       "\t<tr><td>       </td><td> 4.877521</td><td>1.574210e-04</td></tr>\n",
       "\t<tr><td>       </td><td> 4.517112</td><td>9.530196e-03</td></tr>\n",
       "\t<tr><td>       </td><td> 1.548025</td><td>6.029857e-04</td></tr>\n",
       "\t<tr><td>FAM106C</td><td>-1.610026</td><td>3.309201e-02</td></tr>\n",
       "\t<tr><td>       </td><td> 3.263966</td><td>9.363160e-09</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 4367  3\n",
       "\\begin{tabular}{lll}\n",
       " hgnc\\_symbol & log2FoldChange & padj\\\\\n",
       " <chr> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t CFH     &  2.000816 & 2.186195e-07\\\\\n",
       "\t AOC1    & -2.576799 & 5.349419e-03\\\\\n",
       "\t WNT16   &  3.088030 & 1.045158e-06\\\\\n",
       "\t TFPI    &  1.910179 & 6.260387e-10\\\\\n",
       "\t SLC7A2  &  3.866093 & 2.657544e-24\\\\\n",
       "\t ABCB5   &  1.632194 & 3.167950e-02\\\\\n",
       "\t SKAP2   &  2.119249 & 2.514852e-11\\\\\n",
       "\t MPO     &  2.904284 & 1.020554e-03\\\\\n",
       "\t SOX8    &  3.605927 & 2.937395e-05\\\\\n",
       "\t TMEM98  & -2.766673 & 3.800028e-08\\\\\n",
       "\t ABCC8   &  3.675937 & 2.870305e-15\\\\\n",
       "\t DLX6    &  5.108520 & 4.016659e-07\\\\\n",
       "\t USH1C   &  1.834305 & 1.744765e-02\\\\\n",
       "\t PRSS21  &  1.535662 & 1.595898e-03\\\\\n",
       "\t MATK    &  3.243889 & 4.458763e-12\\\\\n",
       "\t SCN4A   &  3.322588 & 5.280402e-08\\\\\n",
       "\t TKTL1   &  3.841916 & 3.806733e-03\\\\\n",
       "\t IL32    & -1.759436 & 2.298351e-07\\\\\n",
       "\t PAX7    &  3.863516 & 2.236451e-04\\\\\n",
       "\t FMO1    &  1.880446 & 1.680261e-05\\\\\n",
       "\t SLC6A7  &  2.498799 & 1.310032e-02\\\\\n",
       "\t GABRA3  & -3.059954 & 3.448751e-10\\\\\n",
       "\t SEMA3B  &  1.570285 & 3.656304e-07\\\\\n",
       "\t LTF     & -1.656888 & 2.354533e-03\\\\\n",
       "\t NR1H4   &  4.332383 & 8.178697e-05\\\\\n",
       "\t ALOX5   & -2.172160 & 1.932958e-03\\\\\n",
       "\t KDM5D   &  3.321876 & 1.024097e-02\\\\\n",
       "\t SLC7A14 &  2.238593 & 1.155544e-02\\\\\n",
       "\t IGF1    & -2.633200 & 5.468134e-05\\\\\n",
       "\t CNTN1   &  1.634947 & 6.128467e-08\\\\\n",
       "\t  &  & \\\\\n",
       "\t         &  2.400793 & 6.245499e-04\\\\\n",
       "\t         &  3.579139 & 1.716630e-03\\\\\n",
       "\t         &  3.577967 & 9.964412e-03\\\\\n",
       "\t         &  2.966998 & 6.134898e-04\\\\\n",
       "\t         &  4.177529 & 1.881429e-06\\\\\n",
       "\t         &  2.256452 & 1.198140e-03\\\\\n",
       "\t         &  2.935816 & 1.260149e-02\\\\\n",
       "\t         &  2.890514 & 1.169877e-02\\\\\n",
       "\t         & -1.970837 & 4.347253e-02\\\\\n",
       "\t         &  2.506229 & 4.712007e-02\\\\\n",
       "\t         & -2.465417 & 6.209511e-05\\\\\n",
       "\t         &  1.722960 & 3.203485e-04\\\\\n",
       "\t         &  3.249567 & 1.666688e-03\\\\\n",
       "\t         &  2.689127 & 1.273936e-02\\\\\n",
       "\t         &  4.528986 & 1.630010e-06\\\\\n",
       "\t         &  1.751495 & 6.059852e-09\\\\\n",
       "\t         &  4.192971 & 5.044360e-04\\\\\n",
       "\t         &  3.358857 & 2.061833e-05\\\\\n",
       "\t         &  1.844520 & 3.014890e-05\\\\\n",
       "\t         & -3.001874 & 2.762255e-06\\\\\n",
       "\t         & -1.882020 & 3.528529e-04\\\\\n",
       "\t         & -2.016672 & 4.233078e-02\\\\\n",
       "\t         & -1.652164 & 8.799479e-03\\\\\n",
       "\t         &  2.678990 & 4.062568e-02\\\\\n",
       "\t         &  3.343057 & 1.712519e-02\\\\\n",
       "\t         &  4.877521 & 1.574210e-04\\\\\n",
       "\t         &  4.517112 & 9.530196e-03\\\\\n",
       "\t         &  1.548025 & 6.029857e-04\\\\\n",
       "\t FAM106C & -1.610026 & 3.309201e-02\\\\\n",
       "\t         &  3.263966 & 9.363160e-09\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 4367  3\n",
       "\n",
       "| hgnc_symbol &lt;chr&gt; | log2FoldChange &lt;dbl&gt; | padj &lt;dbl&gt; |\n",
       "|---|---|---|\n",
       "| CFH     |  2.000816 | 2.186195e-07 |\n",
       "| AOC1    | -2.576799 | 5.349419e-03 |\n",
       "| WNT16   |  3.088030 | 1.045158e-06 |\n",
       "| TFPI    |  1.910179 | 6.260387e-10 |\n",
       "| SLC7A2  |  3.866093 | 2.657544e-24 |\n",
       "| ABCB5   |  1.632194 | 3.167950e-02 |\n",
       "| SKAP2   |  2.119249 | 2.514852e-11 |\n",
       "| MPO     |  2.904284 | 1.020554e-03 |\n",
       "| SOX8    |  3.605927 | 2.937395e-05 |\n",
       "| TMEM98  | -2.766673 | 3.800028e-08 |\n",
       "| ABCC8   |  3.675937 | 2.870305e-15 |\n",
       "| DLX6    |  5.108520 | 4.016659e-07 |\n",
       "| USH1C   |  1.834305 | 1.744765e-02 |\n",
       "| PRSS21  |  1.535662 | 1.595898e-03 |\n",
       "| MATK    |  3.243889 | 4.458763e-12 |\n",
       "| SCN4A   |  3.322588 | 5.280402e-08 |\n",
       "| TKTL1   |  3.841916 | 3.806733e-03 |\n",
       "| IL32    | -1.759436 | 2.298351e-07 |\n",
       "| PAX7    |  3.863516 | 2.236451e-04 |\n",
       "| FMO1    |  1.880446 | 1.680261e-05 |\n",
       "| SLC6A7  |  2.498799 | 1.310032e-02 |\n",
       "| GABRA3  | -3.059954 | 3.448751e-10 |\n",
       "| SEMA3B  |  1.570285 | 3.656304e-07 |\n",
       "| LTF     | -1.656888 | 2.354533e-03 |\n",
       "| NR1H4   |  4.332383 | 8.178697e-05 |\n",
       "| ALOX5   | -2.172160 | 1.932958e-03 |\n",
       "| KDM5D   |  3.321876 | 1.024097e-02 |\n",
       "| SLC7A14 |  2.238593 | 1.155544e-02 |\n",
       "| IGF1    | -2.633200 | 5.468134e-05 |\n",
       "| CNTN1   |  1.634947 | 6.128467e-08 |\n",
       "|  |  |  |\n",
       "| <!----> |  2.400793 | 6.245499e-04 |\n",
       "| <!----> |  3.579139 | 1.716630e-03 |\n",
       "| <!----> |  3.577967 | 9.964412e-03 |\n",
       "| <!----> |  2.966998 | 6.134898e-04 |\n",
       "| <!----> |  4.177529 | 1.881429e-06 |\n",
       "| <!----> |  2.256452 | 1.198140e-03 |\n",
       "| <!----> |  2.935816 | 1.260149e-02 |\n",
       "| <!----> |  2.890514 | 1.169877e-02 |\n",
       "| <!----> | -1.970837 | 4.347253e-02 |\n",
       "| <!----> |  2.506229 | 4.712007e-02 |\n",
       "| <!----> | -2.465417 | 6.209511e-05 |\n",
       "| <!----> |  1.722960 | 3.203485e-04 |\n",
       "| <!----> |  3.249567 | 1.666688e-03 |\n",
       "| <!----> |  2.689127 | 1.273936e-02 |\n",
       "| <!----> |  4.528986 | 1.630010e-06 |\n",
       "| <!----> |  1.751495 | 6.059852e-09 |\n",
       "| <!----> |  4.192971 | 5.044360e-04 |\n",
       "| <!----> |  3.358857 | 2.061833e-05 |\n",
       "| <!----> |  1.844520 | 3.014890e-05 |\n",
       "| <!----> | -3.001874 | 2.762255e-06 |\n",
       "| <!----> | -1.882020 | 3.528529e-04 |\n",
       "| <!----> | -2.016672 | 4.233078e-02 |\n",
       "| <!----> | -1.652164 | 8.799479e-03 |\n",
       "| <!----> |  2.678990 | 4.062568e-02 |\n",
       "| <!----> |  3.343057 | 1.712519e-02 |\n",
       "| <!----> |  4.877521 | 1.574210e-04 |\n",
       "| <!----> |  4.517112 | 9.530196e-03 |\n",
       "| <!----> |  1.548025 | 6.029857e-04 |\n",
       "| FAM106C | -1.610026 | 3.309201e-02 |\n",
       "| <!----> |  3.263966 | 9.363160e-09 |\n",
       "\n"
      ],
      "text/plain": [
       "       hgnc_symbol log2FoldChange padj        \n",
       "1      CFH          2.000816      2.186195e-07\n",
       "2      AOC1        -2.576799      5.349419e-03\n",
       "3      WNT16        3.088030      1.045158e-06\n",
       "4      TFPI         1.910179      6.260387e-10\n",
       "5      SLC7A2       3.866093      2.657544e-24\n",
       "6      ABCB5        1.632194      3.167950e-02\n",
       "7      SKAP2        2.119249      2.514852e-11\n",
       "8      MPO          2.904284      1.020554e-03\n",
       "9      SOX8         3.605927      2.937395e-05\n",
       "10     TMEM98      -2.766673      3.800028e-08\n",
       "11     ABCC8        3.675937      2.870305e-15\n",
       "12     DLX6         5.108520      4.016659e-07\n",
       "13     USH1C        1.834305      1.744765e-02\n",
       "14     PRSS21       1.535662      1.595898e-03\n",
       "15     MATK         3.243889      4.458763e-12\n",
       "16     SCN4A        3.322588      5.280402e-08\n",
       "17     TKTL1        3.841916      3.806733e-03\n",
       "18     IL32        -1.759436      2.298351e-07\n",
       "19     PAX7         3.863516      2.236451e-04\n",
       "20     FMO1         1.880446      1.680261e-05\n",
       "21     SLC6A7       2.498799      1.310032e-02\n",
       "22     GABRA3      -3.059954      3.448751e-10\n",
       "23     SEMA3B       1.570285      3.656304e-07\n",
       "24     LTF         -1.656888      2.354533e-03\n",
       "25     NR1H4        4.332383      8.178697e-05\n",
       "26     ALOX5       -2.172160      1.932958e-03\n",
       "27     KDM5D        3.321876      1.024097e-02\n",
       "28     SLC7A14      2.238593      1.155544e-02\n",
       "29     IGF1        -2.633200      5.468134e-05\n",
       "30     CNTN1        1.634947      6.128467e-08\n",
       "<U+22EE> <U+22EE>    <U+22EE>       <U+22EE>    \n",
       "4338                2.400793      6.245499e-04\n",
       "4339                3.579139      1.716630e-03\n",
       "4340                3.577967      9.964412e-03\n",
       "4341                2.966998      6.134898e-04\n",
       "4342                4.177529      1.881429e-06\n",
       "4343                2.256452      1.198140e-03\n",
       "4344                2.935816      1.260149e-02\n",
       "4345                2.890514      1.169877e-02\n",
       "4346               -1.970837      4.347253e-02\n",
       "4347                2.506229      4.712007e-02\n",
       "4348               -2.465417      6.209511e-05\n",
       "4349                1.722960      3.203485e-04\n",
       "4350                3.249567      1.666688e-03\n",
       "4351                2.689127      1.273936e-02\n",
       "4352                4.528986      1.630010e-06\n",
       "4353                1.751495      6.059852e-09\n",
       "4354                4.192971      5.044360e-04\n",
       "4355                3.358857      2.061833e-05\n",
       "4356                1.844520      3.014890e-05\n",
       "4357               -3.001874      2.762255e-06\n",
       "4358               -1.882020      3.528529e-04\n",
       "4359               -2.016672      4.233078e-02\n",
       "4360               -1.652164      8.799479e-03\n",
       "4361                2.678990      4.062568e-02\n",
       "4362                3.343057      1.712519e-02\n",
       "4363                4.877521      1.574210e-04\n",
       "4364                4.517112      9.530196e-03\n",
       "4365                1.548025      6.029857e-04\n",
       "4366   FAM106C     -1.610026      3.309201e-02\n",
       "4367                3.263966      9.363160e-09"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "prePathfindR_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "953ec01f",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "`n_processes` is set to `iterations` because `iterations` < `n_processes`\n",
      "\n",
      "There is already a directory named \"pathfindR_Results\".\n",
      "Writing the result to \"pathfindR_Results(7)\" not to overwrite any previous results.\n",
      "\n",
      "## Testing input\n",
      "\n",
      "The input looks OK\n",
      "\n",
      "## Processing input. Converting gene symbols,\n",
      "          if necessary (and if human gene symbols provided)\n",
      "\n",
      "Number of genes provided in input: 4367\n",
      "\n",
      "Number of genes in input after p-value filtering: 4367\n",
      "\n",
      "Warning message:\n",
      "\"Duplicated genes found! The lowest p value for each gene was selected\"\n",
      "pathfindR cannot handle p values < 1e-13. These were changed to 1e-13\n",
      "\n",
      "Could not find any interactions for 1199 (37.66%) genes in the PIN\n",
      "\n",
      "Final number of genes in input: 1980\n",
      "\n",
      "## Performing Active Subnetwork Search and Enrichment\n",
      "\n",
      "## Processing the enrichment results over all iterations\n",
      "\n",
      "## Annotating involved genes and visualizing enriched terms\n",
      "\n",
      "## Creating HTML report\n",
      "\n"
     ]
    },
    {
     "ename": "ERROR",
     "evalue": "Error in with_pandoc_safe_environment(system(paste(shQuote(path), \"--version\"), : The 'HOME' environment variable must be set before running Pandoc.\n",
     "output_type": "error",
     "traceback": [
      "Error in with_pandoc_safe_environment(system(paste(shQuote(path), \"--version\"), : The 'HOME' environment variable must be set before running Pandoc.\nTraceback:\n",
      "1. run_pathfindR(prePathfindR_df, visualize_enriched_terms = FALSE)",
      "2. rmarkdown::render(input = system.file(\"rmd\", \"results.Rmd\", package = \"pathfindR\"), \n .     output_dir = \".\")",
      "3. pandoc_available(required_pandoc, error = TRUE)",
      "4. find_pandoc()",
      "5. lapply(sources, function(src) {\n .     if (dir_exists(src)) \n .         get_pandoc_version(src)\n .     else numeric_version(\"0\")\n . })",
      "6. FUN(X[[i]], ...)",
      "7. get_pandoc_version(src)",
      "8. with_pandoc_safe_environment(system(paste(shQuote(path), \"--version\"), \n .     intern = TRUE))",
      "9. stop(\"The 'HOME' environment variable must be set before running Pandoc.\")"
     ]
    }
   ],
   "source": [
    "paths_df <- run_pathfindR(prePathfindR_df, visualize_enriched_terms =FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "8a8402f0",
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'paths_df' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'paths_df' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "paths_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "23cea69c",
   "metadata": {},
   "outputs": [],
   "source": [
    "p <- ggplot(paths_df, aes(x= Fold_Enrichment, y = support, color = support)) +\n",
    "            geom_point() + \n",
    "            scale_size(paths_df$occurrence) +\n",
    "            theme_classic() + \n",
    "            ggtitle(\"Speed Vs Dist coloured by speed\")\n",
    "p"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "df58f349",
   "metadata": {},
   "outputs": [],
   "source": [
    "p"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ebe1f05a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8a66eeb3",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "114d07cc",
   "metadata": {},
   "outputs": [],
   "source": [
    "save.image(file = \"model_env.RData\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9d4c75bc",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.1.2"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {
    "height": "731px",
    "left": "43px",
    "top": "92px",
    "width": "384px"
   },
   "toc_section_display": true,
   "toc_window_display": true
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
